--- a/Makefile	2016-10-19 13:31:56.946185587 +0200
+++ b/Makefile	2016-10-19 21:16:27.098274765 +0200
@@ -3,13 +3,10 @@
 endif
 
 ifeq ($(PLATFORM),rpi3)
-	CPU_FLAGS += -march=armv8-a -mfpu=neon-fp-armv8 -mfloat-abi=hard
 	MORE_CFLAGS += -DRASPBERRY -DCAPSLOCK_DEBIAN_WORKAROUND -DARMV6T2 -DUSE_ARMNEON
 else ifeq ($(PLATFORM),rpi2)
-	CPU_FLAGS += -march=armv7-a -mfpu=neon-vfpv4 -mfloat-abi=hard
 	MORE_CFLAGS += -DRASPBERRY -DCAPSLOCK_DEBIAN_WORKAROUND -DARMV6T2 -DUSE_ARMNEON
 else ifeq ($(PLATFORM),rpi1)
-	CPU_FLAGS += -march=armv6zk -mfpu=vfp -mfloat-abi=hard
 	MORE_CFLAGS += -DRASPBERRY -DCAPSLOCK_DEBIAN_WORKAROUND
 endif
 
@@ -29,16 +26,16 @@
 #GEN_PROFILE=1
 #USE_PROFILE=1
 
-DEFAULT_CFLAGS = $(CFLAGS) `sdl2-config --cflags`
+DEFAULT_CFLAGS = $(CFLAGS) `$(SYSROOT_PREFIX)/usr/bin/sdl2-config --cflags` -std=c++11
 
 MY_LDFLAGS = $(LDFLAGS)
-MY_LDFLAGS += -lSDL2 -lpthread -lz -lSDL2_image -lpng -lrt -lxml2 -lFLAC -lmpg123 -ldl
-MY_LDFLAGS += -lSDL2_ttf -lguisan -L/usr/local/lib -Lsrc/guisan 
+MY_LDFLAGS += -lSDL2 -lpthread -lz -lSDL2_image -lpng -lrt -lxml2 -lFLAC -lmpg123 -ldl -logg
+MY_LDFLAGS += -lSDL2_ttf -lguisan -L$(SYSROOT_PREFIX)/usr/lib
 
-MORE_CFLAGS += -I/usr/include/libxml2
+MORE_CFLAGS += -I$(SYSROOT_PREFIX)/usr/include/libxml2
 MORE_CFLAGS += -DPANDORA -DARMV6_ASSEMBLY -DWITH_INGAME_WARNING
 MORE_CFLAGS += -DCPU_arm -DUSE_SDL
-MORE_CFLAGS += -I/opt/vc/include -I/opt/vc/include/interface/vmcs_host/linux -I/opt/vc/include/interface/vcos/pthreads
+MORE_CFLAGS += -I$(SYSROOT_PREFIX)/usr/include -I$(SYSROOT_PREFIX)/usr/include/interface/vmcs_host/linux -I$(SYSROOT_PREFIX)/usr/include/interface/vcos/pthreads
 MORE_CFLAGS += -Isrc -Isrc/osdep -Isrc/td-sdl -Isrc/include 
 MORE_CFLAGS += -Wno-unused -Wno-format -Wno-write-strings -Wno-multichar
 #MORE_CFLAGS += -fuse-ld=gold
