diff --git a/Makefile.am b/Makefile.am
old mode 100644
new mode 100755
index 5bc486b9..1d5c8639
--- a/Makefile.am
+++ b/Makefile.am
@@ -39,8 +39,6 @@ endif
 SUBDIRS += overlay benchmarks
 endif
 
-SUBDIRS += docs
-
 MAINTAINERCLEANFILES = ChangeLog INSTALL
 
 .PHONY: ChangeLog INSTALL
diff --git a/assembler/Makefile.am b/assembler/Makefile.am
old mode 100644
new mode 100755
index 49422e02..07788f34
--- a/assembler/Makefile.am
+++ b/assembler/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = doc test
+SUBDIRS = test
 
 noinst_LTLIBRARIES = libbrw.la
 
diff --git a/autogen.sh b/autogen.sh
index 53df7525..65fcab2f 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -6,13 +6,6 @@ test -z "$srcdir" && srcdir=.
 ORIGDIR=`pwd`
 cd $srcdir
 
-if ! type gtkdocize > /dev/null 2>&1; then
-        echo "EXTRA_DIST =" > gtk-doc.make
-        echo "CLEANFILES =" >> gtk-doc.make
-else
-        gtkdocize || exit $?
-fi
-
 autoreconf -v --install || exit 1
 cd $ORIGDIR || exit $?
 
diff --git a/configure.ac b/configure.ac
old mode 100644
new mode 100755
index 1ee4e90e..016904f3
--- a/configure.ac
+++ b/configure.ac
@@ -43,14 +43,6 @@ AC_PROG_CC
 AC_PROG_LEX
 AC_PROG_YACC
 
-# check for gtk-doc
-m4_ifdef([GTK_DOC_CHECK], [
-GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
-],[
-AM_CONDITIONAL([ENABLE_GTK_DOC], false)
-enable_gtk_doc=no
-])
-
 # check for rst2man for generating man pages
 AC_CHECK_PROG(RST2MAN, rst2man, yes, no)
 AM_CONDITIONAL(HAVE_RST2MAN, [test "x$RST2MAN" = xyes])
@@ -382,10 +374,6 @@ AC_CONFIG_FILES([
 		 benchmarks/Makefile
 		 benchmarks/ezbench.d/Makefile
 		 benchmarks/wsim/Makefile
-		 docs/Makefile
-		 docs/reference/Makefile
-		 docs/reference/igt-gpu-tools/Makefile
-		 docs/reference/igt-gpu-tools/version.xml
 		 lib/Makefile
 		 lib/tests/Makefile
 		 man/Makefile
@@ -400,7 +388,6 @@ AC_CONFIG_FILES([
 		 debugger/Makefile
 		 debugger/system_routine/Makefile
 		 assembler/Makefile
-		 assembler/doc/Makefile
 		 assembler/test/Makefile
 		 assembler/intel-gen4asm.pc
 		 overlay/Makefile
@@ -428,7 +415,6 @@ echo "       Debugger           : ${enable_debugger}"
 echo "       Overlay            : X: ${enable_overlay_xlib}, Xv: ${enable_overlay_xvlib}"
 echo "       x86-specific tools : ${build_x86}"
 echo ""
-echo " • API-Documentation      : ${enable_gtk_doc}"
 echo " • Fail on warnings       : ${enable_werror}"
 echo ""
 
diff --git a/docs/reference/igt-gpu-tools/Makefile.am b/docs/reference/igt-gpu-tools/Makefile.am
old mode 100644
new mode 100755
index 11ada326..fb2c042c
--- a/docs/reference/igt-gpu-tools/Makefile.am
+++ b/docs/reference/igt-gpu-tools/Makefile.am
@@ -88,7 +88,7 @@ DOC_MAIN_SGML_FILE=$(DOC_MODULE)-docs.xml
 # gtk-doc will search all .c and .h files beneath these paths
 # for inline comments documenting functions and macros.
 # e.g. DOC_SOURCE_DIR=$(top_srcdir)/gtk $(top_srcdir)/gdk
-DOC_SOURCE_DIR=$(top_srcdir)/lib
+#DOC_SOURCE_DIR=$(top_srcdir)/lib
 
 # Extra options to pass to gtkdoc-scangobj. Not normally needed.
 SCANGOBJ_OPTIONS=
@@ -193,7 +193,7 @@ GTKDOC_CFLAGS=
 GTKDOC_LIBS=
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
-include $(top_srcdir)/gtk-doc.make
+#include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
