--- firefox-42.0.orig/tools/profiler/lul/LulElf.cpp	2015-10-29 23:18:01.000000000 +0100
+++ firefox-42.0/tools/profiler/lul/LulElf.cpp	2015-12-17 06:09:09.809776525 +0100
@@ -66,6 +66,9 @@
 #include "LulElfInt.h"
 #include "LulMainInt.h"
 
+#ifndef NT_GNU_BUILD_ID
+#define NT_GNU_BUILD_ID 3
+#endif
 
 #if defined(LUL_PLAT_arm_android) && !defined(SHT_ARM_EXIDX)
 // bionic and older glibsc don't define it
