--- firefox-42.0.orig/tools/profiler/core/platform-linux.cc	2015-10-29 23:18:01.000000000 +0100
+++ firefox-42.0/tools/profiler/core/platform-linux.cc	2015-12-17 05:41:35.091485672 +0100
@@ -57,7 +57,7 @@
 #include <fcntl.h>      // open
 #include <unistd.h>     // sysconf
 #include <semaphore.h>
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
 #include <execinfo.h>   // backtrace, backtrace_symbols
 #endif  // def __GLIBC__
 #include <strings.h>    // index
