diff -Naur irqbalance-v1.2.0-orig/configure.ac irqbalance-v1.2.0/configure.ac
--- irqbalance-v1.2.0-orig/configure.ac	2017-01-18 21:09:00.072325139 +0000
+++ irqbalance-v1.2.0/configure.ac	2017-01-18 21:09:20.972375105 +0000
@@ -39,7 +39,7 @@
   [test "x$with_systemd" = xyes], [
     PKG_CHECK_MODULES([SYSTEMD], [libsystemd], [journal_lib=yes], [journal_lib=no])
     AS_IF([test "x$journal_lib" != "xyes"], [
-      PKG_CHECK_MODULES([SYSTEMD], [libsystemd-journal], [journal_lib=yes])
+      PKG_CHECK_MODULES([SYSTEMD], [libsystemd], [journal_lib=yes])
     ])
     AC_DEFINE(HAVE_LIBSYSTEMD, 1, [systemd support])
     AC_CHECK_LIB([systemd], [sd_journal_print_with_location])
