diff --git a/configure.ac b/configure.ac
old mode 100644
new mode 100755
index eaa76c9..ab2af2e
--- a/configure.ac
+++ b/configure.ac
@@ -54,7 +54,7 @@ AS_IF(
   [test "x$with_systemd" = xyes], [
     PKG_CHECK_MODULES([SYSTEMD], [libsystemd], [journal_lib=yes], [journal_lib=no])
     AS_IF([test "x$journal_lib" != "xyes"], [
-      PKG_CHECK_MODULES([SYSTEMD], [libsystemd-journal], [journal_lib=yes])
+      PKG_CHECK_MODULES([SYSTEMD], [libsystemd], [journal_lib=yes])
     ])
     AC_DEFINE(HAVE_LIBSYSTEMD, 1, [systemd support])
     AC_CHECK_LIB([systemd], [sd_journal_print_with_location])
