Description: skip racy test
Author: James Clarke
Forwarded: no
Last-Update: 2016-08-11

---

--- neon27-0.30.1.orig/test/redirect.c
+++ neon27-0.30.1/test/redirect.c
@@ -181,7 +181,7 @@ static int no_redirect(void)
 
     ONN("redirect non-NULL after non-redir req", ne_redirect_location(sess));
 
-    CALL(process_redir(sess, "/foo", &loc));
+    PRECOND(process_redir(sess, "/foo", &loc) == OK);
     CALL(await_server());
 
     ne_session_destroy(sess);
