--- chromium-59.0.3071.86/third_party/WebKit/Source/platform/text/CharacterEmoji.cpp.system-icu	2017-06-05 21:03:26.000000000 +0200
+++ chromium-59.0.3071.86/third_party/WebKit/Source/platform/text/CharacterEmoji.cpp	2017-06-11 21:13:51.098773019 +0200
@@ -22,7 +22,7 @@
 // temporarily uing our own functions again. Compare crbug.com/628333 Other than
 // that: versions before 56 do not have an API for Emoji properties, but
 // Chrome's copy of ICU 56 does.
-#if defined(USING_SYSTEM_ICU) || (U_ICU_VERSION_MAJOR_NUM <= 57)
+#if defined(USING_SYSTEM_ICU) && (U_ICU_VERSION_MAJOR_NUM <= 63)
 // The following UnicodeSet patterns were compiled from
 // http://www.unicode.org/Public/emoji/3.0//emoji-data.txt
 
