diff --git a/src/libsystemd/sd-hwdb/sd-hwdb.c b/src/libsystemd/sd-hwdb/sd-hwdb.c
index b81786a64..c68b6973f 100644
--- a/src/libsystemd/sd-hwdb/sd-hwdb.c
+++ b/src/libsystemd/sd-hwdb/sd-hwdb.c
@@ -298,7 +298,7 @@ static int trie_search_f(sd_hwdb *hwdb, const char *search) {
 
 static const char hwdb_bin_paths[] =
         "/etc/systemd/hwdb/hwdb.bin\0"
-        "/etc/udev/hwdb.bin\0"
+        "/run/hwdb.bin\0"
         "/usr/lib/systemd/hwdb/hwdb.bin\0"
 #if HAVE_SPLIT_USR
         "/lib/systemd/hwdb/hwdb.bin\0"
diff --git a/src/udev/udevadm-hwdb.c b/src/udev/udevadm-hwdb.c
index e9a1af65f..947f82c83 100644
--- a/src/udev/udevadm-hwdb.c
+++ b/src/udev/udevadm-hwdb.c
@@ -18,7 +18,7 @@ static int help(void) {
                "  -V --version         Print version of the program\n"
                "  -u --update          Update the hardware database\n"
                "  -s --strict          When updating, return non-zero exit value on any parsing error\n"
-               "     --usr             Generate in " UDEVLIBEXECDIR " instead of /etc/udev\n"
+               "     --usr             Generate in " UDEVLIBEXECDIR " instead of /run\n"
                "  -t --test=MODALIAS   Query database and print result\n"
                "  -r --root=PATH       Alternative root path in the filesystem\n\n"
                "NOTE:\n"
