diff --git a/src/libsystemd/sd-hwdb/hwdb-util.c b/src/libsystemd/sd-hwdb/hwdb-util.c
index bfb3ae0e9..295c76fe3 100644
--- a/src/libsystemd/sd-hwdb/hwdb-util.c
+++ b/src/libsystemd/sd-hwdb/hwdb-util.c
@@ -17,7 +17,7 @@
 #include "string-util.h"
 #include "strv.h"
 
-static const char *default_hwdb_bin_dir = "/etc/udev";
+static const char *default_hwdb_bin_dir = "/run";
 static const char * const conf_file_dirs[] = {
         "/etc/udev/hwdb.d",
         UDEVLIBEXECDIR "/hwdb.d",
