commit 56abfa8a80b6768ff870eac4da74460195809733
Author: Lukas Rusak <lorusak@gmail.com>
Date:   Fri Jan 5 10:17:20 2018 -0800

    use /var/media/

diff --git a/src/mount/mount-tool.c b/src/mount/mount-tool.c
index 131f160b5..ffce1d15c 100644
--- a/src/mount/mount-tool.c
+++ b/src/mount/mount-tool.c
@@ -1144,7 +1144,7 @@ static int acquire_mount_where(struct udev_device *d) {
                 if (!filename_is_valid(escaped))
                         return 0;
 
-                arg_mount_where = strjoin("/run/media/system/", escaped);
+                arg_mount_where = strjoin("/var/media/", escaped);
         } else
                 arg_mount_where = strdup(v);
 
@@ -1278,7 +1278,7 @@ static int discover_loop_backing_file(void) {
                         return -EINVAL;
                 }
 
-                arg_mount_where = strjoin("/run/media/system/", escaped);
+                arg_mount_where = strjoin("/var/media/", escaped);
                 if (!arg_mount_where)
                         return log_oom();
 
