From 433db67f6bae394bb755a35ac50ffc98955af4bc Mon Sep 17 00:00:00 2001
From: Maarten Lankhorst <maarten.lankhorst@canonical.com>
Date: Sat, 31 Aug 2013 08:35:59 +0200
Subject: [PATCH 04/41] avrt: Add realtime to stub

---
 dlls/avrt/main.c | 1 +
 1 file changed, 1 insertion(+)

Index: wine1.6-1.7.1-actually1.6/dlls/avrt/main.c
===================================================================
--- wine1.6-1.7.1-actually1.6.orig/dlls/avrt/main.c	2013-09-13 16:19:05.000000000 -0700
+++ wine1.6-1.7.1-actually1.6/dlls/avrt/main.c	2013-09-13 16:20:04.000000000 -0700
@@ -80,6 +80,7 @@
         SetLastError(ERROR_INVALID_HANDLE);
         return NULL;
     }
+    SetThreadPriority(GetCurrentThread(), THREAD_PRIORITY_TIME_CRITICAL);
     return (HANDLE)0x12345678;
 }
 
