From 707d4aeae111fa01b393333bca419311e5e5ce36 Mon Sep 17 00:00:00 2001
From: Maarten Lankhorst <m.b.lankhorst@gmail.com>
Date: Sat, 31 Aug 2013 08:36:01 +0200
Subject: [PATCH 16/41] fix fdels trailing whitespaces

Happy? :P
---
 dlls/winepulse.drv/mmdevdrv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: wine1.6-1.7.1-actually1.6/dlls/winepulse.drv/mmdevdrv.c
===================================================================
--- wine1.6-1.7.1-actually1.6.orig/dlls/winepulse.drv/mmdevdrv.c	2013-09-13 16:19:09.000000000 -0700
+++ wine1.6-1.7.1-actually1.6/dlls/winepulse.drv/mmdevdrv.c	2013-09-13 16:19:56.000000000 -0700
@@ -1898,7 +1898,7 @@
     TRACE("(%p)->(%p)\n", This, frames);
     if (!frames)
         return E_POINTER;
-    
+
     pthread_mutex_lock(&pulse_lock);
     ACImpl_GetCapturePad(This, NULL);
     p = This->locked_ptr;
@@ -1998,7 +1998,7 @@
         else
             *pos += This->pad;
     }
-    
+
     /* Make time never go backwards */
     if (*pos < This->clock_lastpos)
         *pos = This->clock_lastpos;
