From ff09d89482055edfcfcea3b34cee4e82b09ae2da Mon Sep 17 00:00:00 2001
From: Maarten Lankhorst <maarten.lankhorst@canonical.com>
Date: Sat, 31 Aug 2013 08:36:02 +0200
Subject: [PATCH 33/41] dsound: mix float natively

---
 dlls/dsound/mixer.c | 43 ++++++++++++++++++++++++++++++++-----------
 1 file changed, 32 insertions(+), 11 deletions(-)

Index: wine1.6-1.7.1-actually1.6/dlls/dsound/mixer.c
===================================================================
--- wine1.6-1.7.1-actually1.6.orig/dlls/dsound/mixer.c	2013-09-13 16:19:21.371219810 -0700
+++ wine1.6-1.7.1-actually1.6/dlls/dsound/mixer.c	2013-09-13 16:19:45.000000000 -0700
@@ -643,6 +643,7 @@
 	if (device->priolevel != DSSCL_WRITEPRIMARY) {
 		BOOL all_stopped = FALSE;
 		int nfiller;
+		BOOL native = device->normfunction == normfunctions[4];
 		DWORD bpp = device->pwfx->wBitsPerSample>>3;
 
 		/* the sound of silence */
@@ -657,22 +658,42 @@
 			TRACE("Buffer restarting\n");
 		}
 
-		memset(device->mix_buffer, nfiller, maxq);
+		if (native) {
+			void *buffer = NULL;
 
-		/* do the mixing */
-		DSOUND_MixToPrimary(device, device->mix_buffer, writepos, maxq, &all_stopped);
+			hr = IAudioRenderClient_GetBuffer(device->render, maxq / block, (void*)&buffer);
+			if(FAILED(hr)){
+				WARN("GetBuffer failed: %08x\n", hr);
+				LeaveCriticalSection(&device->mixlock);
+				return;
+			}
+			memset(buffer, nfiller, maxq);
+
+			DSOUND_MixToPrimary(device, buffer, writepos, maxq, &all_stopped);
+
+			hr = IAudioRenderClient_ReleaseBuffer(device->render, maxq / block, 0);
+			if(FAILED(hr))
+				ERR("ReleaseBuffer failed: %08x\n", hr);
 
-		if (maxq + writepos > device->buflen) {
-			DWORD todo = device->buflen - writepos;
+			device->pad += maxq;
+		} else {
+			memset(device->mix_buffer, nfiller, maxq);
 
-			device->normfunction(device->mix_buffer, device->buffer + writepos, todo);
-			DSOUND_WaveQueue(device, device->buffer + writepos, todo);
+			/* do the mixing */
+			DSOUND_MixToPrimary(device, device->mix_buffer, writepos, maxq, &all_stopped);
 
-			device->normfunction(device->mix_buffer + todo / bpp, device->buffer, (maxq - todo));
-			DSOUND_WaveQueue(device, device->buffer, maxq - todo);
-		} else {
-			device->normfunction(device->mix_buffer, device->buffer + writepos, maxq);
-			DSOUND_WaveQueue(device, device->buffer + writepos, maxq);
+			if (maxq + writepos > device->buflen) {
+				DWORD todo = device->buflen - writepos;
+
+				device->normfunction(device->mix_buffer, device->buffer + writepos, todo);
+				DSOUND_WaveQueue(device, device->buffer + writepos, todo);
+
+				device->normfunction(device->mix_buffer + todo / bpp, device->buffer, (maxq - todo));
+				DSOUND_WaveQueue(device, device->buffer, maxq - todo);
+			} else {
+				device->normfunction(device->mix_buffer, device->buffer + writepos, maxq);
+				DSOUND_WaveQueue(device, device->buffer + writepos, maxq);
+			}
 		}
 
 		if (maxq) {
