diff --git a/speedtest.go b/speedtest.go
old mode 100644
new mode 100755
index 6b0ed62..4787cd9
--- a/speedtest.go
+++ b/speedtest.go
@@ -41,7 +41,7 @@ func runTest(c *cli.Context, stClient *sthttp.Client, tester *tests.Tester) {
 
 	// if we are *not* running a report then say hello to everyone
 	if !tester.Report {
-		fmt.Printf("github.com/zpeters/speedtest -- unofficial cli for speedtest.net\n")
+		fmt.Printf("● ● ● ● ● ● ●\n")
 	}
 
 	// if we are in debug mode print outa an environment report
