diff -Naurp requests-2.18.4/requests/certs.py requests-2.18.4-patch/requests/certs.py
--- requests-2.18.4/requests/certs.py	2017-08-15 16:22:52.000000000 +0300
+++ requests-2.18.4-patch/requests/certs.py	2017-10-26 09:54:56.000000000 +0300
@@ -12,7 +12,8 @@ If you are packaging Requests, e.g., for
 environment, you can change the definition of where() to return a separately
 packaged CA bundle.
 """
-from certifi import where
+def where():
+    return "/etc/ssl/cert.pem"
 
 if __name__ == '__main__':
     print(where())
diff -Naurp requests-2.18.4/setup.py requests-2.18.4-patch/setup.py
--- requests-2.18.4/setup.py	2017-08-15 16:22:52.000000000 +0300
+++ requests-2.18.4-patch/setup.py	2017-10-26 09:10:39.000000000 +0300
@@ -44,8 +44,7 @@ packages = ['requests']
 requires = [
     'chardet>=3.0.2,<3.1.0',
     'idna>=2.5,<2.7',
-    'urllib3>=1.21.1,<1.23',
-    'certifi>=2017.4.17'
+    'urllib3>=1.21.1,<1.23'
 
 ]
 test_requirements = ['pytest-httpbin==0.0.7', 'pytest-cov', 'pytest-mock', 'pytest-xdist', 'PySocks>=1.5.6, !=1.5.7', 'pytest>=2.8.0']
