diff -Naur nss-3.37.3-orig/nss/lib/freebl/Makefile nss-3.37.3/nss/lib/freebl/Makefile
--- nss-3.37.3-orig/nss/lib/freebl/Makefile	2018-06-08 05:44:34.042760726 +0300
+++ nss-3.37.3/nss/lib/freebl/Makefile	2018-06-05 13:02:17.000000000 +0300
@@ -541,9 +541,7 @@
     ECL_SRCS += curve25519_32.c
 endif
 
-ifndef HAVE_INT128_SUPPORT
-    VERIFIED_SRCS += FStar.c
-endif
+VERIFIED_SRCS += FStar.c
 
 #######################################################################
 # (5) Execute "global" rules. (OPTIONAL)                              #
