--- gtk+-3.22.15/gtk/gtkstylecontext.c~	2017-04-21 22:36:22.000000000 +0000
+++ gtk+-3.22.15/gtk/gtkstylecontext.c	2017-06-14 16:05:29.694158645 +0000
@@ -343,8 +343,10 @@
 
   priv->screen = gdk_screen_get_default ();
 
-  if (priv->screen == NULL)
-    g_error ("Can't create a GtkStyleContext without a display connection");
+  if (priv->screen == NULL) {
+    g_warning ("Can't create a GtkStyleContext without a display connection");
+    exit(1);
+  }
 
   priv->property_cache = g_array_new (FALSE, FALSE, sizeof (PropertyValue));
 
