From 602a8395bd925dac53a240588b70f6940e210577 Mon Sep 17 00:00:00 2001
From: Victor Rodriguez <victor.rodriguez.bahena@intel.com>
Date: Thu, 11 Jun 2015 15:40:16 +0000
Subject: [PATCH] disable update-copyright if perl is too new

Signed-off-by: Victor Rodriguez <victor.rodriguez.bahena@intel.com>
---
 tests/test-update-copyright.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/test-update-copyright.sh b/tests/test-update-copyright.sh
index 60cb372..d5b50f4 100755
--- a/tests/test-update-copyright.sh
+++ b/tests/test-update-copyright.sh
@@ -51,10 +51,10 @@ echo a > $TMP-in
     exit 77
   }
 
-# Skip this test if Perl is too old.  FIXME: 5.8.0 is just a guess.
+# Skip this test if Perl is too old or new.  FIXME: 5.8.0 is just a guess.
 # We have a report that 5.6.1 is inadequate and that 5.8.0 works.
-perl -e 'require 5.8.0' || {
-  echo '$0: skipping this test; Perl version is too old' 1>&2
+perl -e 'require 5.8.0' || perl -e 'require 5.22.0'{
+  echo '$0: skipping this test; Perl version is too old or too new' 1>&2
   exit 77
 }
 
-- 
2.4.3

