--- e2fsprogs-1.43.2.orig/util/subst.c	2016-06-07 07:01:19.000000000 +0200
+++ e2fsprogs-1.43.2/util/subst.c	2016-09-19 13:21:04.000000000 +0200
@@ -22,9 +22,7 @@
 #ifdef HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
-#ifdef HAVE_SYS_STAT_H
 #include <sys/stat.h>
-#endif
 #include <fcntl.h>
 #include <time.h>
 #include <utime.h>
@@ -389,7 +387,7 @@ int main(int argc, char **argv)
 		if (fd > 0) {
 			/* save the original atime, if possible */
 			if (fstat(fd, &stbuf) == 0) {
-#if HAVE_STRUCT_STAT_ST_ATIM
+#if defined(HAVE_STRUCT_STAT_ST_ATIM) && !defined(__APPLE__)
 				tv[0].tv_sec = stbuf.st_atim.tv_sec;
 				tv[0].tv_usec = stbuf.st_atim.tv_nsec / 1000;
 #else
