	When search for wireless-tools, also check $PATH
--- aircrack-ng-1.2-rc2.orig/src/osdep/linux.c	2015-04-09 06:47:28.000000000 +0200
+++ aircrack-ng-1.2-rc2/src/osdep/linux.c	2015-11-07 21:28:22.000000000 +0100
@@ -264,6 +264,7 @@ static char * wiToolsPath(const char * t
                 "/usr/local/bin",
                 "/tmp"
         };
+	char *envpath, *token;
 
 	// Also search in other known location just in case we haven't found it yet
 	nbelems = sizeof(paths) / sizeof(char *);
@@ -274,6 +275,14 @@ static char * wiToolsPath(const char * t
 			return path;
 	}
 
+	/* search again in $PATH */
+	for (envpath = getenv("PATH"); ; envpath = NULL) {
+		if ((token = strtok(envpath, ":")) == NULL)
+			break;
+		if ((path = searchInside(token, tool)) != NULL)
+			return path;
+	}
+
         return NULL;
 }
 
