--- libtasn1-4.9.orig/Makefile.in	2016-07-25 22:28:16.000000000 +0200
+++ libtasn1-4.9/Makefile.in	2016-08-16 14:45:03.404601380 +0200
@@ -222,7 +222,7 @@ am__define_uniq_tagged_files = \
 ETAGS = etags
 CTAGS = ctags
 CSCOPE = cscope
-DIST_SUBDIRS = gl lib src examples tests doc
+DIST_SUBDIRS = gl lib src
 am__DIST_COMMON = $(srcdir)/Makefile.in $(srcdir)/config.h.in \
 	$(top_srcdir)/build-aux/ar-lib $(top_srcdir)/build-aux/compile \
 	$(top_srcdir)/build-aux/config.guess \
@@ -934,7 +934,7 @@ EXTRA_DIST = windows/asn1-parser/asn1-pa
 	windows/test-parser/test-parser.vcproj \
 	windows/test-tree/test-tree.vcproj windows/libtasn14win.mk \
 	cfg.mk maint.mk .clcopying
-SUBDIRS = gl lib src examples tests $(am__append_1)
+SUBDIRS = gl lib src
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4 -I lib/glm4
 TMPFILE = "abi-temp.xml"
 all: config.h
