--- linux-2.4.5.orig/drivers/char/pc_keyb.c	Fri Sep  7 04:36:57 2001
+++ linux-2.4.5/drivers/char/pc_keyb.c	Fri Sep  7 09:02:51 2001
@@ -838,7 +839,7 @@
 /*
  * Check if this is a dual port controller.
  */
-static int __init detect_auxiliary_port(void)
+int detect_auxiliary_port(void)
 {
 	unsigned long flags;
 	int loops = 10;
--- linux-2.4.5.orig/arch/i386/kernel/i386_ksyms.c	Fri Sep  7 04:35:48 2001
+++ linux-2.4.5/arch/i386/kernel/i386_ksyms.c	Fri Sep  7 07:10:40 2001
@@ -30,6 +30,7 @@
 #include <asm/pgalloc.h>
 #include <asm/page.h>
 
+extern int detect_auxiliary_port(void);
 extern void dump_thread(struct pt_regs *, struct user *);
 extern spinlock_t rtc_lock;
 
@@ -170,4 +171,6 @@
 EXPORT_SYMBOL(do_BUG);
 #endif
 
-
+#ifdef CONFIG_PSMOUSE
+EXPORT_SYMBOL(detect_auxiliary_port);
+#endif
\ No newline at end of file
