--- chromium-64.0.3282.71/third_party/WebKit/Source/platform/wtf/text/TextCodecICU.cpp.gcc7	2018-01-04 00:06:33.000000000 +0100
+++ chromium-64.0.3282.71/third_party/WebKit/Source/platform/wtf/text/TextCodecICU.cpp	2018-01-07 10:30:57.354614167 +0100
@@ -594,24 +594,6 @@
   UCNV_FROM_U_CALLBACK_ESCAPE(context, from_unicode_args, code_units, length,
                               code_point, reason, err);
 }
-
-static void GbkCallbackSubstitute(const void* context,
-                                  UConverterFromUnicodeArgs* from_unicode_args,
-                                  const UChar* code_units,
-                                  int32_t length,
-                                  UChar32 code_point,
-                                  UConverterCallbackReason reason,
-                                  UErrorCode* err) {
-  UChar out_char;
-  if (reason == UCNV_UNASSIGNED && (out_char = FallbackForGBK(code_point))) {
-    const UChar* source = &out_char;
-    *err = U_ZERO_ERROR;
-    ucnv_cbFromUWriteUChars(from_unicode_args, &source, source + 1, 0, err);
-    return;
-  }
-  UCNV_FROM_U_CALLBACK_SUBSTITUTE(context, from_unicode_args, code_units,
-                                  length, code_point, reason, err);
-}
 #endif  // USING_SYSTEM_ICU
 
 class TextCodecInput final {
