diff -Naur chromium-66.0.3359.117-orig/third_party/pdfium/fxjs/cjs_global.cpp chromium-66.0.3359.117/third_party/pdfium/fxjs/cjs_global.cpp
--- chromium-66.0.3359.117-orig/third_party/pdfium/fxjs/cjs_global.cpp	2018-04-25 09:25:41.767126823 +0300
+++ chromium-66.0.3359.117/third_party/pdfium/fxjs/cjs_global.cpp	2018-04-25 09:30:03.025118501 +0300
@@ -353,43 +353,6 @@
   }
 }
 
-void CJS_Global::CommitGlobalPersisitentVariables(CJS_Runtime* pRuntime) {
-  for (const auto& iter : m_MapGlobal) {
-    ByteString name = iter.first;
-    JSGlobalData* pData = iter.second.get();
-    if (pData->bDeleted) {
-      m_pGlobalData->DeleteGlobalVariable(name);
-      continue;
-    }
-    switch (pData->nType) {
-      case JS_GlobalDataType::NUMBER:
-        m_pGlobalData->SetGlobalVariableNumber(name, pData->dData);
-        m_pGlobalData->SetGlobalVariablePersistent(name, pData->bPersistent);
-        break;
-      case JS_GlobalDataType::BOOLEAN:
-        m_pGlobalData->SetGlobalVariableBoolean(name, pData->bData);
-        m_pGlobalData->SetGlobalVariablePersistent(name, pData->bPersistent);
-        break;
-      case JS_GlobalDataType::STRING:
-        m_pGlobalData->SetGlobalVariableString(name, pData->sData);
-        m_pGlobalData->SetGlobalVariablePersistent(name, pData->bPersistent);
-        break;
-      case JS_GlobalDataType::OBJECT: {
-        CJS_GlobalVariableArray array;
-        v8::Local<v8::Object> obj =
-            v8::Local<v8::Object>::New(GetIsolate(), pData->pData);
-        ObjectToArray(pRuntime, obj, array);
-        m_pGlobalData->SetGlobalVariableObject(name, array);
-        m_pGlobalData->SetGlobalVariablePersistent(name, pData->bPersistent);
-      } break;
-      case JS_GlobalDataType::NULLOBJ:
-        m_pGlobalData->SetGlobalVariableNull(name);
-        m_pGlobalData->SetGlobalVariablePersistent(name, pData->bPersistent);
-        break;
-    }
-  }
-}
-
 void CJS_Global::ObjectToArray(CJS_Runtime* pRuntime,
                                v8::Local<v8::Object> pObj,
                                CJS_GlobalVariableArray& array) {
