diff -Naur rpcbind-1.2.5-orig/systemd/rpcbind.service.in rpcbind-1.2.5/systemd/rpcbind.service.in
--- rpcbind-1.2.5-orig/systemd/rpcbind.service.in	2018-10-10 06:22:55.087580148 +0300
+++ rpcbind-1.2.5/systemd/rpcbind.service.in	2018-08-15 17:51:19.000000000 +0300
@@ -12,7 +12,7 @@
 [Service]
 Type=notify
 # distro can provide a drop-in adding EnvironmentFile=-/??? if needed.
-ExecStart=@_sbindir@/rpcbind $RPCBIND_OPTIONS -w -f
+ExecStart=@_sbindir@/rpcbind $RPCBIND_OPTIONS -f
 
 [Install]
 WantedBy=multi-user.target
diff -Naur rpcbind-1.2.5-orig/systemd/rpcbind.service.in rpcbind-1.2.5/systemd/rpcbind.service.in
--- rpcbind-1.2.5-orig/systemd/rpcbind.service.in	2019-04-29 05:58:13.916246206 +0300
+++ rpcbind-1.2.5/systemd/rpcbind.service.in	2019-04-29 05:58:06.989579600 +0300
@@ -6,7 +6,7 @@
 
 # Make sure we use the IP addresses listed for
 # rpcbind.socket, no matter how this unit is started.
-Requires=rpcbind.socket
+Requires=rpcbind.socket systemd-tmpfiles-setup.service
 Wants=rpcbind.target
 
 [Service]
