--- a/runtime/defaults.vim	2017-09-11 15:44:48.176302471 +0200
+++ b/runtime/defaults.vim	2017-09-11 15:44:59.582753332 +0200
@@ -75,9 +75,9 @@
 
 " In many terminal emulators the mouse works just fine.  By enabling it you
 " can position the cursor, Visually select and scroll with the mouse.
-if has('mouse')
-  set mouse=a
-endif
+"if has('mouse')
+"  set mouse=a
+"endif
 
 " Switch syntax highlighting on when the terminal has colors or when using the
 " GUI (which always has colors).
