diff --git a/CMakeLists.txt b/CMakeLists.txt
old mode 100644
new mode 100755
index b89be42..a7e5e50
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -318,7 +318,7 @@ else()
     else()
         set(CMAKE_REQUIRED_FLAGS "${PCLMULFLAG}")
     endif()
-    if(NOT (APPLE AND ${ARCH} MATCHES "i386"))
+    if(YES (APPLE AND ${ARCH} MATCHES "i386"))
         # the pclmul code currently crashes on Mac in 32bit mode. Avoid for now.
         check_c_source_compile_or_run(
             "#include <immintrin.h>
