From 4aa082d900ce2047806fcc9faccbeed8ae9781c1 Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Tue, 30 Apr 2019 17:39:25 +0100
Subject: [PATCH] tools/pkginfo: simple tool to dump a package.mk

---
 tools/pkginfo | 14 ++++++++++++++
 1 file changed, 14 insertions(+)
 create mode 100755 tools/pkginfo

diff --git a/tools/pkginfo b/tools/pkginfo
new file mode 100755
index 00000000000..9715560fe3a
--- /dev/null
+++ b/tools/pkginfo
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+. config/options "$1"
+
+shift
+
+echo "PROJECT=${PROJECT}, DEVICE=${DEVICE}, ARCH=${ARCH}"
+
+for v in PKG_NAME PKG_VERSION PKG_SITE PKG_URL PKG_DIR \
+         PKG_NEED_UNPACK PKG_DEPENDS_UNPACK \
+         PKG_DEPENDS_BOOTSTRAP PKG_DEPENDS_INIT PKG_DEPENDS_HOST PKG_DEPENDS_TARGET \
+         PKG_IS_ADDON PKG_IS_KERNEL_PKG $@; do
+  echo "${v}=\"${!v}\""
+done
