From 0ccda584b4e5ae28a23c2e424fe99e8188dc4f12 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=B4me=20Benoit?= <jerome.benoit@piment-noir.org>
Date: Fri, 18 Nov 2016 14:01:23 +0100
Subject: [PATCH 1/2] config/functions: add get_pkg_version helper function
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Jérôme Benoit <jerome.benoit@piment-noir.org>
---
 config/functions | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/config/functions b/config/functions
index 670185beed..518204b4d4 100644
--- a/config/functions
+++ b/config/functions
@@ -137,6 +137,14 @@ get_build_dir() {
   fi
 }
 
+get_pkg_version() {
+  if [ ! -z $1 ] ; then
+    cd $ROOT
+    . config/options $1
+    echo "$PKG_VERSION"
+  fi
+}
+
 tolower(){
   echo "$@" | tr ABCDEFGHIJKLMNOPQRSTUVWXYZ abcdefghijklmnopqrstuvwxyz
 }

From 86832fc2e90107a7d42d45d66f7175d7e250530a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=B4me=20Benoit?= <jerome.benoit@piment-noir.org>
Date: Fri, 18 Nov 2016 14:01:59 +0100
Subject: [PATCH 2/2] cache: add a compiler signature to the cache in order to
 handle compiler upgrade flawlessly        and different toochains in the
 build process
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

    Signed-off-by: Jérôme Benoit <jerome.benoit@piment-noir.org>

Signed-off-by: Jérôme Benoit <jerome.benoit@piment-noir.org>
---
 packages/devel/ccache/package.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packages/devel/ccache/package.mk b/packages/devel/ccache/package.mk
index 0822fc033f..59a8ec7312 100644
--- a/packages/devel/ccache/package.mk
+++ b/packages/devel/ccache/package.mk
@@ -40,6 +40,7 @@ post_makeinstall_host() {
 # setup ccache
   if [ -z "$CCACHE_DISABLE" ]; then
     $ROOT/$TOOLCHAIN/bin/ccache --max-size=$CCACHE_CACHE_SIZE
+    $ROOT/$TOOLCHAIN/bin/ccache --set-config=compiler_check=string:$(gcc -dumpversion)-$(get_pkg_version gcc)
   fi
 
   cat > $ROOT/$TOOLCHAIN/bin/host-gcc <<EOF
