From 8aef8f3371b7865cb9f55a2582335cc7b3747f30 Mon Sep 17 00:00:00 2001
From: "Sascha Kuehndel (InuSasha)" <dev@inusasha.de>
Date: Thu, 8 Mar 2018 10:16:13 +0100
Subject: [PATCH] build: remove unnessary exports

---
 config/functions | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/config/functions b/config/functions
index a1d764b717b..0b5e84410c8 100644
--- a/config/functions
+++ b/config/functions
@@ -1,34 +1,34 @@
 setup_toolchain() {
   # lto flag
   if flag_enabled "lto" "$LTO_SUPPORT" "only-disable"; then
-    export TARGET_CFLAGS+=" $CFLAGS_OPTIM_LTO"
-    export TARGET_CXXFLAGS+=" $CXXFLAGS_OPTIM_LTO"
-    export TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_LTO"
+    TARGET_CFLAGS+=" $CFLAGS_OPTIM_LTO"
+    TARGET_CXXFLAGS+=" $CXXFLAGS_OPTIM_LTO"
+    TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_LTO"
   fi
 
   # gold flag
   if flag_enabled "gold" "$GOLD_SUPPORT" "only-disable"; then
-    export TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_GOLD"
+    TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_GOLD"
   fi
 
   # position-independent code
   if flag_enabled "pic" "no"; then
-    export TARGET_CFLAGS+=" $CFLAGS_OPTIM_PIC"
-    export TARGET_CXXFLAGS+=" $CXXFLAGS_OPTIM_PIC"
-    export TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_PIC"
+    TARGET_CFLAGS+=" $CFLAGS_OPTIM_PIC"
+    TARGET_CXXFLAGS+=" $CXXFLAGS_OPTIM_PIC"
+    TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_PIC"
   fi
   if flag_enabled "pic:host" "no"; then
-    export HOST_CFLAGS+=" $CFLAGS_OPTIM_PIC"
-    export HOST_CXXFLAGS+=" $CXXFLAGS_OPTIM_PIC"
-    export HOST_LDFLAGS+=" $LDFLAGS_OPTIM_PIC"
+    HOST_CFLAGS+=" $CFLAGS_OPTIM_PIC"
+    HOST_CXXFLAGS+=" $CXXFLAGS_OPTIM_PIC"
+    HOST_LDFLAGS+=" $LDFLAGS_OPTIM_PIC"
   fi
 
   # hardening support
   if flag_enabled "hardening" "$HARDENING_SUPPORT"; then
-    export TARGET_CFLAGS+=" $CFLAGS_OPTIM_HARDENING"
-    export TARGET_CXXFLAGS+=" $CXXFLAGS_OPTIM_HARDENING"
-    export TARGET_CFLAGS+=" $CPPFLAGS_OPTIM_HARDENING"
-    export TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_HARDENING"
+    TARGET_CFLAGS+=" $CFLAGS_OPTIM_HARDENING"
+    TARGET_CXXFLAGS+=" $CXXFLAGS_OPTIM_HARDENING"
+    TARGET_CFLAGS+=" $CPPFLAGS_OPTIM_HARDENING"
+    TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_HARDENING"
   fi
 
   # parallel
