From e150120b61d41d207e5b12b3b900c1baac58b13e Mon Sep 17 00:00:00 2001
From: thoradia <thoradia.quack@gmail.com>
Date: Fri, 13 Oct 2017 01:46:08 +0200
Subject: [PATCH] scripts/build: provide for Python packages

---
 scripts/build | 59 ++++++++++++++++++++++++++++++++++++++++++++++++++++-------
 1 file changed, 52 insertions(+), 7 deletions(-)

diff --git a/scripts/build b/scripts/build
index 43f9fbed9d6..22dae51e3d1 100755
--- a/scripts/build
+++ b/scripts/build
@@ -170,6 +170,11 @@ if [ ! -f $STAMP ]; then
       fi
     fi
 
+    if [ "$PKG_IS_PYTHON" = "yes" ]; then
+      PKG_DEPENDS_HOST="toolchain distutilscross:host $PKG_DEPENDS_HOST"
+      PKG_DEPENDS_TARGET="toolchain distutilscross:host Python $PKG_DEPENDS_TARGET"
+    fi
+
     if [ "$TARGET" = "target" ]; then
       for p in $PKG_DEPENDS_TARGET; do
         $SCRIPTS/build $p
@@ -303,11 +308,28 @@ if [ ! -f $STAMP ]; then
         make_$TARGET
       else
         if [ "$TARGET" = "target" ]; then
-          echo "Executing (target): make $PKG_MAKE_OPTS_TARGET" | tr -s " "
-          make $PKG_MAKE_OPTS_TARGET
+          if [ "$PKG_IS_PYTHON" = "yes" ]; then
+            if [ "$PKG_PYTHON_CROSS_COMPILE" != "no" ]; then
+              PKG_PYTHON_OPTS_TARGET="--cross-compile"
+            fi
+            echo "Executing (target): python setup.py build $PKG_PYTHON_OPTS_TARGET" | tr -s " "
+            mkdir -p ".$TARGET_NAME"
+            cd ".$TARGET_NAME"
+            cp -r ../* .
+            export LDSHARED="$CC -shared $LDSHARED"
+            export PYTHONXCPREFIX="$SYSROOT_PREFIX/usr"
+            python setup.py build $PKG_PYTHON_OPTS_TARGET
+            mkdir -p "$INSTALL/lib/python"
+            PYTHONPATH="$INSTALL/lib/python" python setup.py install --home="$INSTALL"
+          else
+            echo "Executing (target): make $PKG_MAKE_OPTS_TARGET" | tr -s " "
+            make $PKG_MAKE_OPTS_TARGET
+          fi
         elif [ "$TARGET" = "host" ]; then
-          echo "Executing (host): make $PKG_MAKE_OPTS_HOST" | tr -s " "
-          make $PKG_MAKE_OPTS_HOST
+          if [ "$PKG_IS_PYTHON" != "yes" ]; then
+            echo "Executing (host): make $PKG_MAKE_OPTS_HOST" | tr -s " "
+            make $PKG_MAKE_OPTS_HOST
+          fi
         elif [ "$TARGET" = "init" ]; then
           echo "Executing (init): make $PKG_MAKE_OPTS_INIT" | tr -s " "
           make $PKG_MAKE_OPTS_INIT
@@ -328,10 +350,33 @@ if [ ! -f $STAMP ]; then
         makeinstall_$TARGET
       else
         if [ "$TARGET" = "target" ]; then
-          $MAKEINSTALL $PKG_MAKEINSTALL_OPTS_TARGET
-          make install DESTDIR=$INSTALL $PKG_MAKEINSTALL_OPTS_TARGET
+          if [ "$PKG_IS_PYTHON" = "yes" ]; then
+            for egg in "$INSTALL"/lib/python/*.egg; do
+              echo "Installing $(basename $egg)"
+              if [ -d "$egg" ]; then
+                cp -r "$egg"/* "$INSTALL/lib"
+              else
+                unzip -oq "$egg" -d "$INSTALL/lib"
+              fi
+            done
+            rm -fr "$INSTALL/lib/python" \
+                   "$INSTALL/lib/EGG-INFO"
+            find $INSTALL/lib -name "*.py" -exec rm -rf "{}" ";"
+          else
+            $MAKEINSTALL $PKG_MAKEINSTALL_OPTS_TARGET
+            make install DESTDIR=$INSTALL $PKG_MAKEINSTALL_OPTS_TARGET
+          fi
         elif [ "$TARGET" = "host" ]; then
-          make install $PKG_MAKEINSTALL_OPTS_HOST
+          if [ "$PKG_IS_PYTHON" = "yes" ]; then
+            mkdir -p ".$HOST_NAME"
+            cd ".$HOST_NAME"
+            cp -r ../* .
+            export LDSHARED="$CC -shared $LDSHARED"
+            python setup.py build
+            python setup.py install
+          else
+            make install $PKG_MAKEINSTALL_OPTS_HOST
+          fi
         elif [ "$TARGET" = "init" ]; then
           make install DESTDIR=$INSTALL $PKG_MAKEINSTALL_OPTS_INIT
         elif [ "$TARGET" = "bootstrap" ]; then
