diff --git a/CMakeLists.txt b/CMakeLists.txt
old mode 100644
new mode 100755
index 3c638cb2..9d42370f
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -164,7 +164,7 @@ INCLUDE(crc32-vpmsum)
 
 # Handle options
 OPTION(DISABLE_SHARED 
- "Don't build shared libraries, compile code as position-dependent" OFF)
+ "Don't build shared libraries, compile code as position-dependent" ON)
 IF(DISABLE_SHARED)
   SET(WITHOUT_DYNAMIC_PLUGINS 1)
 ENDIF()
