diff --git a/Makefile.am b/Makefile.am
old mode 100644
new mode 100755
index 1540bb7..c839eb1
--- a/Makefile.am
+++ b/Makefile.am
@@ -45,7 +45,6 @@ EXTRA_DIST = \
 SUBDIRS = \
 	lib \
 	src \
-	doc \
 	examples \
 	po \
 	tests \
diff --git a/configure.ac b/configure.ac
old mode 100644
new mode 100755
index 5a29b79..901edca
--- a/configure.ac
+++ b/configure.ac
@@ -158,7 +158,6 @@ AC_REPLACE_FUNCS(reallocarray)
 
 AC_CONFIG_FILES(
 Makefile
-doc/Makefile
 examples/Makefile
 examples/fastwc/Makefile
 examples/manual/Makefile
