diff -Naur a/cmake/modules/CrossCompile.cmake b/cmake/modules/CrossCompile.cmake
--- a/cmake/modules/CrossCompile.cmake	2017-12-13 01:02:13.000000000 -0800
+++ b/cmake/modules/CrossCompile.cmake	2018-06-13 23:41:11.303987580 -0700
@@ -6,9 +6,10 @@
     message(STATUS "Setting native build dir to " ${LLVM_${target_name}_BUILD})
   endif(NOT DEFINED LLVM_${target_name}_BUILD)
 
-  if (EXISTS ${LLVM_MAIN_SRC_DIR}/cmake/platforms/${toolchain}.cmake)
+  if (EXISTS ${CMAKE_HOST_TOOLCHAIN_FILE})
     set(CROSS_TOOLCHAIN_FLAGS_INIT
-      -DCMAKE_TOOLCHAIN_FILE=\"${LLVM_MAIN_SRC_DIR}/cmake/platforms/${toolchain}.cmake\")
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_HOST_TOOLCHAIN_FILE}
+      -DCMAKE_INSTALL_RPATH=${CMAKE_HOST_INSTALL_RPATH})
   else()
     set(CROSS_TOOLCHAIN_FLAGS_INIT
       -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
