--- bzip2-1.0.6/Makefile-libbz2_so~	2010-09-10 23:07:52.000000000 +0000
+++ bzip2-1.0.6/Makefile-libbz2_so	2017-01-08 16:08:44.140322653 +0000
@@ -35,7 +35,7 @@
       bzlib.o
 
 all: $(OBJS)
-	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 $(OBJS)
+	$(CC) $(CFLAGS) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 $(OBJS)
 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.6
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.6 libbz2.so.1.0
--- bzip2-1.0.6/Makefile-libbz2_so~	2017-01-08 16:08:44.000000000 +0000
+++ bzip2-1.0.6/Makefile-libbz2_so	2017-01-08 16:14:50.650677489 +0000
@@ -24,7 +24,7 @@
 SHELL=/bin/sh
 CC=gcc
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-fpic -fPIC -Wall -Winline -O2 -g $(BIGFILES)
+CFLAGS+=-fpic -fPIC -Wall -Winline -O2 -g $(BIGFILES)
 
 OBJS= blocksort.o  \
       huffman.o    \
