Index: make-4.1/tests/scripts/features/archives
===================================================================
--- make-4.1.orig/tests/scripts/features/archives	2014-10-05 17:24:51.000000000 +0100
+++ make-4.1/tests/scripts/features/archives	2015-03-12 13:43:18.515175881 +0000
@@ -8,6 +8,14 @@
 # If this instance of make doesn't support archives, skip it
 exists $FEATURES{archives} or return -1;
 
+#
+# We build binutils with --enable-deterministic-archives, which causes
+# these tests to fail.
+# https://bugzilla.redhat.com/show_bug.cgi?id=1124342
+# https://bugzilla.redhat.com/show_bug.cgi?id=1136481
+#
+return -1;
+
 # Create some .o files to work with
 utouch(-60, qw(a1.o a2.o a3.o));
 
