diff --git a/es-app/src/guis/GuiGamelistOptions.cpp b/es-app/src/guis/GuiGamelistOptions.cpp
index 1c8948c..cb8f45d 100644
--- a/es-app/src/guis/GuiGamelistOptions.cpp
+++ b/es-app/src/guis/GuiGamelistOptions.cpp
@@ -104,7 +104,7 @@ void GuiGamelistOptions::jumpToLetter()
 
 		if(checkLetter < letter)
 			min = mid + 1;
-		else if(checkLetter > letter)
+		else if(checkLetter > letter || (mid > 0 && (letter == toupper(files.at(mid - 1)->getName()[0]))))
 			max = mid - 1;
 		else
 			break; //exact match found
