diff --git a/src/crypto/x509/root_unix.go b/src/crypto/x509/root_unix.go
old mode 100644
new mode 100755
index f77d6c0..96eca2e
--- a/src/crypto/x509/root_unix.go
+++ b/src/crypto/x509/root_unix.go
@@ -23,6 +23,7 @@ var certFiles = []string{
 // reading at least one file from a directory.
 var certDirectories = []string{
 	"/system/etc/security/cacerts", // Android
+	"/etc/ssl",                     // LibreELEC
 
 }
 
