diff --git a/xorg-server.m4 b/xorg-server.m4
index 195bda5..0c8d845 100644
--- a/xorg-server.m4
+++ b/xorg-server.m4
@@ -31,7 +31,8 @@ dnl
 AC_DEFUN([XORG_DRIVER_CHECK_EXT],[
 	AC_REQUIRE([PKG_PROG_PKG_CONFIG])
 	SAVE_CFLAGS="$CFLAGS"
-	CFLAGS="$CFLAGS `$PKG_CONFIG --cflags xorg-server`"
+	m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+	CFLAGS="$CFLAGS -I$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --cflags xorg-server`"
 	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "xorg-server.h"
 #if !defined $1
