--- branches/gcc-5-branch/gcc/cp/Make-lang.in	2016/02/25 15:23:47	233719
+++ branches/gcc-5-branch/gcc/cp/Make-lang.in	2016/02/25 15:33:50	233720
@@ -111,7 +111,7 @@
 # deleting the $(srcdir)/cp/cfns.h file.
 $(srcdir)/cp/cfns.h:
 endif
-	gperf -o -C -E -k '1-6,$$' -j1 -D -N 'libc_name_p' -L ANSI-C \
+	gperf -o -C -E -k '1-6,$$' -j1 -D -N 'libc_name_p' -L C++ \
 		$(srcdir)/cp/cfns.gperf --output-file $(srcdir)/cp/cfns.h
 
 #
