diff -Naur gtk+-3.22.16-orig/Makefile.am gtk+-3.22.16/Makefile.am
--- gtk+-3.22.16-orig/Makefile.am	2017-06-21 20:44:05.287284334 +0300
+++ gtk+-3.22.16/Makefile.am	2017-06-21 20:44:26.409929474 +0300
@@ -1,8 +1,8 @@
 ## Makefile.am for GTK+
 include $(top_srcdir)/Makefile.decl
 
-SRC_SUBDIRS = gdk gtk libgail-util modules demos tests testsuite examples
-SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
+SRC_SUBDIRS = gdk gtk libgail-util modules demos tests
+SUBDIRS = po po-properties $(SRC_SUBDIRS) m4macros build
 
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
