--- a/src/xdr_sizeof.c	2017-07-17 18:42:58.707608652 +0000
+++ b/src/xdr_sizeof.c	2017-07-17 18:43:25.907227510 +0000
@@ -90,7 +90,7 @@
 	if (xdrs->x_op != XDR_ENCODE) {
 		return (NULL);
 	}
-	if (len < (uintptr_t)xdrs->x_base) {
+	if (len < (__intptr_t)xdrs->x_base) {
 		/* x_private was already allocated */
 		xdrs->x_handy += len;
 		return ((int32_t *) xdrs->x_private);
@@ -102,7 +102,7 @@
 			xdrs->x_base = 0;
 			return (NULL);
 		}
-		xdrs->x_base = (caddr_t)(uintptr_t)len;
+		xdrs->x_base = (caddr_t)(__intptr_t)len;
 		xdrs->x_handy += len;
 		return ((int32_t *) xdrs->x_private);
 	}
