diff -Naur docker-17.10.0.orig/components/engine/cmd/dockerd/daemon_unix.go docker-17.10.0/components/engine/cmd/dockerd/daemon_unix.go
--- docker-17.10.0.orig/components/engine/cmd/dockerd/daemon_unix.go	2017-11-04 12:56:10.459236350 +0100
+++ docker-17.10.0/components/engine/cmd/dockerd/daemon_unix.go	2017-11-04 12:56:43.774958858 +0100
@@ -57,7 +57,7 @@
 		opts = append(opts, libcontainerd.WithStartDaemon(true))
 	}
 	if daemon.UsingSystemd(cli.Config) {
-		args := []string{"--systemd-cgroup=true"}
+		args := []string{"--systemd-cgroup=false"}
 		opts = append(opts, libcontainerd.WithRuntimeArgs(args))
 	}
 	if cli.Config.LiveRestoreEnabled {
diff -Naur docker-17.10.0.orig/components/engine/daemon/start_unix.go docker-17.10.0/components/engine/daemon/start_unix.go
--- docker-17.10.0.orig/components/engine/daemon/start_unix.go	2017-10-17 20:51:33.000000000 +0200
+++ docker-17.10.0/components/engine/daemon/start_unix.go	2017-11-04 12:56:59.994823475 +0100
@@ -23,7 +23,7 @@
 		return nil, validationError{errors.Errorf("no such runtime '%s'", container.HostConfig.Runtime)}
 	}
 	if UsingSystemd(daemon.configStore) {
-		rt.Args = append(rt.Args, "--systemd-cgroup=true")
+		rt.Args = append(rt.Args, "--systemd-cgroup=false")
 	}
 	createOptions = append(createOptions, libcontainerd.WithRuntime(rt.Path, rt.Args))
 
