--- chromium-64.0.3282.71/chrome/browser/ui/libgtkui/native_theme_gtk2.cc.gtk2	2018-01-07 23:08:05.778818732 +0100
+++ chromium-64.0.3282.71/chrome/browser/ui/libgtkui/native_theme_gtk2.cc	2018-01-07 23:08:28.008440326 +0100
@@ -214,6 +214,13 @@
       return SK_ColorTRANSPARENT;
     case kColorId_ProminentButtonColor:
       return GetSystemColor(kColorId_LinkEnabled);
+    // TabbedPane
+    case kColorId_TabTitleColorActive:
+      return GetTextColor(GetEntry(), NORMAL);
+    case kColorId_TabTitleColorInactive:
+      return GetTextColor(GetLabel(), INSENSITIVE);
+    case kColorId_TabBottomBorder:
+      return GetTextColor(GetMenuItem(), INSENSITIVE);
     case kColorId_TextOnProminentButtonColor:
       return GetTextColor(GetLabel(), SELECTED);
     case kColorId_ButtonPressedShade:
