From: Santiago Vila <sanvila@debian.org>
Subject: Manual is wrong. Option -dj does not take a parameter
Bug-Debian: http://bugs.debian.org/536527

--- a/doc/indent.texinfo
+++ b/doc/indent.texinfo
@@ -1731,7 +1731,7 @@
 Break the line after the last argument in a declaration.@*
 @xref{Declarations}.
 
-@item -dj@var{n}
+@item -dj
 @itemx --left-justify-declarations
 If -cd 0 is used then comments after declarations are left justified
 behind the declaration.@*
