diff -Naur pango-1.40.14-orig/Makefile.am pango-1.40.14/Makefile.am
--- pango-1.40.14-orig/Makefile.am	2018-01-01 09:40:47.156372043 +0300
+++ pango-1.40.14/Makefile.am	2018-01-01 09:41:10.832242039 +0300
@@ -1,6 +1,6 @@
 ## Process this file with automake to create Makefile.in.
 
-SUBDIRS= pango pango-view examples tools tests win32
+SUBDIRS= pango win32
 
 meson_dist_files = \
 	pango/pango-features.h.meson \
diff -Naur pango-1.40.14-orig/pango/Makefile.am pango-1.40.14/pango/Makefile.am
--- pango-1.40.14-orig/pango/Makefile.am	2018-01-01 09:40:47.149705376 +0300
+++ pango-1.40.14/pango/Makefile.am	2018-01-01 09:42:15.248984296 +0300
@@ -33,7 +33,7 @@
 	$(X_CFLAGS)					\
 	$(PANGO_HIDDEN_VISIBILITY_CFLAGS)
 
-BUILT_SOURCES = pango-enum-types.h pango-enum-types.c pango-features.h
+BUILT_SOURCES = pango-features.h
 
 pangoincludedir=$(includedir)/pango-1.0/pango
 
@@ -134,22 +134,6 @@
 	pango-features.h   \
 	pango-enum-types.h
 
-pango-enum-types.h: s-enum-types-h
-s-enum-types-h: $(pango_headers) Makefile.am pango-enum-types.h.template
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) \
-			 --template pango-enum-types.h.template \
-			$(pango_headers) ) > pango-enum-types.h.tmp \
-			&& mv pango-enum-types.h.tmp pango-enum-types.h \
-			&& echo timestamp > $@
-
-pango-enum-types.c: s-enum-types-c
-s-enum-types-c: $(pango_headers) Makefile.am pango-enum-types.c.template
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) \
-			 --template pango-enum-types.c.template \
-			$(pango_headers) ) > pango-enum-types.c.tmp \
-			&& mv pango-enum-types.c.tmp pango-enum-types.c \
-			&& echo timestamp > $@
-
 pango-features.h:
 	$(AM_V_GEN) cd $(top_builddir) && ./config.status pango/pango-features.h
 
diff -Naur pango-1.40.14-orig/pango/pango-context.c pango-1.40.14/pango/pango-context.c
--- pango-1.40.14-orig/pango/pango-context.c	2018-01-01 09:40:47.153038709 +0300
+++ pango-1.40.14/pango/pango-context.c	2018-01-01 09:42:46.833658796 +0300
@@ -39,6 +39,7 @@
 #include "pango-engine-private.h"
 #include "pango-script-private.h"
 #include "pango-emoji-private.h"
+#include "pango-enum-types.h"
 
 /**
  * PangoContext:
diff -Naur pango-1.40.14-orig/pango/pangofc-font.c pango-1.40.14/pango/pangofc-font.c
--- pango-1.40.14-orig/pango/pangofc-font.c	2018-01-01 09:40:47.153038709 +0300
+++ pango-1.40.14/pango/pangofc-font.c	2018-01-01 09:43:36.287884394 +0300
@@ -45,6 +45,7 @@
 #include "pango-engine.h"
 #include "pango-layout.h"
 #include "pango-impl-utils.h"
+#include "pango-enum-types.h"
 
 #include <fontconfig/fcfreetype.h>
 
diff -Naur pango-1.40.14-orig/pango/pango-layout.c pango-1.40.14/pango/pango-layout.c
--- pango-1.40.14-orig/pango/pango-layout.c	2018-01-01 09:40:47.149705376 +0300
+++ pango-1.40.14/pango/pango-layout.c	2018-01-01 09:43:57.841171728 +0300
@@ -77,6 +77,7 @@
 #include <string.h>
 
 #include "pango-layout-private.h"
+#include "pango-enum-types.h"
 
 
 typedef struct _ItemProperties ItemProperties;
