diff --git a/Makefile b/Makefile
old mode 100644
new mode 100755
index bf67bc6..8742798
--- a/Makefile
+++ b/Makefile
@@ -26,8 +26,8 @@
 # set the base of the Installation
 # BASEDIR = /mnt
 BASEDIR ?= $(DESTDIR)
-BINDIR ?= /bin
-SBINDIR ?= /sbin
+BINDIR ?= /usr/bin
+SBINDIR ?= /usr/bin
 
 # path to the net-lib support library. Default: lib
 NET_LIB_PATH = lib
diff --git a/config.in b/config.in
old mode 100644
new mode 100755
index b942b25..6344fe3
--- a/config.in
+++ b/config.in
@@ -42,14 +42,14 @@
 * course, welcome.  Answer `n' here if you have no support for
 * internationalization on your system.
 *
-bool 'Does your system support GNU gettext?' I18N n
+bool 'Does your system support GNU gettext?' I18N y
 *
 *
 *             Protocol Families.
 *
 bool 'UNIX protocol family' HAVE_AFUNIX y
 bool 'INET (TCP/IP) protocol family' HAVE_AFINET y
-bool 'INET6 (IPv6) protocol family' HAVE_AFINET6 y
+bool 'INET6 (IPv6) protocol family' HAVE_AFINET6 n
 bool 'Novell IPX/SPX protocol family' HAVE_AFIPX y
 bool 'Appletalk DDP protocol family' HAVE_AFATALK y
 bool 'AX25 (packet radio) protocol family' HAVE_AFAX25 y
@@ -76,7 +76,7 @@ bool 'Rose (packet radio) support' HAVE_HWROSE y
 bool 'NET/ROM (packet radio) support' HAVE_HWNETROM y
 bool 'X.25 (generic) support' HAVE_HWX25 y
 bool 'DLCI/FRAD (frame relay) support' HAVE_HWFR y
-bool 'SIT (IPv6-in-IPv4) support' HAVE_HWSIT y
+bool 'SIT (IPv6-in-IPv4) support' HAVE_HWSIT n
 bool 'FDDI (generic) support' HAVE_HWFDDI y
 bool 'HIPPI (generic) support' HAVE_HWHIPPI y
 bool 'Ash hardware support' HAVE_HWASH y
