diff --git a/zlib.h b/zlib.h
old mode 100644
new mode 100755
index 3e49ea3..92e2f35
--- a/zlib.h
+++ b/zlib.h
@@ -37,15 +37,8 @@
 extern "C" {
 #endif
 
-#define ZLIBNG_VERSION "1.9.9"
-#define ZLIBNG_VERNUM 0x1990
-#define ZLIBNG_VER_MAJOR 1
-#define ZLIBNG_VER_MINOR 9
-#define ZLIBNG_VER_REVISION 9
-#define ZLIBNG_VER_SUBREVISION 0
-
-#define ZLIB_VERSION "1.2.11.zlib-ng"
-#define ZLIB_VERNUM 0x12bf
+#define ZLIB_VERSION "1.2.11"
+#define ZLIB_VERNUM 0x12b0
 #define ZLIB_VER_MAJOR 1
 #define ZLIB_VER_MINOR 2
 #define ZLIB_VER_REVISION 11
