diff -Naurp kodi-7e52c1d/addons/resource.language.en_gb/resources/strings.po kodi-7e52c1d-patch/addons/resource.language.en_gb/resources/strings.po
--- kodi-7e52c1d/addons/resource.language.en_gb/resources/strings.po	2018-04-15 17:13:09.000000000 +0300
+++ kodi-7e52c1d-patch/addons/resource.language.en_gb/resources/strings.po	2018-04-15 17:11:50.000000000 +0300
@@ -5596,7 +5596,7 @@ msgstr ""
 #: system/settings/settings.xml
 #: addons/skin.estuary/1080i/DialogButtonMenu.xml
 msgctxt "#13010"
-msgid "Hibernate"
+msgid "Games"
 msgstr ""
 
 #: system/settings/settings.xml
@@ -5609,7 +5609,7 @@ msgstr ""
 #: addons/skin.estuary/1080i/DialogButtonMenu.xml
 #: addons/skin.estouchy/xml/Home.xml
 msgctxt "#13012"
-msgid "Exit"
+msgid "TV"
 msgstr ""
 
 #: addons/skin.estuary/1080i/DialogButtonMenu.xml
diff -Naurp kodi-7e52c1d/xbmc/Application.cpp kodi-7e52c1d-patch/xbmc/Application.cpp
--- kodi-7e52c1d/xbmc/Application.cpp	2018-04-15 17:13:09.000000000 +0300
+++ kodi-7e52c1d-patch/xbmc/Application.cpp	2018-04-15 17:15:02.000000000 +0300
@@ -223,6 +223,9 @@
 #include "pictures/GUIWindowSlideShow.h"
 #include "windows/GUIWindowLoginScreen.h"
 
+#include <fstream>
+#include <fcntl.h>
+
 using namespace ADDON;
 using namespace XFILE;
 #ifdef HAS_DVD_DRIVE
@@ -2428,7 +2431,8 @@ void CApplication::OnApplicationMessage(
     break;
 
   case TMSG_QUIT:
-    SetExitCode(EXITCODE_QUIT);
+    SetExitCode(EXITCODE_TV);
+    Stop(EXITCODE_TV);
     break;
   
   case TMSG_SHUTDOWN:
@@ -2440,7 +2444,8 @@ void CApplication::OnApplicationMessage(
     break;
 
   case TMSG_HIBERNATE:
-    g_powerManager.Hibernate();
+    SetExitCode(EXITCODE_GAMES);
+    Stop(EXITCODE_GAMES);
     break;
 
   case TMSG_SUSPEND:
@@ -3009,6 +3014,22 @@ void CApplication::Stop(int exitCode)
 
   cleanup_emu_environ();
 
+  // start Games and VDR
+  switch (m_ExitCode)
+  {
+  case EXITCODE_GAMES:
+    remove("/var/lock/start.kodi");
+    creat("/var/lock/start.retro", S_IRUSR|S_IROTH);
+    system("systemctl start retroarch.service");
+    break;
+  
+  case EXITCODE_TV:
+    remove("/var/lock/start.kodi");
+    creat("/var/lock/start.vdr", S_IRUSR|S_IROTH);
+    system("systemctl start vdr.service");
+    break;
+  }
+
   Sleep(200);
 }
 
diff -Naurp kodi-7e52c1d/xbmc/powermanagement/linux/LogindUPowerSyscall.cpp kodi-7e52c1d-patch/xbmc/powermanagement/linux/LogindUPowerSyscall.cpp
--- kodi-7e52c1d/xbmc/powermanagement/linux/LogindUPowerSyscall.cpp	2018-02-12 20:41:29.000000000 +0200
+++ kodi-7e52c1d-patch/xbmc/powermanagement/linux/LogindUPowerSyscall.cpp	2018-04-15 17:11:50.000000000 +0300
@@ -50,10 +50,10 @@ CLogindUPowerSyscall::CLogindUPowerSysca
   if (!m_hasUPower)
     CLog::Log(LOGINFO, "LogindUPowerSyscall - UPower not found, battery information will not be available");
 
-  m_canPowerdown = LogindCheckCapability("CanPowerOff");
-  m_canReboot    = LogindCheckCapability("CanReboot");
-  m_canHibernate = LogindCheckCapability("CanHibernate");
-  m_canSuspend   = LogindCheckCapability("CanSuspend");
+  m_canPowerdown = true;
+  m_canReboot    = true;
+  m_canHibernate = true;
+  m_canSuspend   = true;
 
   InhibitDelayLock();
 
diff -Naurp kodi-7e52c1d/xbmc/powermanagement/PowerManager.cpp kodi-7e52c1d-patch/xbmc/powermanagement/PowerManager.cpp
--- kodi-7e52c1d/xbmc/powermanagement/PowerManager.cpp	2018-02-12 20:41:29.000000000 +0200
+++ kodi-7e52c1d-patch/xbmc/powermanagement/PowerManager.cpp	2018-04-15 17:11:50.000000000 +0300
@@ -312,17 +312,6 @@ void CPowerManager::OnLowBattery()
 
 void CPowerManager::SettingOptionsShutdownStatesFiller(const CSetting *setting, std::vector< std::pair<std::string, int> > &list, int &current, void *data)
 {
-  if (g_powerManager.CanPowerdown())
-    list.push_back(make_pair(g_localizeStrings.Get(13005), POWERSTATE_SHUTDOWN));
-  if (g_powerManager.CanHibernate())
-    list.push_back(make_pair(g_localizeStrings.Get(13010), POWERSTATE_HIBERNATE));
-  if (g_powerManager.CanSuspend())
-    list.push_back(make_pair(g_localizeStrings.Get(13011), POWERSTATE_SUSPEND));
-  if (!g_application.IsStandAlone())
-  {
-    list.push_back(make_pair(g_localizeStrings.Get(13009), POWERSTATE_QUIT));
-#if !defined(TARGET_DARWIN_IOS)
-    list.push_back(make_pair(g_localizeStrings.Get(13014), POWERSTATE_MINIMIZE));
-#endif
-  }
+  list.push_back(make_pair(g_localizeStrings.Get(13005), POWERSTATE_SHUTDOWN));
+  list.push_back(make_pair(g_localizeStrings.Get(13011), POWERSTATE_SUSPEND));
 }
diff -Naurp kodi-7e52c1d/xbmc/XBApplicationEx.h kodi-7e52c1d-patch/xbmc/XBApplicationEx.h
--- kodi-7e52c1d/xbmc/XBApplicationEx.h	2018-04-15 17:13:09.000000000 +0300
+++ kodi-7e52c1d-patch/xbmc/XBApplicationEx.h	2018-04-15 17:11:50.000000000 +0300
@@ -31,6 +31,8 @@ enum {
   EXITCODE_POWERDOWN = 64,
   EXITCODE_RESTARTAPP= 65,
   EXITCODE_REBOOT    = 66,
+  EXITCODE_TV        = 67,
+  EXITCODE_GAMES     = 68,
 };
 
 class CXBApplicationEx : public IWindowManagerCallback
