diff --git a/bfd/configure b/bfd/configure
index d6d4e295..981bb55e 100755
--- a/bfd/configure
+++ b/bfd/configure
@@ -3330,7 +3330,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
@@ -4815,7 +4815,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
diff --git a/binutils/configure b/binutils/configure
index 7854d5c3..ff286f82 100755
--- a/binutils/configure
+++ b/binutils/configure
@@ -3314,7 +3314,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
@@ -4582,7 +4582,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
diff --git a/config/acinclude.m4 b/config/acinclude.m4
index 8242b2c7..6ba1c9bd 100644
--- a/config/acinclude.m4
+++ b/config/acinclude.m4
@@ -24,7 +24,7 @@ dnl normal versions of a library), tasteless as that idea is.
   if test "$ac_test_CXXFLAGS" = set; then
     CXXFLAGS="$ac_save_CXXFLAGS"
   elif test $ac_cv_prog_cxx_g = yes; then
-    CXXFLAGS="-g -O2"
+    CXXFLAGS="-g1 -O2"
   else
     CXXFLAGS="-g1 -O2"
   fi
@@ -121,7 +121,7 @@ dnl normal versions of a library), tasteless as that idea is.
   if test "$ac_test_CFLAGS" = set; then
     CFLAGS="$ac_save_CFLAGS"
   elif test $ac_cv_prog_cc_g = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g1 -O2"
   fi
diff --git a/configure b/configure
index be9dd89d..a8b3a11b 100755
--- a/configure
+++ b/configure
@@ -4620,7 +4620,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
@@ -4968,7 +4968,7 @@ if test "$ac_test_CXXFLAGS" = set; then
   CXXFLAGS=$ac_save_CXXFLAGS
 elif test $ac_cv_prog_cxx_g = yes; then
   if test "$GXX" = yes; then
-    CXXFLAGS="-g -O2"
+    CXXFLAGS="-g1 -O2"
   else
     CXXFLAGS="-g"
   fi
@@ -6688,7 +6688,7 @@ fi
 # CFLAGS_FOR_TARGET and CXXFLAGS_FOR_TARGET.
 if test "x$CFLAGS_FOR_TARGET" = x; then
   if test "x${is_cross_compiler}" = xyes; then
-    CFLAGS_FOR_TARGET="-g -O2"
+    CFLAGS_FOR_TARGET="-g1 -O2"
   else
     CFLAGS_FOR_TARGET=$CFLAGS
     case " $CFLAGS " in
@@ -6705,7 +6705,7 @@ fi
 
 if test "x$CXXFLAGS_FOR_TARGET" = x; then
   if test "x${is_cross_compiler}" = xyes; then
-    CXXFLAGS_FOR_TARGET="-g -O2"
+    CXXFLAGS_FOR_TARGET="-g1 -O2"
   else
     CXXFLAGS_FOR_TARGET=$CXXFLAGS
     case " $CXXFLAGS " in
diff --git a/configure.ac b/configure.ac
index 532c5c27..3acf65ba 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2282,7 +2282,7 @@ AC_SUBST(DEBUG_PREFIX_CFLAGS_FOR_TARGET)
 # CFLAGS_FOR_TARGET and CXXFLAGS_FOR_TARGET.
 if test "x$CFLAGS_FOR_TARGET" = x; then
   if test "x${is_cross_compiler}" = xyes; then
-    CFLAGS_FOR_TARGET="-g -O2"
+    CFLAGS_FOR_TARGET="-g1 -O2"
   else
     CFLAGS_FOR_TARGET=$CFLAGS
     case " $CFLAGS " in
@@ -2299,7 +2299,7 @@ AC_SUBST(CFLAGS_FOR_TARGET)
 
 if test "x$CXXFLAGS_FOR_TARGET" = x; then
   if test "x${is_cross_compiler}" = xyes; then
-    CXXFLAGS_FOR_TARGET="-g -O2"
+    CXXFLAGS_FOR_TARGET="-g1 -O2"
   else
     CXXFLAGS_FOR_TARGET=$CXXFLAGS
     case " $CXXFLAGS " in
diff --git a/gas/configure b/gas/configure
index e6e985e7..640e4c5d 100755
--- a/gas/configure
+++ b/gas/configure
@@ -3075,7 +3075,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
@@ -4343,7 +4343,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
diff --git a/gold/configure b/gold/configure
index cb020be5..b0eb28eb 100755
--- a/gold/configure
+++ b/gold/configure
@@ -4405,7 +4405,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
@@ -4944,7 +4944,7 @@ if test "$ac_test_CXXFLAGS" = set; then
   CXXFLAGS=$ac_save_CXXFLAGS
 elif test $ac_cv_prog_cxx_g = yes; then
   if test "$GXX" = yes; then
-    CXXFLAGS="-g -O2"
+    CXXFLAGS="-g1 -O2"
   else
     CXXFLAGS="-g"
   fi
diff --git a/gprof/configure b/gprof/configure
index 6abea208..c14ce5cf 100755
--- a/gprof/configure
+++ b/gprof/configure
@@ -2986,7 +2986,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
@@ -4257,7 +4257,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
diff --git a/intl/configure b/intl/configure
index d7af03fc..705b7e05 100755
--- a/intl/configure
+++ b/intl/configure
@@ -3288,7 +3288,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
diff --git a/ld/configure b/ld/configure
index 5087f017..3278d18e 100755
--- a/ld/configure
+++ b/ld/configure
@@ -3400,7 +3400,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
@@ -4691,7 +4691,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
@@ -5039,7 +5039,7 @@ if test "$ac_test_CXXFLAGS" = set; then
   CXXFLAGS=$ac_save_CXXFLAGS
 elif test $ac_cv_prog_cxx_g = yes; then
   if test "$GXX" = yes; then
-    CXXFLAGS="-g -O2"
+    CXXFLAGS="-g1 -O2"
   else
     CXXFLAGS="-g"
   fi
diff --git a/ld/testsuite/config/default.exp b/ld/testsuite/config/default.exp
index f5cb1884..f6c543c2 100644
--- a/ld/testsuite/config/default.exp
+++ b/ld/testsuite/config/default.exp
@@ -86,7 +86,7 @@ if {![info exists CC]} {
     set CC [find_gcc]
 }
 if {![info exists CFLAGS]} {
-    set CFLAGS "-g -O2"
+    set CFLAGS "-g1 -O2"
 }
 if {![info exists CXX]} {
     set CXX [find_g++]
diff --git a/libiberty/configure b/libiberty/configure
index ccadfaa8..521ac8ce 100755
--- a/libiberty/configure
+++ b/libiberty/configure
@@ -3619,7 +3619,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
diff --git a/libtool.m4 b/libtool.m4
index 24d13f34..ebae02ce 100644
--- a/libtool.m4
+++ b/libtool.m4
@@ -7052,7 +7052,7 @@ AC_DEFUN([LT_PROG_GCJ],
 [m4_ifdef([AC_PROG_GCJ], [AC_PROG_GCJ],
   [m4_ifdef([A][M_PROG_GCJ], [A][M_PROG_GCJ],
     [AC_CHECK_TOOL(GCJ, gcj,)
-      test "x${GCJFLAGS+set}" = xset || GCJFLAGS="-g -O2"
+      test "x${GCJFLAGS+set}" = xset || GCJFLAGS="-g1 -O2"
       AC_SUBST(GCJFLAGS)])])[]dnl
 ])
 
diff --git a/opcodes/configure b/opcodes/configure
index 7e5e2538..9dfa0318 100755
--- a/opcodes/configure
+++ b/opcodes/configure
@@ -3052,7 +3052,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
@@ -4320,7 +4320,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
diff --git a/zlib/configure b/zlib/configure
index bed9e3ea..e5eba2e1 100755
--- a/zlib/configure
+++ b/zlib/configure
@@ -3599,7 +3599,7 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g1 -O2"
   else
     CFLAGS="-g"
   fi
