--- binutils.orig/bfd/elf64-ppc.c	2018-01-25 13:23:13.575860963 +0000
+++ binutils-2.29.1/bfd/elf64-ppc.c	2018-01-25 13:24:36.564874275 +0000
@@ -12805,6 +12805,9 @@ ppc64_elf_size_stubs (struct bfd_link_in
 	    stub_sec->flags &= ~SEC_RELOC;
 	  }
 
+      if (htab->stub_iteration <= STUB_SHRINK_ITER
+	  || htab->brlt->rawsize < htab->brlt->size)
+	htab->brlt->rawsize = htab->brlt->size;
       htab->brlt->size = 0;
       htab->brlt->reloc_count = 0;
       htab->brlt->flags &= ~SEC_RELOC;
@@ -12858,6 +12861,9 @@ ppc64_elf_size_stubs (struct bfd_link_in
 	  break;
 
       if (group == NULL
+	  && (htab->brlt->rawsize == htab->brlt->size
+	      || (htab->stub_iteration > STUB_SHRINK_ITER
+		  && htab->brlt->rawsize > htab->brlt->size))
 	  && (htab->glink_eh_frame == NULL
 	      || htab->glink_eh_frame->rawsize == htab->glink_eh_frame->size))
 	break;
