Description: turn off POSIX_IDENTIFIERS option in completion
Origin: 2e76fb80b4c228944b57b8f669e75456a23f147d
Author: Peter Stephenson <pws@zsh.org>
Reviewed-By: Axel Beckert <abe@debian.org>

--- a/Completion/compinit
+++ b/Completion/compinit
@@ -153,6 +153,7 @@
     NO_markdirs
     NO_octalzeroes
     NO_posixbuiltins
+    NO_posixidentifiers
     NO_shwordsplit
     NO_shglob
     NO_warncreateglobal
