--- mozilla-release.orig/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.cc	2013-09-11 01:15:25.000000000 +0200
+++ mozilla-release/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.cc	2013-10-24 12:08:16.000000000 +0200
@@ -34,13 +34,31 @@
 #include "common/stabs_reader.h"
 
 #include <assert.h>
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
 #include <stab.h>
+#else
+#define __define_stab(NAME, CODE, STRING) NAME=CODE,
+enum __stab_debug_code
+{
+__define_stab (N_FUN, 0x24, "FUN")
+__define_stab (N_SLINE, 0x44, "SLINE")
+__define_stab (N_SOL, 0x84, "SOL")
+__define_stab (N_SO, 0x64, "SO")
+LAST_UNUSED_STAB_CODE
+};
+#undef __define_stab
+#endif
 #include <string.h>
 
 #include <string>
 
 #include "common/using_std_string.h"
 
+#ifndef N_UNDF
+#define N_UNDF 0
+#endif
+
+
 using std::vector;
 
 namespace google_breakpad {
