@@ -1,44 +0,0 @@
--- vdr-plugin-softhddevice.orig/video.c       2015-10-28 17:43:37.352439593 +0100
+++ vdr-plugin-softhddevice/video.c    2015-10-28 17:42:18.074438509 +0100
@@ -3493,6 +3493,9 @@
         vaUnmapBuffer(VaDisplay, *decoder->vpp_deinterlace_buf);
     }
 
+    if (!filter_count) {
+       return NULL; // no postprocessing if no filters applied
+    }
 
     va_status = VaapiPostprocessSurface(decoder->vpp_ctx, decoder->PlaybackSurface, *surface,
                   filters_to_run, filter_count,
