diff -Naurp vdr-plugin-softhddevice-4de20b9/softhddevice.cpp vdr-plugin-softhddevice-4de20b9-patch/softhddevice.cpp
--- vdr-plugin-softhddevice-4de20b9/softhddevice.cpp	2015-11-01 19:40:45.000000000 +0200
+++ vdr-plugin-softhddevice-4de20b9-patch/softhddevice.cpp	2016-10-01 19:05:32.000000000 +0300
@@ -3592,6 +3592,19 @@ bool cPluginSoftHdDevice::Service(const
 {
     //dsyslog("[softhddev]%s: id %s\n", __FUNCTION__, id);
 
+	if (strcmp(id, OSD1_3DMODE_SERVICE) == 0) {
+		SoftHDDevice_Osd3DModeService_v1_1_t *r;
+
+		r = (SoftHDDevice_Osd3DModeService_v1_1_t *) data;
+		if (r->GetMode) {
+			r->Mode = VideoGetOsd3DMode();
+		}
+		else {
+			VideoSetOsd3DMode(r->Mode);
+			return true;
+		}
+    }
+
     if (strcmp(id, OSD_3DMODE_SERVICE) == 0) {
 	SoftHDDevice_Osd3DModeService_v1_0_t *r;
 
diff -Naurp vdr-plugin-softhddevice-4de20b9/softhddevice_service.h vdr-plugin-softhddevice-4de20b9-patch/softhddevice_service.h
--- vdr-plugin-softhddevice-4de20b9/softhddevice_service.h	2015-11-01 19:40:45.000000000 +0200
+++ vdr-plugin-softhddevice-4de20b9-patch/softhddevice_service.h	2016-10-01 19:07:04.000000000 +0300
@@ -25,6 +25,7 @@
 #define ATMO_GRAB_SERVICE	"SoftHDDevice-AtmoGrabService-v1.0"
 #define ATMO1_GRAB_SERVICE	"SoftHDDevice-AtmoGrabService-v1.1"
 #define OSD_3DMODE_SERVICE	"SoftHDDevice-Osd3DModeService-v1.0"
+#define OSD1_3DMODE_SERVICE	"SoftHDDevice-Osd3DModeService-v1.1"
 
 enum
 { GRAB_IMG_RGBA_FORMAT_B8G8R8A8 };
@@ -52,6 +53,12 @@ typedef struct
 
 typedef struct
 {
+    int Mode;
+    bool GetMode;
+} SoftHDDevice_Osd3DModeService_v1_1_t;
+
+typedef struct
+{
     // request/reply data
 
     int width;
diff -Naurp vdr-plugin-softhddevice-4de20b9/video.c vdr-plugin-softhddevice-4de20b9-patch/video.c
--- vdr-plugin-softhddevice-4de20b9/video.c	2016-10-01 19:02:20.000000000 +0300
+++ vdr-plugin-softhddevice-4de20b9-patch/video.c	2016-10-01 19:08:25.000000000 +0300
@@ -11573,6 +11573,16 @@ void VideoSetOsdSize(int width, int heig
 }
 
 ///
+///	Set active 3d OSD mode.
+///
+///	@return mode	OSD mode (0=off, 1=SBS, 2=Top Bottom)
+///
+int VideoGetOsd3DMode(void)
+{
+    return Osd3DMode;
+}
+
+///
 ///	Set the 3d OSD mode.
 ///
 ///	@param mode	OSD mode (0=off, 1=SBS, 2=Top Bottom)
diff -Naurp vdr-plugin-softhddevice-4de20b9/video.h vdr-plugin-softhddevice-4de20b9-patch/video.h
--- vdr-plugin-softhddevice-4de20b9/video.h	2015-11-01 19:40:45.000000000 +0200
+++ vdr-plugin-softhddevice-4de20b9-patch/video.h	2016-10-01 19:09:30.000000000 +0300
@@ -223,6 +223,9 @@ extern void VideoSetOsdSize(int, int);
     /// Set Osd 3D Mode
 extern void VideoSetOsd3DMode(int);
 
+    /// Get Osd 3D Mode
+extern int VideoGetOsd3DMode(void);
+
     /// Set video clock.
 extern void VideoSetClock(VideoHwDecoder *, int64_t);
 
