From 0592680436af2de8c553a8cca2fcf0293e60be32 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Fri, 21 Oct 2016 21:54:41 +0100
Subject: [PATCH 2/3] Use sane defaults throughout the system

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
---
 conf.d/60-latin.conf | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/conf.d/60-latin.conf b/conf.d/60-latin.conf
index 35600ea..d5d7361 100644
--- a/conf.d/60-latin.conf
+++ b/conf.d/60-latin.conf
@@ -4,6 +4,7 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
+			<family>Clear Sans</family>
 			<family>Bitstream Vera Serif</family>
 			<family>DejaVu Serif</family>
 			<family>Times New Roman</family>
@@ -17,6 +18,7 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
+			<family>Clear Sans</family>
 			<family>Bitstream Vera Sans</family>
 			<family>DejaVu Sans</family>
 			<family>Verdana</family>
@@ -34,6 +36,7 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
+			<family>Droid Sans Mono for Powerline</family>
 			<family>Bitstream Vera Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
 			<family>Inconsolata</family>
-- 
2.10.1

