--- iproute2-4.9.0.orig/ip/iplink_can.c	2016-12-13 00:07:42.000000000 +0100
+++ iproute2-4.9.0/ip/iplink_can.c	2016-12-26 01:27:39.734025890 +0100
@@ -119,7 +119,7 @@ static int can_parse_opt(struct link_uti
 			if (get_u32(&bt.bitrate, *argv, 0))
 				invarg("invalid \"bitrate\" value\n", *argv);
 		} else if (matches(*argv, "sample-point") == 0) {
-			float sp;
+			float sp = 0;
 
 			NEXT_ARG();
 			if (get_float(&sp, *argv))
