--- chromium-66.0.3359.66/build/linux/unbundle/ffmpeg.gn.buildflag_header	2018-03-29 22:00:27.309041504 +0200
+++ chromium-66.0.3359.66/build/linux/unbundle/ffmpeg.gn	2018-03-30 12:30:19.959072993 +0200
@@ -14,8 +14,8 @@
   ]
 }
 
-buildflag_header("ffmpeg_buildflags") {
-  header = "ffmpeg_buildflags.h"
+buildflag_header("ffmpeg_features") {
+  header = "ffmpeg_features.h"
   flags = [ "USE_SYSTEM_FFMPEG=true" ]
 }
 
@@ -36,7 +36,7 @@
 
 source_set("ffmpeg") {
   deps = [
-    ":ffmpeg_buildflags",
+    ":ffmpeg_features",
     ":ffmpeg_shim",
   ]
   public_configs = [ ":system_ffmpeg", ":ffmpeg_deprecations" ]
--- chromium-66.0.3359.66/build/linux/unbundle/jsoncpp.gn.overrides	2018-03-29 22:00:27.291041781 +0200
+++ chromium-66.0.3359.66/build/linux/unbundle/jsoncpp.gn	2018-03-30 12:35:59.317987225 +0200
@@ -20,9 +20,13 @@
   ]
 }
 
+config("jsoncpp_config") {
+}
+
 source_set("jsoncpp") {
   deps = [
     ":jsoncpp_shim",
   ]
+  public_configs = [ ":jsoncpp_config" ]
   libs = [ "jsoncpp" ]
 }
