diff -ru glew-2.0.0_backup/Makefile glew-2.0.0/Makefile
--- glew-2.0.0_backup/Makefile	2017-11-15 04:36:46.781150418 +0000
+++ glew-2.0.0/Makefile	2017-11-15 05:13:05.527365350 +0000
@@ -81,7 +81,7 @@
 OPT = $(POPT)
 endif
 INCLUDE = -Iinclude
-CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
+CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA) $(DEFAULTFLAGS)
 
 all debug: glew.lib glew.bin
 
@@ -114,7 +114,7 @@
 endif
 
 lib/$(LIB.SHARED): $(LIB.SOBJS)
-	$(LD) $(LDFLAGS.SO) -o $@ $^ $(LIB.LDFLAGS) $(LIB.LIBS)
+	$(LD) $(CFLAGS) $(LDFLAGS.SO) -o $@ $^ $(LIB.LDFLAGS) $(LIB.LIBS)
 ifneq ($(LN),)
 	$(LN) $(LIB.SHARED) lib/$(LIB.SONAME)
 	$(LN) $(LIB.SHARED) lib/$(LIB.DEVLNK)
