diff -Naur gcc-9-20190120-orig/gcc/opts.c gcc-9-20190120/gcc/opts.c
--- gcc-9-20190120-orig/gcc/opts.c	2019-01-23 23:44:59.183029675 +0300
+++ gcc-9-20190120/gcc/opts.c	2019-01-06 19:44:51.000000000 +0300
@@ -2315,6 +2315,9 @@
 			       opts, opts_set, loc, dc);
       break;
 
+    case OPT_fhonour_copts:
+      break;
+
     case OPT_Wfatal_errors:
       dc->fatal_errors = value;
       break;
