diff --git a/p11-kit/Makefile.am b/p11-kit/Makefile.am
old mode 100644
new mode 100755
index 6496e8d..c5e26c5
--- a/p11-kit/Makefile.am
+++ b/p11-kit/Makefile.am
@@ -197,17 +197,6 @@ p11_kit_LIBS = \
 	$(LTLIBINTL)
 
 CHECK_PROGS += \
-	test-progname \
-	test-util \
-	test-conf \
-	test-uri \
-	test-pin \
-	test-init \
-	test-modules \
-	test-deprecated \
-	test-proxy \
-	test-iter \
-	test-rpc \
 	$(NULL)
 
 test_conf_SOURCES = p11-kit/test-conf.c
@@ -244,8 +233,7 @@ test_util_SOURCES = p11-kit/test-util.c
 test_util_LDADD = $(p11_kit_LIBS)
 
 noinst_PROGRAMS += \
-	print-messages \
-	frob-setuid
+	$(NULL)
 
 print_messages_SOURCES = p11-kit/print-messages.c
 print_messages_LDADD = $(p11_kit_LIBS)
@@ -254,11 +242,6 @@ frob_setuid_SOURCES = p11-kit/frob-setuid.c
 frob_setuid_LDADD = $(p11_kit_LIBS)
 
 CHECK_PROGS += \
-	test-virtual \
-	test-managed \
-	test-log \
-	test-filter \
-	test-transport \
 	$(NULL)
 
 test_log_SOURCES = p11-kit/test-log.c
