--- gtk+-3.22.0/gtk/gtkstyleproviderprivate.c~	2016-03-26 21:32:35.000000000 +0000
+++ gtk+-3.22.0/gtk/gtkstyleproviderprivate.c	2017-04-08 18:03:12.258391211 +0000
@@ -120,6 +120,9 @@
 
   gtk_internal_return_val_if_fail (GTK_IS_STYLE_PROVIDER_PRIVATE (provider), NULL);
 
+  if (!provider)
+	return NULL;
+
   iface = GTK_STYLE_PROVIDER_PRIVATE_GET_INTERFACE (provider);
 
   if (!iface->get_settings)
--- gtk+-3.22.15/gtk/gtkcssvalue.c~	2016-10-22 04:17:52.000000000 +0000
+++ gtk+-3.22.15/gtk/gtkcssvalue.c	2017-06-04 12:53:04.513036527 +0000
@@ -90,6 +90,7 @@
 {
 
   gtk_internal_return_val_if_fail (value != NULL, NULL);
+  gtk_internal_return_val_if_fail (value->class != NULL, NULL);
   gtk_internal_return_val_if_fail (GTK_IS_STYLE_PROVIDER_PRIVATE (provider), NULL);
   gtk_internal_return_val_if_fail (GTK_IS_CSS_STYLE (style), NULL);
   gtk_internal_return_val_if_fail (parent_style == NULL || GTK_IS_CSS_STYLE (parent_style), NULL);
--- gtk+-3.22.15/gtk/gtksettings.c~	2017-04-21 22:36:22.000000000 +0000
+++ gtk+-3.22.15/gtk/gtksettings.c	2017-06-09 22:56:24.451954721 +0000
@@ -3264,6 +3264,8 @@
 GdkScreen *
 _gtk_settings_get_screen (GtkSettings *settings)
 {
+  if (!settings || !settings->priv)
+	return NULL;
   return settings->priv->screen;
 }
 
