--- a/CMakeLists.txt	2017-05-08 15:14:49.636557670 +0200
+++ b/CMakeLists.txt	2017-05-08 15:15:00.959993592 +0200
@@ -7,7 +7,7 @@
 
   # Rely on llvm-config.
   set(CONFIG_OUTPUT)
-  find_program(LLVM_CONFIG "llvm-config")
+  set(LLVM_CONFIG "${CMAKE_FIND_ROOT_PATH}/usr/bin/llvm-config-host")
   if(LLVM_CONFIG)
     message(STATUS "Found LLVM_CONFIG as ${LLVM_CONFIG}")
     set(CONFIG_COMMAND ${LLVM_CONFIG}
