diff -ruN vdr-plugin-xvdr-xvdr-0.9.9/src/xvdr/xvdrclient.c vdr-plugin-xvdr-xvdr-0.9.9_new/src/xvdr/xvdrclient.c
--- vdr-plugin-xvdr-xvdr-0.9.9/src/xvdr/xvdrclient.c	2013-02-21 20:13:55.000000000 +0100
+++ vdr-plugin-xvdr-xvdr-0.9.9_new/src/xvdr/xvdrclient.c	2013-12-15 20:11:10.348273061 +0100
@@ -1683,7 +1683,7 @@
       sep++;
       *sep = 0;
     }
-    strncat(filename_new, newtitle, 512);
+    strncat(filename_new, newtitle, sizeof(filename_new)-strlen(filename_new)-1);
 
     INFOLOG("renaming recording '%s' to '%s'", filename_old, filename_new);
     r = rename(filename_old, filename_new);
