From f1978e83082f1a33874465db951563eee13ceac9 Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Mon, 20 Feb 2017 13:38:23 +1100
Subject: [PATCH 14/24] Make writeback throttling default enabled.

---
 block/Kconfig |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-4.11-ck2/block/Kconfig
===================================================================
--- linux-4.11-ck2.orig/block/Kconfig	2017-05-26 19:04:59.014232702 +1000
+++ linux-4.11-ck2/block/Kconfig	2017-05-26 19:04:59.004232775 +1000
@@ -128,7 +128,7 @@ config BLK_CMDLINE_PARSER
 
 config BLK_WBT
 	bool "Enable support for block device writeback throttling"
-	default n
+	default y
 	---help---
 	Enabling this option enables the block layer to throttle buffered
 	background writeback from the VM, making it more smooth and having
@@ -138,7 +138,7 @@ config BLK_WBT
 
 config BLK_WBT_SQ
 	bool "Single queue writeback throttling"
-	default n
+	default y
 	depends on BLK_WBT
 	---help---
 	Enable writeback throttling by default on legacy single queue devices
