From d9634fef71bb3de8ce1a34d72e03baf9264a5a96 Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Mon, 20 Feb 2017 13:44:25 +1100
Subject: [PATCH 17/24] Make BFQ default IO scheduler.

---
 block/Kconfig.iosched |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-4.11-ck2/block/Kconfig.iosched
===================================================================
--- linux-4.11-ck2.orig/block/Kconfig.iosched	2017-05-26 19:04:59.424229739 +1000
+++ linux-4.11-ck2/block/Kconfig.iosched	2017-05-26 19:04:59.424229739 +1000
@@ -41,7 +41,7 @@ config CFQ_GROUP_IOSCHED
 
 config IOSCHED_BFQ
 	tristate "BFQ I/O scheduler"
-	default n
+	default y
 	---help---
 	The BFQ I/O scheduler distributes bandwidth among all
 	processes according to their weights, regardless of the
@@ -60,7 +60,7 @@ config BFQ_GROUP_IOSCHED
 
 choice
 	prompt "Default I/O scheduler"
-	default DEFAULT_CFQ
+	default DEFAULT_BFQ
 	help
 	  Select the I/O scheduler which will be used by default for all
 	  block devices.
