diff -Naur pulseaudio-10.0-orig/build-aux/ltmain.sh pulseaudio-10.0/build-aux/ltmain.sh
--- pulseaudio-10.0-orig/build-aux/ltmain.sh	2017-01-19 02:38:21.211083934 +0000
+++ pulseaudio-10.0/build-aux/ltmain.sh	2017-01-19 02:56:42.891636512 +0000
@@ -8367,7 +8367,7 @@
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir=-L$libdir
+	      #add_dir=-L$libdir
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
