diff -Naur --strip-trailing-cr pulseaudio-11.1-orig/aclocal.m4 pulseaudio-11.1/aclocal.m4
--- pulseaudio-11.1-orig/aclocal.m4	2018-03-20 03:52:08.392204974 +0300
+++ pulseaudio-11.1/aclocal.m4	2017-09-18 17:42:40.000000000 +0300
@@ -1,4 +1,4 @@
-# generated automatically by aclocal 1.15.1 -*- Autoconf -*-
+# generated automatically by aclocal 1.16.1 -*- Autoconf -*-
 
 # Copyright (C) 1996-2017 Free Software Foundation, Inc.
 
@@ -311,7 +311,7 @@
 [am__api_version='1.15'
 dnl Some users find AM_AUTOMAKE_VERSION and mistake it for a way to
 dnl require some minimum version.  Point them to the right macro.
-m4_if([$1], [1.15.1], [],
+m4_if([$1], [1.16.1], [],
       [AC_FATAL([Do not call $0, use AM_INIT_AUTOMAKE([$1]).])])dnl
 ])
 
@@ -327,7 +327,7 @@
 # Call AM_AUTOMAKE_VERSION and AM_AUTOMAKE_VERSION so they can be traced.
 # This function is AC_REQUIREd by AM_INIT_AUTOMAKE.
 AC_DEFUN([AM_SET_CURRENT_AUTOMAKE_VERSION],
-[AM_AUTOMAKE_VERSION([1.15.1])dnl
+[AM_AUTOMAKE_VERSION([1.16.1])dnl
 m4_ifndef([AC_AUTOCONF_VERSION],
   [m4_copy([m4_PACKAGE_VERSION], [AC_AUTOCONF_VERSION])])dnl
 _AM_AUTOCONF_VERSION(m4_defn([AC_AUTOCONF_VERSION]))])
diff -Naur --strip-trailing-cr pulseaudio-11.1-orig/configure pulseaudio-11.1/configure
--- pulseaudio-11.1-orig/configure	2018-03-20 03:52:08.392204974 +0300
+++ pulseaudio-11.1/configure	2017-09-18 17:42:42.000000000 +0300
@@ -3372,7 +3372,7 @@
 ac_configure="$SHELL $ac_aux_dir/configure"  # Please don't use this var.
 
 
-am__api_version='1.15'
+am__api_version='1.16'
 
 # Find a good install program.  We prefer a C program (faster),
 # so one script is as good as another.  But avoid the broken or
