diff --git a/config.mk b/config.mk
index 81025f5..f015898 100644
--- a/config.mk
+++ b/config.mk
@@ -38,7 +38,7 @@ NAME=pstate-frequency
 #
 # [/usr/local] | /usr | other
 ##
-PREFIX?=/usr/local
+PREFIX?=/usr
 
 ##
 # power plan files
@@ -84,7 +84,7 @@ INCLUDE_SYSTEMD_UNIT?=1
 #
 # 0 NO | [1 YES]
 ##
-INCLUDE_UDEV_RULE?=1
+INCLUDE_UDEV_RULE?=0
 
 ##
 # Install README.md
@@ -94,7 +94,7 @@ INCLUDE_UDEV_RULE?=1
 #
 # 0 NO | [1 YES]
 ##
-INCLUDE_DOC?=1
+INCLUDE_DOC?=0
 
 ##
 # Install copy of the MIT License
@@ -104,7 +104,7 @@ INCLUDE_DOC?=1
 #
 # 0 NO | [1 YES]
 ##
-INCLUDE_LICENSE?=1
+INCLUDE_LICENSE?=0
 
 ##
 # Make Flags
