diff -Nur glibc-2.22.orig/sunrpc/rpc_main.c glibc-2.22/sunrpc/rpc_main.c
--- glibc-2.22.orig/sunrpc/rpc_main.c	2015-08-05 08:42:21.000000000 +0200
+++ glibc-2.22/sunrpc/rpc_main.c	2015-12-08 06:21:38.000000000 +0100
@@ -38,7 +38,11 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#if defined(__APPLE__) || defined(__CYGWIN__)
+#define gettext(X) (X)
+#else
 #include <libintl.h>
+#endif
 #include <locale.h>
 #include <ctype.h>
 #include <sys/types.h>
@@ -51,6 +55,10 @@
 #include "rpc_scan.h"
 #include "proto.h"
 
+#ifdef __CYGWIN__
+#define stat64 stat
+#endif
+
 #include "../version.h"
 #define PACKAGE _libc_intl_domainname
 
diff -Nur glibc-2.22.orig/sunrpc/rpc_scan.c glibc-2.22/sunrpc/rpc_scan.c
--- glibc-2.22.orig/sunrpc/rpc_scan.c	2015-08-05 08:42:21.000000000 +0200
+++ glibc-2.22/sunrpc/rpc_scan.c	2015-12-08 06:22:36.000000000 +0100
@@ -37,7 +37,11 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <string.h>
+#if defined(__APPLE__) || defined(__CYGWIN__)
+#define gettext(X) (X)
+#else
 #include <libintl.h>
+#endif
 #include "rpc_scan.h"
 #include "rpc_parse.h"
 #include "rpc_util.h"
