diff -urN ./gpm.cvs.orig/src/liblow.c ./gpm/src/liblow.c
--- ./gpm.cvs.orig/src/liblow.c	2002-05-28 15:13:58.000000000 -0400
+++ ./gpm/src/liblow.c	2002-06-13 16:28:45.000000000 -0400
@@ -243,9 +243,9 @@
       conn->vc=0;                 /* default handler */
       if (flag > 0) {  /* forced vc number */
          conn->vc=flag;
-         if((tty = malloc(strlen(consolename)+Gpm_cnt_digits(flag))) != NULL)
+         if((tty = malloc(strlen(consolename)+Gpm_cnt_digits(flag))) == NULL)
             gpm_report(GPM_PR_OOPS,GPM_MESS_NO_MEM);
-         snprintf(tty,strlen(consolename)-1,"%s",consolename);
+         memcpy(tty,consolename,strlen(consolename)-1);
          sprintf(&tty[strlen(consolename)-1],"%i",flag);
       } else { /* use your current vc */ 
          if (isatty(0)) tty = ttyname(0);             /* stdin */
@@ -262,7 +262,7 @@
             goto err;
          }
           
-         conn->vc=atoi(&tty[strlen(consolename)]);
+         conn->vc=atoi(&tty[strlen(consolename)-1]);
       }
 
       if (gpm_consolefd == -1)
