diff --git a/Makefile.am b/Makefile.am
old mode 100644
new mode 100755
index a479b97d..19475cf6
--- a/Makefile.am
+++ b/Makefile.am
@@ -58,9 +58,7 @@ SCRIPT_ENV	= GREP='$(GREP)' SED='$(SED)'
 
 build_scripts	= $(srcdir)/$(aux_dir)/announce-gen \
 		  $(srcdir)/$(aux_dir)/do-release-commit-and-tag \
-		  $(srcdir)/$(aux_dir)/gendocs.sh \
 		  $(srcdir)/$(aux_dir)/git-version-gen \
-		  $(srcdir)/$(aux_dir)/gnu-web-doc-update \
 		  $(srcdir)/$(aux_dir)/gnupload \
 		  $(srcdir)/$(aux_dir)/useless-if-before-free \
 		  $(srcdir)/$(aux_dir)/vc-list-files
@@ -382,51 +380,18 @@ $(lt_config_h_in): $(lt_configure_deps)
 ## Documentation. ##
 ## -------------- ##
 
-doc_dir		= $(srcdir)/doc
-
-libtool_1	= $(doc_dir)/libtool.1
-libtoolize_1	= $(doc_dir)/libtoolize.1
-notes_texi	= $(doc_dir)/notes.texi
-notes_txt	= $(doc_dir)/notes.txt
-
-EXTRA_DIST	+= $(doc_dir)/gendocs_template
-
 # A bug in automake 1.11.1 (at least) prevents us from using:
 #
 #   info_TEXINFOS	= $(doc_dir)/libtool.texi
 #
 # Producing the following error, even though srcdir is implicitly set:
 # "cannot open < ./$(srcdir)/doc/libtool.texi: No such file or directory"
-info_TEXINFOS		= doc/libtool.texi
-doc_libtool_TEXINFOS	= $(doc_dir)/PLATFORMS $(doc_dir)/fdl.texi \
-			  $(notes_texi)
-EXTRA_DIST	       += $(notes_txt)
 ## Workaround for a texinfo bug described here:
 ## http://thread.gmane.org/gmane.comp.tex.texinfo.bugs/3930
 DISTCLEANFILES	       += libtool.dvi
 
 all-local: $(notes_txt)
 
-$(notes_txt): $(notes_texi)
-	$(AM_V_GEN)$(MAKEINFO) -P '$(srcdir)/doc' --no-headers \
-	    $(MAKEINFOFLAGS) -o '$@' '$(notes_texi)'
-
-dist_man1_MANS		= $(libtool_1) $(libtoolize_1)
-MAINTAINERCLEANFILES	+= $(dist_man1_MANS)
-update_mans = \
-  PATH=".$(PATH_SEPARATOR)$$PATH"; export PATH; \
-  $(HELP2MAN) --output='$@'
-
-# It's wrong to make distributed files (e.g. $(libtool_1)) rely on
-# files created in the build tree, so instead we regenerate the
-# manual pages if the sources for the build-tree files we want to
-# run have changed.
-$(libtool_1): $(ltmain_sh)
-	$(AM_V_GEN)$(update_mans) --help-option=--help-all libtool
-$(libtoolize_1): $(libtoolize_in)
-	$(AM_V_GEN)$(update_mans) libtoolize
-
-
 ## ------------- ##
 ## Installation. ##
 ## ------------- ##
diff --git a/bootstrap.conf b/bootstrap.conf
old mode 100644
new mode 100755
index 67b88fc2..1fe340e4
--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -44,9 +44,7 @@ checkout_only_file=HACKING
 # List of programs (and minimum versions) required to bootstrap, maintain
 # and release Libtool.
 buildreq="
-        help2man   1.29        http://www.gnu.org/s/help2man
         make       3.81        http://www.gnu.org/s/make
-        makeinfo   4.8         http://www.gnu.org/s/texinfo
         xz         4.999.8beta http://tukaani.org/xz
 "
 
diff --git a/build-aux/ltmain.in b/build-aux/ltmain.in
old mode 100644
new mode 100755
index bbc76433..d53be823
--- a/build-aux/ltmain.in
+++ b/build-aux/ltmain.in
@@ -6469,7 +6469,7 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir=-L$libdir
+	      #add_dir=-L$libdir
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
diff --git a/configure.ac b/configure.ac
old mode 100644
new mode 100755
index ce34990a..f5577a6b
--- a/configure.ac
+++ b/configure.ac
@@ -177,8 +177,6 @@ GL_INIT
 ## ------------- ##
 ## Documentation ##
 ## ------------- ##
-AM_MISSING_PROG([HELP2MAN], [help2man])
-
 ## -------- ##
 ## Outputs. ##
 ## -------- ##
