--- libtool-2.4.6.orig/m4/libtool.m4	2015-01-20 17:15:19.000000000 +0100
+++ libtool-2.4.6/m4/libtool.m4	2015-06-04 11:48:09.000000000 +0200
@@ -4950,7 +4950,7 @@ m4_if([$1], [CXX], [
   _LT_TAGVAR(enable_shared_with_static_runtimes, $1)=no
   _LT_TAGVAR(export_dynamic_flag_spec, $1)=
   _LT_TAGVAR(export_symbols_cmds, $1)='$NM $libobjs $convenience | $global_symbol_pipe | $SED '\''s/.* //'\'' | sort | uniq > $export_symbols'
-  _LT_TAGVAR(hardcode_automatic, $1)=no
+  _LT_TAGVAR(hardcode_automatic, $1)=yes
   _LT_TAGVAR(hardcode_direct, $1)=no
   _LT_TAGVAR(hardcode_direct_absolute, $1)=no
   _LT_TAGVAR(hardcode_libdir_flag_spec, $1)=
@@ -6308,7 +6308,7 @@ _LT_TAGVAR(hardcode_libdir_flag_spec, $1
 _LT_TAGVAR(hardcode_libdir_separator, $1)=
 _LT_TAGVAR(hardcode_minus_L, $1)=no
 _LT_TAGVAR(hardcode_shlibpath_var, $1)=unsupported
-_LT_TAGVAR(hardcode_automatic, $1)=no
+_LT_TAGVAR(hardcode_automatic, $1)=yes
 _LT_TAGVAR(inherit_rpath, $1)=no
 _LT_TAGVAR(module_cmds, $1)=
 _LT_TAGVAR(module_expsym_cmds, $1)=
