--- xbmc-16.0-Jarvis.orig/Makefile.in	2016-02-20 16:21:19.000000000 +0100
+++ xbmc-16.0-Jarvis/Makefile.in	2016-04-08 13:41:57.000000000 +0200
@@ -314,17 +314,9 @@ $(DVDPLAYER_ARCHIVES)         : dvdpcode
 lib/cpluff/libcpluff/.libs/libcpluff.a: force
 	$(MAKE) -C lib/cpluff/libcpluff
 system/libcpluff-@ARCH@.so: lib/cpluff/libcpluff/.libs/libcpluff.a exports
-ifeq ($(findstring osx,@ARCH@), osx)
-  ifeq (@ARCH@, arm-osx)
-	$(CXX) $(LDFLAGS) -all_load -bundle -flat_namespace -undefined dynamic_lookup -read_only_relocs suppress -o $@ $<
-  else
-	$(CXX) $(LDFLAGS) -all_load -bundle -flat_namespace -undefined dynamic_lookup -lexpat $(BUNDLE1_O) -o $@ $<
-  endif
-else
 	$(CXX) $(LDFLAGS) -shared -o $@ -Wl,--whole-archive $< \
 		-Wl,--no-whole-archive  \
 		`cat xbmc/cores/DllLoader/exports/wrapper.def` xbmc/cores/DllLoader/exports/wrapper.o -lexpat
-endif
 
 exports: xbmc/cores/DllLoader/exports/exports.a xbmc/cores/DllLoader/exports/util/exports_utils.a
 	$(MAKE) -C xbmc/cores/DllLoader/exports wrapper.def
@@ -348,13 +340,8 @@ dvdpextcodecs:
 endif
 
 eventclients:
-ifeq ($(findstring osx,@ARCH@), osx)
-ifneq ($(findstring arm,@ARCH@), arm)
-	$(MAKE) -C tools/EventClients/Clients/OSXRemote
-endif
-else
 	$(MAKE) -C tools/EventClients
-endif
+
 libexif: dllloader
 	$(MAKE) -C lib/libexif
 
@@ -423,19 +410,10 @@ $(NWAOBJSXBMC) $(DIRECTORY_ARCHIVES) $(M
 include xbmc/addons/addon-bindings.mk
 
 lib@APP_NAME_LC@.so: $(OBJSXBMC) $(DYNOBJSXBMC) $(NWAOBJSXBMC) $(MAINOBJS)
-ifeq ($(findstring osx,@ARCH@), osx)
-	$(SILENT_LD) $(CXX) $(LDFLAGS) -bundle -o $@ $(MAINOBJS) -Wl,-all_load,-ObjC $(MAINOBJS) $(DYNOBJSXBMC) $(NWAOBJSXBMC) $(OBJSXBMC) $(LIBS) -read_only_relocs suppress
-else
 	$(SILENT_LD) $(CXX) $(CXXFLAGS) $(LDFLAGS) -shared -o $@ -Wl,--whole-archive $(MAINOBJS) -Wl,--no-whole-archive,--start-group $(MAINOBJS) $(DYNOBJSXBMC) $(OBJSXBMC) -Wl,--end-group -Wl,--no-undefined $(NWAOBJSXBMC) $(LIBS) -Wl,-Bsymbolic
-endif
 
 @APP_NAME_LC@.bin: $(OBJSXBMC) $(DYNOBJSXBMC) $(NWAOBJSXBMC) $(MAINOBJS)
-
-ifeq ($(findstring osx,@ARCH@), osx)
-	$(SILENT_LD) $(CXX) $(LDFLAGS) -o @APP_NAME_LC@.bin $(MAINOBJS) -Wl,-all_load,-ObjC $(MAINOBJS) $(DYNOBJSXBMC) $(NWAOBJSXBMC) $(OBJSXBMC) $(LIBS) -rdynamic
-else
-	$(SILENT_LD) $(CXX) $(CXXFLAGS) $(LDFLAGS) -o @APP_NAME_LC@.bin $(MAINOBJS) -Wl,--start-group $(MAINOBJS) $(DYNOBJSXBMC) $(OBJSXBMC) -Wl,--end-group $(NWAOBJSXBMC) $(LIBS) -rdynamic
-endif
+	$(SILENT_LD) $(CXX) $(CXXFLAGS) $(LDFLAGS) -o @APP_NAME_LC@.bin $(MAINOBJS) -Wl,--start-group $(MAINOBJS) $(DYNOBJSXBMC) $(OBJSXBMC) -Wl,--end-group $(NWAOBJSXBMC) $(LIBS) -rdynamic
 
 @APP_NAME_LC@-xrandr: xbmc-xrandr.c
 ifneq (1,@USE_X11@)
@@ -450,9 +428,6 @@ install-bin: @APP_NAME_LC@.bin # develop
 	sudo install -d $(DESTDIR)$(libdir)
 	sudo install @APP_NAME_LC@.bin $(DESTDIR)$(libdir)/@APP_NAME_LC@
 
-ifeq ($(findstring osx,@ARCH@), osx)
-	# TODO: add osx install
-else
 install: install-binaries install-arch install-datas
 
 apk obb apk-unsigned apk-obb apk-obb-unsigned apk-noobb: install
@@ -474,30 +449,11 @@ else
 	@install @APP_NAME_LC@.bin $(DESTDIR)$(libdir)/@APP_NAME_LC@/@APP_NAME_LC@.bin
 	@echo "You can run @APP_NAME_LC@ with the command '@APP_NAME_LC@'"
 endif
-endif
 
 install-arch:
 	@# Arch dependent files
-ifeq ($(findstring freebsd,@ARCH@), freebsd)
-	@find -E system addons -type f -not -iregex ".*\.git.*" \
-		-iregex ".*@ARCH@.*|.*\.vis|.*\.xbs" \
-		-exec sh -c "install -d \"$(DESTDIR)$(libdir)/@APP_NAME_LC@/\`dirname '{}'\`\"" \; \
-		-and \
-		-exec install "{}" $(DESTDIR)$(libdir)/@APP_NAME_LC@/"{}" \; \
-		-exec printf " -- %-75.75s\r" "{}" \;
-else
-ifeq ($(findstring Darwin,$(shell uname -s)),Darwin)
-	@find -E system addons -type f -not -iregex ".*\.git.*" \
-		-iregex ".*@ARCH@.*|.*\.vis|.*\.xbs" \
-		-exec sh -c "install -d \"$(DESTDIR)$(libdir)/@APP_NAME_LC@/\`dirname '{}'\`\"" \; \
-		-and \
-		-exec install "{}" $(DESTDIR)$(libdir)/xbmc/"{}" \; \
-		-exec printf " -- %-75.75s\r" "{}" \;
-else
 	@find system addons -regextype posix-extended -type f -not -iregex ".*\.git.*" -iregex ".*\.so|.*\.so\.[0-9].*|.*\.vis|.*\.xbs" -exec install -D "{}" $(DESTDIR)$(libdir)/@APP_NAME_LC@/"{}" \; -printf " -- %-75.75f\r"
 	@find -L addons -regextype posix-extended -type f -not -iregex ".*\.git.*" -iregex ".*\.so|.*\.so\.[0-9].*" -exec cp -d "{}" $(DESTDIR)$(libdir)/@APP_NAME_LC@/"{}" \; -printf " -- %-75.75f\r"
-endif
-endif
 
 install-scripts:
 	@install -d $(DESTDIR)$(bindir)
@@ -620,11 +576,7 @@ $(CHECK_LIBS): force
 	@$(MAKE) CXXFLAGS="$(CXXFLAGS) -DGTEST_USE_OWN_TR1_TUPLE=1" $(if $(V),,-s) -C $(@D)
 
 @APP_NAME_LC@-test: $(CHECK_LIBS) $(OBJSXBMC) $(DYNOBJSXBMC) $(NWAOBJSXBMC) $(GTEST_LIBS)
-ifeq ($(findstring osx,@ARCH@), osx)
-	$(SILENT_LD) $(CXX) $(CXXFLAGS) $(LDFLAGS) $(GTEST_INCLUDES) -o $@ -Wl,-all_load,-ObjC $(DYNOBJSXBMC) $(NWAOBJSXBMC) $(OBJSXBMC) $(GTEST_LIBS) $(CHECK_LIBS) $(LIBS) $(CHECK_LIBADD) -rdynamic
-else
-	$(SILENT_LD) $(CXX) $(CXXFLAGS) $(LDFLAGS) $(GTEST_INCLUDES) -o $@ -Wl,--whole-archive $(DYNOBJSXBMC) $(OBJSXBMC) $(GTEST_LIBS) $(CHECK_LIBS) -Wl,--no-whole-archive $(NWAOBJSXBMC) $(LIBS) $(CHECK_LIBADD) -rdynamic
-endif
+	$(SILENT_LD) $(CXX) $(CXXFLAGS) $(LDFLAGS) $(GTEST_INCLUDES) -o $@ -Wl,--whole-archive $(DYNOBJSXBMC) $(OBJSXBMC) $(GTEST_LIBS) $(CHECK_LIBS) -Wl,--no-whole-archive $(NWAOBJSXBMC) $(LIBS) $(CHECK_LIBADD)
 else
 # Give a message that the framework is not configured, but don't fail.
 check testsuite testframework:
