diff -Naur tiff-4.0.8-orig/configure.ac tiff-4.0.8/configure.ac
--- tiff-4.0.8-orig/configure.ac	2017-06-22 20:17:38.601392414 +0300
+++ tiff-4.0.8/configure.ac	2017-06-22 20:19:59.717869850 +0300
@@ -885,13 +885,13 @@
 
 dnl AX_CHECK_GL sets GL_CFLAGS & GL_LIBS.  Also PTHREAD_LIBS,
 dnl PTHREAD_CFLAGS, & PTHREAD_CC as a side-effect
-AX_CHECK_GL
+AX_CHECK_GL([no_gl=no],[no_gl=yes])
 
 dnl AX_CHECK_GLU sets GLU_CFLAGS & GLU_LIBS
-AX_CHECK_GLU
+AX_CHECK_GLU([no_glu=no],[no_glu=yes])
 
 dnl AX_CHECK_GLUT sets GLUT_CFLAGS & GLUT_LIBS
-AX_CHECK_GLUT
+AX_CHECK_GLUT([no_glut=no],[no_glut=yes])
 
 if test "$no_x" != "yes" -a "$no_gl" != "yes" \
 	-a "$no_glu" != "yes" -a "$no_glut" != "yes" ; then
