diff --git a/speedtest.go b/speedtest.go
old mode 100644
new mode 100755
index 8d4c2a3..03921eb
--- a/speedtest.go
+++ b/speedtest.go
@@ -33,7 +33,7 @@ func runTest(c *cli.Context) {
 
 	// if we are *not* running a report then say hello to everyone
 	if !debug.REPORT {
-		fmt.Printf("github.com/zpeters/speedtest -- unofficial cli for speedtest.net\n")
+		fmt.Printf("ΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞΞ\n")
 	}
 
 	// if we are in debug mode print outa an environment report
