--- firefox-45.0.orig/xpcom/ds/nsMathUtils.h	2016-03-03 22:48:07.000000000 +0100
+++ firefox-45.0/xpcom/ds/nsMathUtils.h	2016-03-13 21:15:29.543246177 +0100
@@ -103,12 +103,16 @@ NS_finite(double aNum)
 #ifdef WIN32
   // NOTE: '!!' casts an int to bool without spamming MSVC warning C4800.
   return !!_finite(aNum);
-#elif defined(XP_DARWIN)
+#elif defined(XP_DARWIN) || defined(__UCLIBC__)
   // Darwin has deprecated |finite| and recommends |isfinite|. The former is
   // not present in the iOS SDK.
   return std::isfinite(aNum);
 #else
-  return finite(aNum);
+#ifdef _GLIBCXX_CMATH
+  return std::isfinite(aNum);
+#else
+  return isfinite(aNum);
+#endif
 #endif
 }
 
