diff -Naur squid-4.0.18-orig/configure squid-4.0.18/configure
--- squid-4.0.18-orig/configure	2017-03-31 14:56:59.109107987 +0300
+++ squid-4.0.18/configure	2017-03-31 15:01:13.291741972 +0300
@@ -29393,7 +29393,7 @@
     if test "$cross_compiling" = yes; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
+_as_fn_error $? "cannot run test program while cross compiling
 See \`config.log' for more details" "$LINENO" 5; }
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
@@ -29418,7 +29418,7 @@
     if test "$cross_compiling" = yes; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
+_as_fn_error $? "cannot run test program while cross compiling
 See \`config.log' for more details" "$LINENO" 5; }
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
diff -Naur squid-4.0.18-orig/src/Makefile.in squid-4.0.18/src/Makefile.in
--- squid-4.0.18-orig/src/Makefile.in	2017-03-31 14:56:59.065108781 +0300
+++ squid-4.0.18/src/Makefile.in	2017-03-31 14:59:03.920696104 +0300
@@ -7630,7 +7630,7 @@
 
 # cf_gen builds the configuration files.
 cf_gen$(EXEEXT): $(cf_gen_SOURCES) $(cf_gen_DEPENDENCIES) cf_gen_defines.cci
-	$(BUILDCXX) $(BUILDCXXFLAGS) -o $@ $(srcdir)/cf_gen.cc -I$(srcdir) -I$(top_builddir)/include/ -I$(top_builddir)/src
+	g++ -o $@ $(srcdir)/cf_gen.cc -I$(srcdir) -I$(top_builddir)/include/ -I$(top_builddir)/src
 
 # squid.conf.default is built by cf_gen when making cf_parser.cci
 squid.conf.default squid.conf.documented: cf_parser.cci
diff -Naur squid-4.0.18-orig/src/mime.conf.default squid-4.0.18/src/mime.conf.default
--- squid-4.0.18-orig/src/mime.conf.default	2017-03-31 14:56:59.089108349 +0300
+++ squid-4.0.18/src/mime.conf.default	2017-03-31 14:58:23.542524084 +0300
@@ -1,20 +1,3 @@
-## Copyright (C) 1996-2017 The Squid Software Foundation and contributors
-##
-## Squid software is distributed under GPLv2+ license and includes
-## contributions from numerous individuals and organizations.
-## Please see the COPYING and CONTRIBUTORS files for details.
-##
-#
-# This file associates URL patterns for servers or services
-# that don't automatically include Content-Type (like ftp) with a mime type
-# and a graphical icon.
-#
-# Content-Encodings are taken from section 3.1 of RFC2068 (HTTP/1.1)
-#
-# This file has the format :
-#
-# regexp	content-type			icon				encoding mode	actions
-#--------------------------------------------------------------------------------------------------------
 \.gif$			image/gif		silk/image.png			-	image	+download
 \.mime$			www/mime		silk/page_white_text.png	-	ascii	+download
 ^internal-dirup$	-			silk/arrow_up.png		-	-
@@ -190,7 +173,4 @@
 \.xhtml$	application/xhtml		silk/page_world.png		-	ascii	+download
 \.xml$		text/xml			silk/page_world.png		-	ascii	+download
 \.xsl$		text/xml			silk/layout.png			-	ascii	+download
-\.xyz$		chemical/x-xyz			silk/chart_line.png		-	image	+download
-#
-# the default
-.		text/plain			silk/bullet_red.png		-	image	+download +view
+\.xyz$		chemical/x-xyz			silk/chart_line.png		-	image	+download
\ No newline at end of file
