diff -Naur libpcap-libpcap-1.8.1-orig/configure libpcap-libpcap-1.8.1/configure
--- libpcap-libpcap-1.8.1-orig/configure	2017-01-19 18:58:41.640474757 +0000
+++ libpcap-libpcap-1.8.1/configure	2017-01-19 18:58:59.897826766 +0000
@@ -5744,7 +5744,7 @@
 	if test x$with_libnl != xno ; then
 		have_any_nl="no"
 
-                incdir=-I/usr/include/libnl3
+                #incdir=-I/usr/include/libnl3
                 libnldir=
                 if test x$withval != x ; then
                   libnldir=-L${withval}/lib/.libs
diff -Naur libpcap-libpcap-1.8.1-orig/configure.ac libpcap-libpcap-1.8.1/configure.ac
--- libpcap-libpcap-1.8.1-orig/configure.ac	2017-01-19 18:58:41.640474757 +0000
+++ libpcap-libpcap-1.8.1/configure.ac	2017-01-19 18:59:13.719603229 +0000
@@ -491,7 +491,7 @@
 	if test x$with_libnl != xno ; then
 		have_any_nl="no"
 
-                incdir=-I/usr/include/libnl3
+                #incdir=-I/usr/include/libnl3
                 libnldir=
                 if test x$withval != x ; then
                   libnldir=-L${withval}/lib/.libs
