diff -Nur newlib-2.2.0-1.orig/configure newlib-2.2.0-1/configure
--- newlib-2.2.0-1.orig/configure	2014-07-05 23:09:07.000000000 +0200
+++ newlib-2.2.0-1/configure	2015-11-25 06:33:11.461178398 +0100
@@ -6985,7 +6985,7 @@
 
 # Some systems (e.g., one of the i386-aix systems the gas testers are
 # using) don't handle "\$" correctly, so don't use it here.
-tooldir='${exec_prefix}'/${target_noncanonical}
+tooldir='${exec_prefix}'
 build_tooldir=${tooldir}
 
 # Create a .gdbinit file which runs the one in srcdir
