--- a/CMakeLists.txt	2016-10-05 14:43:09.403647787 +0200
+++ b/CMakeLists.txt	2016-10-05 14:43:22.947132376 +0200
@@ -263,7 +263,7 @@
 set(LLVM_TARGET_ARCH "host"
   CACHE STRING "Set target to use for LLVM JIT or use \"host\" for automatic detection.")
 
-option(LLVM_ENABLE_TERMINFO "Use terminfo database if available." ON)
+option(LLVM_ENABLE_TERMINFO "Use terminfo database if available." OFF)
 
 option(LLVM_ENABLE_THREADS "Use threads if available." ON)
 
