From bda4fd60c4c798f7eb0078f727e08c3e4d228669 Mon Sep 17 00:00:00 2001
From: arvidn <arvid@cs.umu.se>
Date: Sun, 12 Mar 2017 13:02:17 -0400
Subject: [PATCH] fix build with newer boost

---
 test/test_ssl.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/test/test_ssl.cpp b/test/test_ssl.cpp
index 064e9afebb..618e2866af 100644
--- a/test/test_ssl.cpp
+++ b/test/test_ssl.cpp
@@ -44,7 +44,6 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <boost/asio/connect.hpp>
 
 #ifdef TORRENT_USE_OPENSSL
-#include <boost/asio/ssl/error.hpp> // for asio::error::get_ssl_category()
 #include <boost/asio/ssl.hpp>
 
 using namespace libtorrent;
