From 8736a59adc65a9668d8230ce045433508c21fb96 Mon Sep 17 00:00:00 2001
From: arvidn <arvid@cs.umu.se>
Date: Sun, 21 May 2017 23:19:33 -0400
Subject: [PATCH] fix faulty fix to enum_routes

---
 src/enum_net.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/enum_net.cpp b/src/enum_net.cpp
index 7dc7fdad3c..26beccfd24 100644
--- a/src/enum_net.cpp
+++ b/src/enum_net.cpp
@@ -825,7 +825,8 @@ namespace libtorrent
 	{
 		rtm = (rt_msghdr*)next;
 		if (rtm->rtm_version != RTM_VERSION
-			|| rtm->rtm_type != RTM_ADD)
+			|| (rtm->rtm_type != RTM_ADD
+			&& rtm->rtm_type != RTM_GET))
 		{
 			continue;
 		}
