Index: gcc-6-6.2.1-4.1/src/libgo/runtime/netpoll.goc
===================================================================
--- gcc-6-6.2.1-4.1.orig/src/libgo/runtime/netpoll.goc
+++ gcc-6-6.2.1-4.1/src/libgo/runtime/netpoll.goc
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package net
 
@@ -98,7 +98,7 @@ func runtime_pollServerInit() {
 	runtime_netpollinit();
 }
 
-func runtime_pollOpen(fd uintptr) (pd *PollDesc, errno int) {
+func runtime_pollOpen(fd uintptr) (pd *PollDesc, errno1 int) {
 	pd = allocPollDesc();
 	runtime_lock(pd);
 	if(pd->wg != nil && pd->wg != READY)
@@ -114,7 +114,7 @@ func runtime_pollOpen(fd uintptr) (pd *P
 	pd->wd = 0;
 	runtime_unlock(pd);
 
-	errno = runtime_netpollopen(fd, pd);
+	errno1 = runtime_netpollopen(fd, pd);
 }
 
 func runtime_pollClose(pd *PollDesc) {
