diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
old mode 100644
new mode 100755
index 600fc047e5..a8090f93c9
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1439,7 +1439,7 @@ config("chromium_code") {
   if (is_win) {
     cflags = [ "/W4" ]  # Warning level 4.
   } else {
-    cflags = [ "-Wall" ]
+    cflags = [ "" ]
     if (treat_warnings_as_errors) {
       cflags += [ "-Werror" ]
 
@@ -1524,7 +1524,7 @@ config("no_chromium_code") {
     if (is_clang && !is_nacl) {
       # TODO(thakis): Remove !is_nacl once
       # https://codereview.webrtc.org/1552863002/ made its way into chromium.
-      cflags += [ "-Wall" ]
+      # cflags += [ "-Wall" ]
     }
   }
 
