diff -Naur coreutils-8.26-orig/Makefile.in coreutils-8.26/Makefile.in
--- coreutils-8.26-orig/Makefile.in	2017-02-01 23:09:27.679046822 +0000
+++ coreutils-8.26/Makefile.in	2017-02-01 23:10:40.258649613 +0000
@@ -5895,7 +5895,7 @@
   tests/pr/w72l24f-ll
 
 all: $(BUILT_SOURCES)
-	$(MAKE) $(AM_MAKEFLAGS) all-recursive
+	$(MAKE) all-recursive
 
 .SUFFIXES:
 .SUFFIXES: .1 .c .dvi .log .o .obj .pl .pl$(EXEEXT) .ps .sed .sh .sh$(EXEEXT) .sin .trs .x .xpl .xpl$(EXEEXT) .y
@@ -5936,7 +5936,7 @@
 
 lib/config.h: lib/stamp-h1
 	@test -f $@ || rm -f lib/stamp-h1
-	@test -f $@ || $(MAKE) $(AM_MAKEFLAGS) lib/stamp-h1
+	@test -f $@ || $(MAKE) lib/stamp-h1
 
 lib/stamp-h1: $(top_srcdir)/lib/config.hin $(top_builddir)/config.status
 	@rm -f lib/stamp-h1
@@ -11347,7 +11347,7 @@
 	fi
 $(srcdir)/doc/version.texi:  $(srcdir)/doc/stamp-vti
 $(srcdir)/doc/stamp-vti: doc/coreutils.texi $(top_srcdir)/configure
-	@test -f doc/$(am__dirstamp) || $(MAKE) $(AM_MAKEFLAGS) doc/$(am__dirstamp)
+	@test -f doc/$(am__dirstamp) || $(MAKE) doc/$(am__dirstamp)
 	@(dir=.; test -f ./doc/coreutils.texi || dir=$(srcdir); \
 	set `$(SHELL) $(top_srcdir)/build-aux/mdate-sh $$dir/doc/coreutils.texi`; \
 	echo "@set UPDATED $$1 $$2 $$3"; \
@@ -11530,11 +11530,11 @@
 	  else \
 	    local_target="$$target"; \
 	  fi; \
-	  ($(am__cd) $$subdir && $(MAKE) $(AM_MAKEFLAGS) $$local_target) \
+	  ($(am__cd) $$subdir && $(MAKE) $$local_target) \
 	  || eval $$failcom; \
 	done; \
 	if test "$$dot_seen" = "no"; then \
-	  $(MAKE) $(AM_MAKEFLAGS) "$$target-am" || exit 1; \
+	  $(MAKE) "$$target-am" || exit 1; \
 	fi; test -z "$$fail"
 
 ID: $(am__tagged_files)
@@ -11615,7 +11615,7 @@
 # to avoid problems with "make -n".
 .log.trs:
 	rm -f $< $@
-	$(MAKE) $(AM_MAKEFLAGS) $<
+	$(MAKE) $<
 
 # Leading 'am--fnord' is there to ensure the list of targets does not
 # expand to empty, as could happen e.g. with make check TESTS=''.
@@ -11640,7 +11640,7 @@
 	  echo "fatal: making $(TEST_SUITE_LOG): possible infinite" \
 	       "recursion detected" >&2; \
 	elif test -n "$$redo_logs"; then \
-	  am__remaking_logs=yes $(MAKE) $(AM_MAKEFLAGS) $$redo_logs; \
+	  am__remaking_logs=yes $(MAKE) $$redo_logs; \
 	fi; \
 	if $(am__make_dryrun); then :; else \
 	  st=0;  \
@@ -11737,7 +11737,7 @@
 	log_list=`for i in $$bases; do echo $$i.log; done`; \
 	trs_list=`for i in $$bases; do echo $$i.trs; done`; \
 	log_list=`echo $$log_list`; trs_list=`echo $$trs_list`; \
-	$(MAKE) $(AM_MAKEFLAGS) $(TEST_SUITE_LOG) TEST_LOGS="$$log_list"; \
+	$(MAKE) $(TEST_SUITE_LOG) TEST_LOGS="$$log_list"; \
 	exit $$?;
 recheck: all 
 	@test -z "$(TEST_SUITE_LOG)" || rm -f $(TEST_SUITE_LOG)
@@ -11746,7 +11746,7 @@
 	         | $(am__list_recheck_tests)` || exit 1; \
 	log_list=`for i in $$bases; do echo $$i.log; done`; \
 	log_list=`echo $$log_list`; \
-	$(MAKE) $(AM_MAKEFLAGS) $(TEST_SUITE_LOG) \
+	$(MAKE) $(TEST_SUITE_LOG) \
 	        am__force_recheck=am--force-recheck \
 	        TEST_LOGS="$$log_list"; \
 	exit $$?
@@ -11837,10 +11837,10 @@
 	    dir1=$$subdir; dir2="$(top_distdir)"; \
 	    $(am__relativize); \
 	    new_top_distdir=$$reldir; \
-	    echo " (cd $$subdir && $(MAKE) $(AM_MAKEFLAGS) top_distdir="$$new_top_distdir" distdir="$$new_distdir" \\"; \
+	    echo " (cd $$subdir && $(MAKE) top_distdir="$$new_top_distdir" distdir="$$new_distdir" \\"; \
 	    echo "     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)"; \
 	    ($(am__cd) $$subdir && \
-	      $(MAKE) $(AM_MAKEFLAGS) \
+	      $(MAKE) \
 	        top_distdir="$$new_top_distdir" \
 	        distdir="$$new_distdir" \
 		am__remove_distdir=: \
@@ -11850,7 +11850,7 @@
 	      || exit 1; \
 	  fi; \
 	done
-	$(MAKE) $(AM_MAKEFLAGS) \
+	$(MAKE) \
 	  top_distdir="$(top_distdir)" distdir="$(distdir)" \
 	  dist-info dist-hook
 	-test -n "$(am__skip_mode_fix)" \
@@ -11895,7 +11895,7 @@
 	$(am__post_remove_distdir)
 
 dist dist-all:
-	$(MAKE) $(AM_MAKEFLAGS) $(DIST_TARGETS) am__post_remove_distdir='@:'
+	$(MAKE) $(DIST_TARGETS) am__post_remove_distdir='@:'
 	$(am__post_remove_distdir)
 
 # This target untars the dist file and tries a VPATH configuration.  Then
@@ -11925,33 +11925,33 @@
 	test -d $(distdir)/_build || exit 0; \
 	dc_install_base=`$(am__cd) $(distdir)/_inst && pwd | sed -e 's,^[^:\\/]:[\\/],/,'` \
 	  && dc_destdir="$${TMPDIR-/tmp}/am-dc-$$$$/" \
-	  && $(MAKE) $(AM_MAKEFLAGS) distcheck-hook \
+	  && $(MAKE) distcheck-hook \
 	  && am__cwd=`pwd` \
 	  && $(am__cd) $(distdir)/_build/sub \
 	  && ../../configure \
 	    $(AM_DISTCHECK_CONFIGURE_FLAGS) \
 	    $(DISTCHECK_CONFIGURE_FLAGS) \
 	    --srcdir=../.. --prefix="$$dc_install_base" \
-	  && $(MAKE) $(AM_MAKEFLAGS) \
-	  && $(MAKE) $(AM_MAKEFLAGS) dvi \
-	  && $(MAKE) $(AM_MAKEFLAGS) check \
-	  && $(MAKE) $(AM_MAKEFLAGS) install \
-	  && $(MAKE) $(AM_MAKEFLAGS) installcheck \
-	  && $(MAKE) $(AM_MAKEFLAGS) uninstall \
-	  && $(MAKE) $(AM_MAKEFLAGS) distuninstallcheck_dir="$$dc_install_base" \
+	  && $(MAKE) \
+	  && $(MAKE) dvi \
+	  && $(MAKE) check \
+	  && $(MAKE) install \
+	  && $(MAKE) installcheck \
+	  && $(MAKE) uninstall \
+	  && $(MAKE) distuninstallcheck_dir="$$dc_install_base" \
 	        distuninstallcheck \
 	  && chmod -R a-w "$$dc_install_base" \
 	  && ({ \
 	       (cd ../.. && umask 077 && mkdir "$$dc_destdir") \
-	       && $(MAKE) $(AM_MAKEFLAGS) DESTDIR="$$dc_destdir" install \
-	       && $(MAKE) $(AM_MAKEFLAGS) DESTDIR="$$dc_destdir" uninstall \
-	       && $(MAKE) $(AM_MAKEFLAGS) DESTDIR="$$dc_destdir" \
+	       && $(MAKE) DESTDIR="$$dc_destdir" install \
+	       && $(MAKE) DESTDIR="$$dc_destdir" uninstall \
+	       && $(MAKE) DESTDIR="$$dc_destdir" \
 	            distuninstallcheck_dir="$$dc_destdir" distuninstallcheck; \
 	      } || { rm -rf "$$dc_destdir"; exit 1; }) \
 	  && rm -rf "$$dc_destdir" \
-	  && $(MAKE) $(AM_MAKEFLAGS) dist \
+	  && $(MAKE) dist \
 	  && rm -rf $(DIST_ARCHIVES) \
-	  && $(MAKE) $(AM_MAKEFLAGS) distcleancheck \
+	  && $(MAKE) distcleancheck \
 	  && cd "$$am__cwd" \
 	  || exit 1
 	$(am__post_remove_distdir)
@@ -11985,9 +11985,9 @@
 	       $(distcleancheck_listfiles) ; \
 	       exit 1; } >&2
 check-am: all-am
-	$(MAKE) $(AM_MAKEFLAGS) check-TESTS check-local
+	$(MAKE) check-TESTS check-local
 check: $(BUILT_SOURCES)
-	$(MAKE) $(AM_MAKEFLAGS) check-recursive
+	$(MAKE) check-recursive
 all-am: Makefile $(INFO_DEPS) $(LIBRARIES) $(PROGRAMS) $(MANS) \
 		$(HEADERS) all-local
 installdirs: installdirs-recursive
@@ -11996,22 +11996,22 @@
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
 install: $(BUILT_SOURCES)
-	$(MAKE) $(AM_MAKEFLAGS) install-recursive
+	$(MAKE) install-recursive
 install-exec: install-exec-recursive
 install-data: install-data-recursive
 uninstall: uninstall-recursive
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) install-exec-am install-data-am
 
 installcheck: installcheck-recursive
 install-strip:
 	if test -z '$(STRIP)'; then \
-	  $(MAKE) $(AM_MAKEFLAGS) INSTALL_PROGRAM="$(INSTALL_STRIP_PROGRAM)" \
+	  $(MAKE) INSTALL_PROGRAM="$(INSTALL_STRIP_PROGRAM)" \
 	    install_sh_PROGRAM="$(INSTALL_STRIP_PROGRAM)" INSTALL_STRIP_FLAG=-s \
 	      install; \
 	else \
-	  $(MAKE) $(AM_MAKEFLAGS) INSTALL_PROGRAM="$(INSTALL_STRIP_PROGRAM)" \
+	  $(MAKE) INSTALL_PROGRAM="$(INSTALL_STRIP_PROGRAM)" \
 	    install_sh_PROGRAM="$(INSTALL_STRIP_PROGRAM)" INSTALL_STRIP_FLAG=-s \
 	    "INSTALL_PROGRAM_ENV=STRIPPROG='$(STRIP)'" install; \
 	fi
@@ -12096,7 +12096,7 @@
 install-exec-am: install-binPROGRAMS install-exec-local \
 	install-pkglibexecPROGRAMS
 	@$(NORMAL_INSTALL)
-	$(MAKE) $(AM_MAKEFLAGS) install-exec-hook
+	$(MAKE) install-exec-hook
 install-html: install-html-recursive
 
 install-html-am: $(HTMLS)
