diff -Naur coreutils-8.27-orig/Makefile.am coreutils-8.27/Makefile.am
--- coreutils-8.27-orig/Makefile.am	2017-06-22 02:42:20.496825003 +0300
+++ coreutils-8.27/Makefile.am	2017-06-22 02:43:08.226872608 +0300
@@ -17,7 +17,7 @@
 
 ALL_RECURSIVE_TARGETS =
 
-SUBDIRS = po . gnulib-tests
+SUBDIRS = po
 
 changelog_etc =				\
   build-aux/ChangeLog-2007		\
@@ -213,6 +213,3 @@
 
 include $(top_srcdir)/lib/local.mk
 include $(top_srcdir)/src/local.mk
-include $(top_srcdir)/doc/local.mk
-include $(top_srcdir)/man/local.mk
-include $(top_srcdir)/tests/local.mk
