diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
old mode 100644
new mode 100755
index 20d504455b..9a58255fa4
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -644,7 +644,7 @@ config("compiler_cpu_abi") {
     if (current_cpu == "x64") {
       cflags += [
         "-m64",
-        "-march=x86-64",
+        "-march=westmere",
       ]
       ldflags += [ "-m64" ]
     } else if (current_cpu == "x86") {
