--- iproute2-3.12.0.orig/include/iptables_common.h	2013-11-23 02:10:33.000000000 +0100
+++ iproute2-3.12.0/include/iptables_common.h	2014-04-05 05:21:46.946376336 +0200
@@ -1,5 +1,8 @@
 #ifndef _IPTABLES_COMMON_H
 #define _IPTABLES_COMMON_H
+
+#include <stdint.h>
+
 /* Shared definitions between ipv4 and ipv6. */
 
 enum exittype {
@@ -43,9 +46,9 @@ extern char *lib_dir;
   extern void init_extensions(void);
 #endif
 
-#define __be32	u_int32_t
-#define __le32	u_int32_t
-#define __be16	u_int16_t
-#define __le16	u_int16_t
+#define __be32	uint32_t
+#define __le32	uint32_t
+#define __be16	uint16_t
+#define __le16	uint16_t
 
 #endif /*_IPTABLES_COMMON_H*/
