--- a/support/nsm/rpc.c	2017-07-18 06:57:01.877281996 +0000
+++ b/support/nsm/rpc.c	2017-07-18 06:57:24.540290393 +0000
@@ -466,7 +466,7 @@
 	if (xdr_u_long(xdrs, &port) == FALSE)
 		xlog(L_ERROR, "%s: can't decode pmap reply",
 			__func__);
-	if (port > UINT16_MAX) {
+	if (port > UINT_MAX) {
 		xlog(L_ERROR, "%s: bad port number",
 			__func__);
 		port = 0;
@@ -503,7 +503,7 @@
 
 	xdr_free((xdrproc_t)xdr_wrapstring, (char *)&uaddr);
 
-	if (port < 0 || port > UINT16_MAX) {
+	if (port < 0 || port > UINT_MAX) {
 		xlog(L_ERROR, "%s: bad port number",
 			__func__);
 		return 0;
