--- a/utils/osd_login/Makefile.in.orig	2017-05-29 13:36:19.363634178 +0000
+++ b/utils/osd_login/Makefile.in	2017-05-29 13:36:46.459906798 +0000
@@ -339,7 +339,7 @@
 
 # These binaries go in /sbin (not /usr/sbin), and that cannot be
 # overridden at config time.
-sbindir = /sbin
+sbindir = /usr/sbin
 sharedstatedir = @sharedstatedir@
 srcdir = @srcdir@
 startstatd = @startstatd@
--- a/utils/nfsdcltrack/Makefile.in.orig	2017-05-29 13:36:27.866843594 +0000
+++ b/utils/nfsdcltrack/Makefile.in	2017-05-29 13:36:57.323082809 +0000
@@ -394,7 +394,7 @@
 
 # These binaries go in /sbin (not /usr/sbin), and that cannot be
 # overridden at config time. The kernel "knows" the /sbin name.
-sbindir = /sbin
+sbindir = /usr/sbin
 sharedstatedir = @sharedstatedir@
 srcdir = @srcdir@
 startstatd = @startstatd@
--- a/utils/mount/Makefile.in.orig	2017-05-29 13:36:33.436762554 +0000
+++ b/utils/mount/Makefile.in	2017-05-29 13:37:06.322952738 +0000
@@ -416,7 +416,7 @@
 
 # These binaries go in /sbin (not /usr/sbin), and that cannot be
 # overridden at config time.
-sbindir = /sbin
+sbindir = /usr/sbin
 sharedstatedir = @sharedstatedir@
 srcdir = @srcdir@
 startstatd = @startstatd@
