diff --git a/Makefile b/Makefile
old mode 100644
new mode 100755
index 39989bf..2af9f0e
--- a/Makefile
+++ b/Makefile
@@ -54,7 +54,7 @@ DFLAGS?= -DDEBUG_PPPD -DTRUST_PPPD_TO_DIE
 #
 #KERNELSRC=/lib/modules/`uname -r`/build/
 KERNELSRC?=./linux
-OSFLAGS?= -DLINUX -I$(KERNELSRC)/include/
+OSFLAGS?= -DLINUX -I$(kernel_path)/include/
 #
 # Uncomment the following to use the kernel interface under Linux
 # This requires the pppol2tp-linux-2.4.27.patch patch from contrib
@@ -107,7 +107,7 @@ LDLIBS+= -lm
 EXEC=xl2tpd
 CONTROL_EXEC=xl2tpd-control
 
-PREFIX?=/usr/local
+PREFIX?=/usr
 SBINDIR?=$(DESTDIR)${PREFIX}/sbin
 BINDIR?=$(DESTDIR)${PREFIX}/bin
 MANDIR?=$(DESTDIR)${PREFIX}/share/man
