diff -Naur harfbuzz-1.7.6-orig/test/api/Makefile.in harfbuzz-1.7.6/test/api/Makefile.in
--- harfbuzz-1.7.6-orig/test/api/Makefile.in	2018-04-29 00:48:59.546795684 +0300
+++ harfbuzz-1.7.6/test/api/Makefile.in	2018-03-07 17:56:53.000000000 +0300
@@ -134,11 +134,9 @@
 @HAVE_FREETYPE_TRUE@@HAVE_GLIB_TRUE@@HAVE_OT_TRUE@am__EXEEXT_3 = test-ot-math$(EXEEXT) \
 @HAVE_FREETYPE_TRUE@@HAVE_GLIB_TRUE@@HAVE_OT_TRUE@	$(am__EXEEXT_1)
 @HAVE_GLIB_TRUE@am__EXEEXT_4 = test-blob$(EXEEXT) test-buffer$(EXEEXT) \
-@HAVE_GLIB_TRUE@	test-common$(EXEEXT) test-font$(EXEEXT) \
 @HAVE_GLIB_TRUE@	test-object$(EXEEXT) test-set$(EXEEXT) \
 @HAVE_GLIB_TRUE@	test-shape$(EXEEXT) test-subset-cmap$(EXEEXT) \
 @HAVE_GLIB_TRUE@	test-subset-glyf$(EXEEXT) \
-@HAVE_GLIB_TRUE@	test-subset-hdmx$(EXEEXT) \
 @HAVE_GLIB_TRUE@	test-subset-hmtx$(EXEEXT) \
 @HAVE_GLIB_TRUE@	test-subset-os2$(EXEEXT) test-unicode$(EXEEXT) \
 @HAVE_GLIB_TRUE@	test-version$(EXEEXT) $(am__EXEEXT_1) \
@@ -183,8 +181,6 @@
 @HAVE_GLIB_TRUE@test_cplusplus_DEPENDENCIES =  \
 @HAVE_GLIB_TRUE@	$(top_builddir)/src/libharfbuzz.la \
 @HAVE_GLIB_TRUE@	$(am__DEPENDENCIES_1)
-test_font_SOURCES = test-font.c
-test_font_OBJECTS = test-font.$(OBJEXT)
 test_font_LDADD = $(LDADD)
 @HAVE_GLIB_TRUE@test_font_DEPENDENCIES =  \
 @HAVE_GLIB_TRUE@	$(top_builddir)/src/libharfbuzz.la \
@@ -234,8 +230,6 @@
 test_subset_glyf_OBJECTS = test-subset-glyf.$(OBJEXT)
 @HAVE_GLIB_TRUE@test_subset_glyf_DEPENDENCIES = $(am__DEPENDENCIES_2) \
 @HAVE_GLIB_TRUE@	$(top_builddir)/src/libharfbuzz-subset.la
-test_subset_hdmx_SOURCES = test-subset-hdmx.c
-test_subset_hdmx_OBJECTS = test-subset-hdmx.$(OBJEXT)
 @HAVE_GLIB_TRUE@test_subset_hdmx_DEPENDENCIES = $(am__DEPENDENCIES_2) \
 @HAVE_GLIB_TRUE@	$(top_builddir)/src/libharfbuzz-subset.la
 test_subset_hmtx_SOURCES = test-subset-hmtx.c
@@ -311,16 +305,16 @@
 am__v_CXXLD_0 = @echo "  CXXLD   " $@;
 am__v_CXXLD_1 = 
 SOURCES = test-blob.c test-buffer.c test-c.c test-common.c \
-	$(test_cplusplus_SOURCES) test-font.c test-object.c \
+	$(test_cplusplus_SOURCES) test-object.c \
 	test-ot-color.c test-ot-math.c test-ot-tag.c test-set.c \
 	test-shape.c test-subset-cmap.c test-subset-glyf.c \
-	test-subset-hdmx.c test-subset-hmtx.c test-subset-os2.c \
+	test-subset-hmtx.c test-subset-os2.c \
 	test-unicode.c test-version.c
 DIST_SOURCES = test-blob.c test-buffer.c test-c.c test-common.c \
-	$(am__test_cplusplus_SOURCES_DIST) test-font.c test-object.c \
+	$(am__test_cplusplus_SOURCES_DIST) test-object.c \
 	test-ot-color.c test-ot-math.c test-ot-tag.c test-set.c \
 	test-shape.c test-subset-cmap.c test-subset-glyf.c \
-	test-subset-hdmx.c test-subset-hmtx.c test-subset-os2.c \
+	test-subset-hmtx.c test-subset-os2.c \
 	test-unicode.c test-version.c
 am__can_run_installinfo = \
   case $$AM_UPDATE_INFO_DIR in \
@@ -748,9 +742,7 @@
 
 # Tests for header compilation
 @HAVE_GLIB_TRUE@TEST_PROGS = test-blob test-buffer test-common \
-@HAVE_GLIB_TRUE@	test-font test-object test-set test-shape \
 @HAVE_GLIB_TRUE@	test-subset-cmap test-subset-glyf \
-@HAVE_GLIB_TRUE@	test-subset-hdmx test-subset-hmtx \
 @HAVE_GLIB_TRUE@	test-subset-os2 test-unicode test-version \
 @HAVE_GLIB_TRUE@	$(NULL) $(am__append_4) $(am__append_5) test-c \
 @HAVE_GLIB_TRUE@	test-cplusplus $(NULL)
@@ -867,10 +859,6 @@
 	@rm -f test-cplusplus$(EXEEXT)
 	$(AM_V_CXXLD)$(CXXLINK) $(test_cplusplus_OBJECTS) $(test_cplusplus_LDADD) $(LIBS)
 
-test-font$(EXEEXT): $(test_font_OBJECTS) $(test_font_DEPENDENCIES) $(EXTRA_test_font_DEPENDENCIES) 
-	@rm -f test-font$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(test_font_OBJECTS) $(test_font_LDADD) $(LIBS)
-
 test-object$(EXEEXT): $(test_object_OBJECTS) $(test_object_DEPENDENCIES) $(EXTRA_test_object_DEPENDENCIES) 
 	@rm -f test-object$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_object_OBJECTS) $(test_object_LDADD) $(LIBS)
@@ -903,10 +891,6 @@
 	@rm -f test-subset-glyf$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_subset_glyf_OBJECTS) $(test_subset_glyf_LDADD) $(LIBS)
 
-test-subset-hdmx$(EXEEXT): $(test_subset_hdmx_OBJECTS) $(test_subset_hdmx_DEPENDENCIES) $(EXTRA_test_subset_hdmx_DEPENDENCIES) 
-	@rm -f test-subset-hdmx$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(test_subset_hdmx_OBJECTS) $(test_subset_hdmx_LDADD) $(LIBS)
-
 test-subset-hmtx$(EXEEXT): $(test_subset_hmtx_OBJECTS) $(test_subset_hmtx_DEPENDENCIES) $(EXTRA_test_subset_hmtx_DEPENDENCIES) 
 	@rm -f test-subset-hmtx$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_subset_hmtx_OBJECTS) $(test_subset_hmtx_LDADD) $(LIBS)
@@ -932,7 +916,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-blob.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-buffer.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-common.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-font.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-object.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-ot-color.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-ot-tag.Po@am__quote@
@@ -940,7 +923,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-shape.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-subset-cmap.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-subset-glyf.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-subset-hdmx.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-subset-hmtx.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-subset-os2.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-version.Po@am__quote@
@@ -1267,13 +1249,6 @@
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
 	"$$tst" $(AM_TESTS_FD_REDIRECT)
-test-font.log: test-font$(EXEEXT)
-	@p='test-font$(EXEEXT)'; \
-	b='test-font'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
 test-object.log: test-object$(EXEEXT)
 	@p='test-object$(EXEEXT)'; \
 	b='test-object'; \
@@ -1308,13 +1283,6 @@
 	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
-test-subset-hdmx.log: test-subset-hdmx$(EXEEXT)
-	@p='test-subset-hdmx$(EXEEXT)'; \
-	b='test-subset-hdmx'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
 	"$$tst" $(AM_TESTS_FD_REDIRECT)
 test-subset-hmtx.log: test-subset-hmtx$(EXEEXT)
 	@p='test-subset-hmtx$(EXEEXT)'; \
