diff --git a/CMakeLists.txt b/CMakeLists.txt
index 420fc1c..b977464 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -596,7 +596,6 @@ if(WITH_OPTIM)
             endif()
         endif()
         if(HAVE_PCLMULQDQ_INTRIN)
-            add_definitions(-DX86_PCLMULQDQ_CRC)
             set(ZLIB_ARCH_SRCS ${ZLIB_ARCH_SRCS} ${ARCHDIR}/crc_folding.c)
             add_intrinsics_option("${PCLMULFLAG}")
             if(HAVE_SSE42_INTRIN)
