--- a/cmake/boost.cmake_orig	2017-02-26 18:54:13.297691646 +0100
+++ b/cmake/boost.cmake	2017-02-26 18:54:37.296778263 +0100
@@ -34,7 +34,7 @@
 SET(BOOST_PACKAGE_NAME "boost_1_59_0")
 SET(BOOST_TARBALL "${BOOST_PACKAGE_NAME}.tar.gz")
 SET(BOOST_DOWNLOAD_URL
-  "http://sourceforge.net/projects/boost/files/boost/1.59.0/${BOOST_TARBALL}"
+  "http://mirror.nienbo.com/boost/1.59.0/${BOOST_TARBALL}"
   )
 
 SET(OLD_PACKAGE_NAMES "boost_1_55_0 boost_1_56_0 boost_1_57_0 boost_1_58_0")

