--- patch/src/patch.c	2014-04-11 11:24:37.000000000 +0100
+++ patch/src/patch.c	2014-04-11 11:24:00.589844724 +0100
@@ -1975,5 +1975,6 @@ cleanup (void)
   remove_if_needed (TMPOUTNAME, &TMPOUTNAME_needs_removal);
   remove_if_needed (TMPPATNAME, &TMPPATNAME_needs_removal);
   remove_if_needed (TMPREJNAME, &TMPREJNAME_needs_removal);
-  forget_output_files ();
+  if (files_to_output)
+      forget_output_files ();
 }
