--- busybox-1.18.1/include/platform.h~	Mon Dec 20 00:41:03 2010
+++ busybox-1.18.1/include/platform.h	Thu Jan  6 20:03:58 2011
@@ -155,6 +155,7 @@
 # include <sex.h>
 #elif defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) \
    || defined(__APPLE__)
+# include <sys/param.h>		/* contains needed types for the following */
 # include <sys/resource.h>  /* rlimit */
 # include <machine/endian.h>
 # define bswap_64 __bswap64
