diff --git a/Makefile b/Makefile
index 989f8586..1f882fdd 100644
--- a/Makefile
+++ b/Makefile
@@ -15,9 +15,9 @@ include filelist.mk
 
 USE_AESNI	:= yes
 USE_AVX		:= yes
-USE_STATIC	:= no
+USE_STATIC	:= yes
 USE_MESHNET	:= no
-USE_UPNP	:= no
+USE_UPNP	:= yes
 
 ifeq ($(WEBSOCKETS),1)
 	NEEDED_CXXFLAGS += -DWITH_EVENTS
diff --git a/Makefile.linux b/Makefile.linux
index 782b0203..12deb616 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -20,7 +20,7 @@ else ifeq ($(shell expr match ${CXXVER} "4\.[7-9]"),3) # >= 4.7
 else ifeq ($(shell expr match ${CXXVER} "4\.6"),3) # = 4.6
 	NEEDED_CXXFLAGS += -std=c++0x
 else ifeq ($(shell expr match ${CXXVER} "[5-6]\.[0-9]"),3) # gcc >= 5.0
-	NEEDED_CXXFLAGS += -std=c++11
+	NEEDED_CXXFLAGS += -std=c++14
 else # not supported
 	$(error Compiler too old)
 endif
