diff --git a/src/IconButton.cc b/src/IconButton.cc
index d8205fb0..4b0b4236 100644
--- a/src/IconButton.cc
+++ b/src/IconButton.cc
@@ -25,7 +25,6 @@
 #include "IconbarTheme.hh"
 
 #include "Screen.hh"
-#include "Window.hh"
 
 #include "FbTk/App.hh"
 #include "FbTk/Command.hh"
@@ -59,10 +58,6 @@ IconButton::IconButton(const FbTk::FbWindow &parent,
     m_signals.join(m_win.titleSig(),
                    MemFunIgnoreArgs(m_title_update_timer, &FbTk::Timer::start));
 
-    if (m_win.fbwindow())
-        m_signals.join(m_win.fbwindow()->stateSig(),
-                       MemFunIgnoreArgs(*this, &IconButton::clientTitleChanged));
-
     m_signals.join(m_win.focusSig(),
                    MemFunIgnoreArgs(*this, &IconButton::reconfigAndClear));
 
@@ -255,10 +250,7 @@ void IconButton::clientTitleChanged() {
 
 void IconButton::setupWindow() {
     m_icon_window.clear();
-    FbTk::FbString title = m_win.title().logical();
-    if (m_win.fbwindow() && m_win.fbwindow()->isIconic())
-        title = IconbarTool::iconifiedPrefix() + title + IconbarTool::iconifiedSuffix();
-    setText(title);
+    setText(m_win.title());
     FbTk::TextButton::clear();
 }
 
diff --git a/src/IconbarTool.hh b/src/IconbarTool.hh
index 60440050..21c8078e 100644
--- a/src/IconbarTool.hh
+++ b/src/IconbarTool.hh
@@ -68,8 +68,6 @@ public:
 
     void setOrientation(FbTk::Orientation orient);
     FbTk::Container::Alignment alignment() const { return m_icon_container.alignment(); }
-    static std::string &iconifiedPrefix() { return s_iconifiedDecoration[0]; }
-    static std::string &iconifiedSuffix() { return s_iconifiedDecoration[1]; }
 
     const BScreen &screen() const { return m_screen; }
 private:
@@ -100,8 +98,6 @@ private:
     /// called when the list emits a signal
     void update(UpdateReason reason, Focusable *win);
 
-    void updateIconifiedPattern();
-
     void themeReconfigured();
 
     BScreen &m_screen;
@@ -122,7 +118,6 @@ private:
     FbTk::Resource<bool> m_rc_use_pixmap; ///< if iconbar should use win pixmap or not
     FbMenu m_menu;
     int m_alpha;
-    static std::string s_iconifiedDecoration[2];
 };
 
 #endif // ICONBARTOOL_HH
