diff -Naur intel-gpu-tools-1.16.orig/configure.ac intel-gpu-tools-1.16/configure.ac
--- intel-gpu-tools-1.16.orig/configure.ac	2016-09-13 04:06:46.308854895 +0300
+++ intel-gpu-tools-1.16/configure.ac	2016-09-13 04:06:58.560224844 +0300
@@ -315,10 +315,6 @@
 		 Makefile
 		 benchmarks/Makefile
 		 demos/Makefile
-		 docs/Makefile
-		 docs/reference/Makefile
-		 docs/reference/intel-gpu-tools/Makefile
-		 docs/reference/intel-gpu-tools/version.xml
 		 lib/Makefile
 		 lib/tests/Makefile
 		 man/Makefile
