diff --git a/configure.ac b/configure.ac
index b75f9c644d..44d3056082 100644
--- a/configure.ac
+++ b/configure.ac
@@ -715,7 +715,7 @@ oLIBS="$LIBS"
 LIBS="$LIBS $GDK_WLIBS"
 # The following is necessary for Linux libc-5.4.38
 AC_MSG_CHECKING(if iswalnum() and friends are properly defined)
-AC_TRY_LINK([#include <stdlib.h>],[
+AC_TRY_LINK([#include <stdlib.h>
 #if (defined(HAVE_WCTYPE_H) || defined(HAVE_WCHAR_H))
 #  ifdef HAVE_WCTYPE_H
 #    include <wctype.h>
@@ -726,7 +726,7 @@ AC_TRY_LINK([#include <stdlib.h>],[
 #  endif
 #else
 #  define iswalnum(c) ((wchar_t)(c) <= 0xFF && isalnum(c))
-#endif
+#endif],[
 iswalnum((wchar_t) 0);
 ], gdk_working_wctype=yes, gdk_working_wctype=no)
 LIBS="$oLIBS"
diff --git a/gdk/win32/gdkevents-win32.c b/gdk/win32/gdkevents-win32.c
index 9b09edd052..ac91fa64cb 100644
--- a/gdk/win32/gdkevents-win32.c
+++ b/gdk/win32/gdkevents-win32.c
@@ -2274,6 +2274,14 @@ gdk_event_translate (MSG  *msg,
 	  in_ime_composition)
 	break;
 
+      /* Ignore autorepeats on modifiers */
+      if (msg->message == WM_KEYDOWN &&
+          (msg->wParam == VK_MENU ||
+           msg->wParam == VK_CONTROL ||
+           msg->wParam == VK_SHIFT) &&
+           ((HIWORD(msg->lParam) & KF_REPEAT) >= 1))
+        break;
+
       if (!propagate (&window, msg,
 		      _gdk_display->keyboard_grab.window,
 		      _gdk_display->keyboard_grab.owner_events,
@@ -3622,8 +3630,7 @@ _gdk_events_queue (GdkDisplay *display)
   if (modal_win32_dialog != NULL)
     return;
   
-  while (!_gdk_event_queue_find_first (display) &&
-	 PeekMessageW (&msg, NULL, 0, 0, PM_REMOVE))
+  while (PeekMessageW (&msg, NULL, 0, 0, PM_REMOVE))
     {
       TranslateMessage (&msg);
       DispatchMessageW (&msg);
diff --git a/gtk/abicheck.sh b/gtk/abicheck.sh
index 23b2b0a098..0d033fb0a1 100755
--- a/gtk/abicheck.sh
+++ b/gtk/abicheck.sh
@@ -1,5 +1,5 @@
 #! /bin/sh
 
 cpp -DINCLUDE_VARIABLES -P -DG_OS_UNIX -DGTK_WINDOWING_X11 -DALL_FILES ${srcdir:-.}/gtk.symbols | sed -e '/^$/d' -e 's/ G_GNUC.*$//' -e 's/ PRIVATE//' | sort > expected-abi
-nm -D -g --defined-only .libs/libgtk-x11-2.0.so | cut -d ' ' -f 3 | egrep -v '^(__bss_start|_edata|_end)' | egrep -v '^g_cclosure_marshal' | sort > actual-abi
+nm -D -g --defined-only .libs/libgtk-x11-2.0.so | cut -d ' ' -f 3 | egrep -v '^(__bss_start|_edata|_end)' | sort > actual-abi
 diff -u expected-abi actual-abi && rm -f expected-abi actual-abi
diff --git a/gtk/gtk.symbols b/gtk/gtk.symbols
index 5f9ba56d47..6d5d5b6fa2 100644
--- a/gtk/gtk.symbols
+++ b/gtk/gtk.symbols
@@ -2510,16 +2510,10 @@ gtk_marshal_BOOLEAN__VOID
 gtk_marshal_ENUM__ENUM
 gtk_marshal_INT__POINTER
 gtk_marshal_INT__POINTER_CHAR_CHAR
-gtk_marshal_VOID__BOOLEAN
-gtk_marshal_VOID__BOXED
-gtk_marshal_VOID__ENUM
 gtk_marshal_VOID__ENUM_FLOAT
 gtk_marshal_VOID__ENUM_FLOAT_BOOLEAN
-gtk_marshal_VOID__INT
 gtk_marshal_VOID__INT_INT
 gtk_marshal_VOID__INT_INT_POINTER
-gtk_marshal_VOID__OBJECT
-gtk_marshal_VOID__POINTER
 gtk_marshal_VOID__POINTER_INT
 gtk_marshal_VOID__POINTER_INT_INT_POINTER_UINT_UINT
 gtk_marshal_VOID__POINTER_POINTER
@@ -2529,13 +2523,10 @@ gtk_marshal_VOID__POINTER_STRING_STRING
 gtk_marshal_VOID__POINTER_UINT
 gtk_marshal_VOID__POINTER_UINT_ENUM
 gtk_marshal_VOID__POINTER_UINT_UINT
-gtk_marshal_VOID__STRING
 gtk_marshal_VOID__STRING_INT_POINTER
-gtk_marshal_VOID__UINT
 gtk_marshal_VOID__UINT_POINTER_UINT_ENUM_ENUM_POINTER
 gtk_marshal_VOID__UINT_POINTER_UINT_UINT_ENUM
 gtk_marshal_VOID__UINT_STRING
-gtk_marshal_VOID__VOID
 #endif
 #endif
 #endif
diff --git a/gtk/gtkcalendar.c b/gtk/gtkcalendar.c
index 2dd68d6394..28baba16f1 100644
--- a/gtk/gtkcalendar.c
+++ b/gtk/gtkcalendar.c
@@ -689,6 +689,7 @@ gtk_calendar_init (GtkCalendar *calendar)
 #ifdef G_OS_WIN32
   wchar_t wbuffer[100];
 #else
+  static const char *month_format = NULL;
   char buffer[255];
   time_t tmp_time;
 #endif
@@ -714,7 +715,7 @@ gtk_calendar_init (GtkCalendar *calendar)
       {
 #ifndef G_OS_WIN32
 	tmp_time= (i+3)*86400;
-	strftime ( buffer, sizeof (buffer), "%a", gmtime (&tmp_time));
+	strftime (buffer, sizeof (buffer), "%a", gmtime (&tmp_time));
 	default_abbreviated_dayname[i] = g_locale_to_utf8 (buffer, -1, NULL, NULL, NULL);
 #else
 	if (!GetLocaleInfoW (GetThreadLocale (), LOCALE_SABBREVDAYNAME1 + (i+6)%7,
@@ -730,7 +731,21 @@ gtk_calendar_init (GtkCalendar *calendar)
       {
 #ifndef G_OS_WIN32
 	tmp_time=i*2764800;
-	strftime ( buffer, sizeof (buffer), "%B", gmtime (&tmp_time));
+	if (G_UNLIKELY (month_format == NULL))
+	  {
+	    buffer[0] = '\0';
+	    month_format = "%OB";
+	    strftime (buffer, sizeof (buffer), month_format, gmtime (&tmp_time));
+	    /* "%OB" is not supported in Linux with glibc < 2.27  */
+	    if (!strcmp (buffer, "%OB") || !strcmp (buffer, "OB") || !strcmp (buffer, ""))
+	      {
+		month_format = "%B";
+		strftime (buffer, sizeof (buffer), month_format, gmtime (&tmp_time));
+	      }
+	  }
+	else
+	  strftime (buffer, sizeof (buffer), month_format, gmtime (&tmp_time));
+
 	default_monthname[i] = g_locale_to_utf8 (buffer, -1, NULL, NULL, NULL);
 #else
 	if (!GetLocaleInfoW (GetThreadLocale (), LOCALE_SMONTHNAME1 + i,
diff --git a/gtk/queryimmodules.c b/gtk/queryimmodules.c
index a91ea47b63..e87391b28b 100644
--- a/gtk/queryimmodules.c
+++ b/gtk/queryimmodules.c
@@ -204,13 +204,19 @@ int main (int argc, char **argv)
             if (dir)
               {
                 const char *dent;
+                GList *list = NULL, *iterator = NULL;
 
                 while ((dent = g_dir_read_name (dir)))
+                  list = g_list_prepend (list, g_strdup (dent));
+
+                list = g_list_sort (list, (GCompareFunc) strcmp);
+                for (iterator = list; iterator; iterator = iterator->next)
                   {
-                    if (g_str_has_suffix (dent, SOEXT))
-                      error |= query_module (dirs[i], dent, contents);
+                    if (g_str_has_suffix (iterator->data, SOEXT))
+                      error |= query_module (dirs[i], iterator->data, contents);
                   }
 
+                g_list_free_full (list, g_free);
                 g_dir_close (dir);
               }
 
diff --git a/modules/printbackends/cups/gtkprintbackendcups.c b/modules/printbackends/cups/gtkprintbackendcups.c
index 24d4d8b487..2a000dc16f 100644
--- a/modules/printbackends/cups/gtkprintbackendcups.c
+++ b/modules/printbackends/cups/gtkprintbackendcups.c
@@ -2184,7 +2184,7 @@ cups_printer_handle_attribute (GtkPrintBackendCups *cups_backend,
   else
     {
       GTK_NOTE (PRINTING,
-		g_print ("CUPS Backend: Attribute %s ignored", ippGetName (attr)));
+		g_print ("CUPS Backend: Attribute %s ignored\n", ippGetName (attr)));
     }
 
 }
