--- libressl-2.4.2.orig/apps/openssl/Makefile.in	2016-08-01 01:08:10.000000000 +0200
+++ libressl-2.4.2/apps/openssl/Makefile.in	2016-08-22 19:40:55.019614700 +0200
@@ -844,7 +844,7 @@ install-exec-hook:
 	@if [ "@OPENSSLDIR@x" != "x" ]; then \
 		OPENSSLDIR="$(DESTDIR)/@OPENSSLDIR@"; \
 	else \
-		OPENSSLDIR="$(DESTDIR)/$(sysconfdir)/ssl"; \
+		OPENSSLDIR="$(DESTDIR)$(sysconfdir)/ssl"; \
 	fi; \
 	mkdir -p "$$OPENSSLDIR/certs"; \
 	for i in cert.pem openssl.cnf x509v3.cnf; do \
