From 592e35aa24c50ccaf8e08365087b41a340280435 Mon Sep 17 00:00:00 2001
From: Rob Loach <robloach@gmail.com>
Date: Wed, 29 Mar 2017 03:32:21 -0400
Subject: [PATCH] Use the "Commodore - 64" database for VICE

---
 dist/info/vice_x64_libretro.info | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dist/info/vice_x64_libretro.info b/dist/info/vice_x64_libretro.info
index c212298..4fb2de1 100644
--- a/dist/info/vice_x64_libretro.info
+++ b/dist/info/vice_x64_libretro.info
@@ -5,7 +5,7 @@ corename = "VICE"
 manufacturer = "Commodore"
 categories = "Emulator"
 systemname = "C64"
-database = "Commodore - C64"
+database = "Commodore - 64"
 license = "GPLv2"
 permissions = ""
 display_version = "v3.0"
