From 70fd808f7d502188a6bdade6b6b343fc97ecfac9 Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Wed, 8 May 2019 01:42:27 +0100
Subject: [PATCH] f2fs: Disable gcc9 -Wmaybe-uninitialized

---
 grub-core/fs/f2fs.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/grub-core/fs/f2fs.c b/grub-core/fs/f2fs.c
index 644653d..d0308fe 100644
--- a/grub-core/fs/f2fs.c
+++ b/grub-core/fs/f2fs.c
@@ -848,6 +848,11 @@ grub_f2fs_mount (grub_disk_t disk)
   return NULL;
 }
 
+#if __GNUC__ >= 9
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
+
 /* Guarantee inline_data was handled by caller. */
 static grub_disk_addr_t
 grub_f2fs_get_block (grub_fshelp_node_t node, grub_disk_addr_t block_ofs)
@@ -888,6 +893,10 @@ grub_f2fs_get_block (grub_fshelp_node_t node, grub_disk_addr_t block_ofs)
   return block_addr;
 }
 
+#if __GNUC__ >= 9
+#pragma GCC diagnostic pop
+#endif
+
 static grub_ssize_t
 grub_f2fs_read_file (grub_fshelp_node_t node,
                      grub_disk_read_hook_t read_hook, void *read_hook_data,
-- 
2.7.4

