diff --git a/src/main.conf b/src/main.conf
index 68870b28..694432f8 100644
--- a/src/main.conf
+++ b/src/main.conf
@@ -63,7 +63,7 @@
 # Allow connman to change the system hostname. This can
 # happen for example if we receive DHCP hostname option.
 # Default value is true.
-# AllowHostnameUpdates = true
+AllowHostnameUpdates = false
 
 # Keep only a single connected technology at any time. When a new
 # service is connected by the user or a better one is found according
@@ -100,7 +100,7 @@
 # the use of 6to4 will generally lead to a severe degradation of connection
 # quality. See RFC6343. Default value is false (as recommended by RFC6343
 # section 4.1).
-# Enable6to4 = false
+Enable6to4 = false
 
 # Enable use of http get as on online status check.
 # When a service is in a READY state, and is selected as default,
