diff -Naur qBittorrent-release-4.1.2/src/lang/lang.pri qBittorrent-release-4.1.2-lrelease/src/lang/lang.pri
--- qBittorrent-release-4.1.2/src/lang/lang.pri	2018-08-12 20:16:29.000000000 +0200
+++ qBittorrent-release-4.1.2-lrelease/src/lang/lang.pri	2018-08-16 19:40:41.926188661 +0200
@@ -11,7 +11,7 @@
     else:QMAKE_LRELEASE = $$[QT_INSTALL_BINS]/lrelease
     unix {
       equals(QT_MAJOR_VERSION, 5) {
-        !exists($$QMAKE_LRELEASE) { QMAKE_LRELEASE = lrelease-qt5 }
+        !exists($$QMAKE_LRELEASE) { QMAKE_LRELEASE = lrelease }
       }
     } else {
         !exists($$QMAKE_LRELEASE) { QMAKE_LRELEASE = lrelease }
