diff --git a/configure.py b/configure.py
index 9e23a5a..573219c 100755
--- a/configure.py
+++ b/configure.py
@@ -347,12 +347,12 @@ else:
              '-o', '/dev/null'],
             stdout=open(os.devnull, 'wb'), stderr=subprocess.STDOUT)
         if proc.wait() == 0:
-            cflags += ['-fdiagnostics-color']
+            cflags += ['-fdiagnostics-color=2']
     except:
         pass
     if platform.is_mingw():
         cflags += ['-D_WIN32_WINNT=0x0501']
-    ldflags = ['-L$builddir']
+    ldflags = ['-L$builddir -s']
     if platform.uses_usr_local():
         cflags.append('-I/usr/local/include')
         ldflags.append('-L/usr/local/lib')
