diff -Naur bison-3.4.1-orig/build-aux/missing bison-3.4.1/build-aux/missing
--- bison-3.4.1-orig/build-aux/missing	2019-06-03 03:51:50.761708031 +0300
+++ bison-3.4.1/build-aux/missing	2019-05-22 08:27:25.000000000 +0300
@@ -55,7 +55,7 @@
 
 Supported PROGRAM values:
   aclocal   autoconf  autoheader   autom4te  automake  makeinfo
-  bison     yacc      flex         lex       help2man
+  bison     yacc      flex         lex
 
 Version suffixes to PROGRAM as well as the prefixes 'gnu-', 'gnu', and
 'g' are ignored when checking the name.
diff -Naur bison-3.4.1-orig/configure bison-3.4.1/configure
--- bison-3.4.1-orig/configure	2019-06-03 03:51:50.761708031 +0300
+++ bison-3.4.1/configure	2019-05-22 08:28:29.000000000 +0300
@@ -11502,8 +11502,6 @@
    as_fn_error $? "perl not found" "$LINENO" 5
 fi
 
-HELP2MAN=${HELP2MAN-"${am_missing_run}help2man"}
-
 # Extract the first word of "xsltproc", so it can be a program name with args.
 set dummy xsltproc; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
diff -Naur bison-3.4.1-orig/configure.ac bison-3.4.1/configure.ac
--- bison-3.4.1-orig/configure.ac	2019-06-03 03:51:50.761708031 +0300
+++ bison-3.4.1/configure.ac	2019-05-19 19:32:10.000000000 +0300
@@ -228,7 +228,6 @@
 if test -z "$PERL"; then
    AC_MSG_ERROR([perl not found])
 fi
-AM_MISSING_PROG([HELP2MAN], [help2man])
 AC_PATH_PROG([XSLTPROC], [xsltproc])
 AC_SUBST([XSLTPROC])
 
diff -Naur bison-3.4.1-orig/Makefile.in bison-3.4.1/Makefile.in
--- bison-3.4.1-orig/Makefile.in	2019-06-03 03:51:50.785707712 +0300
+++ bison-3.4.1/Makefile.in	2019-05-22 08:28:33.000000000 +0300
@@ -1955,7 +1955,6 @@
 HAVE_WMEMSET = @HAVE_WMEMSET@
 HAVE__BOOL = @HAVE__BOOL@
 HAVE__EXIT = @HAVE__EXIT@
-HELP2MAN = @HELP2MAN@
 INCLUDE_NEXT = @INCLUDE_NEXT@
 INCLUDE_NEXT_AS_FIRST_DIRECTIVE = @INCLUDE_NEXT_AS_FIRST_DIRECTIVE@
 INSTALL = @INSTALL@
@@ -6777,18 +6776,6 @@
 @CROSS_COMPILING_FALSE@	      -e '/translation bugs/d'  >>doc/bison.help.tmp
 @CROSS_COMPILING_FALSE@	$(AM_V_at)$(top_srcdir)/build-aux/move-if-change doc/bison.help.tmp $@
 
-$(top_srcdir)/doc/bison.1: $(MAN_DEPS)
-	$(AM_V_GEN)$(HELP2MAN)			\
-	    --include=$(top_srcdir)/doc/bison.x	\
-	    --output=$@.tmp tests/bison
-	$(AM_V_at)if $(remove_time_stamp) $@ >$@a.tmp 2>/dev/null &&		\
-	   $(remove_time_stamp) $@.tmp | cmp $@a.tmp - >/dev/null 2>&1; then	\
-	  touch $@;								\
-	else									\
-	  mv $@.tmp $@;								\
-	fi
-	$(AM_V_at)rm -f $@*.tmp
-
 .gv.eps:
 	$(AM_V_GEN) $(MKDIR_P) `echo "./$@" | sed -e 's,/[^/]*$$,,'`
 	$(AM_V_at) $(DOT) -Gmargin=0 -Teps $< >$@.tmp
