diff --git a/Makefile b/Makefile
old mode 100644
new mode 100755
index 7701e162..1eef1653
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
-OPT_SUBDIRS ?= dbus gui mcstrans python restorecond sandbox semodule-utils
-SUBDIRS=libsepol libselinux libsemanage checkpolicy secilc policycoreutils $(OPT_SUBDIRS)
+fOPT_SUBDIRS ?= dbus gui mcstrans python restorecond sandbox semodule-utils
+SUBDIRS=libsepol libselinux $(OPT_SUBDIRS)
 PYSUBDIRS=libselinux libsemanage
 DISTCLEANSUBDIRS=libselinux libsemanage
 
