Last-Update: 2014-04-02
Forwarded: no
Bug-Debian: http://bugs.debian.org/413258
Author: Patrick Winnertz <winnie@debian.org>
Reviewed-by: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: Disable internal editor by default

--- a/src/setup.c
+++ b/src/setup.c
@@ -178,9 +178,9 @@
 
 /* If true use the internal viewer */
 int use_internal_view = 1;
 /* If set, use the builtin editor */
-int use_internal_edit = 1;
+int use_internal_edit = 0;
 
 #ifdef HAVE_CHARSET
 /* Numbers of (file I/O) and (input/display) codepages. -1 if not selected */
 int default_source_codepage = -1;
