Description: Test for libtoolize
 libtool is not used anywhere.
Bug-Debian: https://bugs.debian.org/761757
Author: Sebastian Ramacher <sramacher@debian.org>
Last-Update: 2014-09-28

--- a/make/configure.py
+++ b/make/configure.py
@@ -1471,7 +1471,7 @@
         autoconf = ToolProbe( 'AUTOCONF.exe', 'autoconf', abort=False )
         automake = ToolProbe( 'AUTOMAKE.exe', 'automake', abort=False )
         cmake    = ToolProbe( 'CMAKE.exe',    'cmake', abort=False )
-        libtool  = ToolProbe( 'LIBTOOL.exe',  'libtool', abort=False )
+        libtool  = ToolProbe( 'LIBTOOLIZE.exe',  'libtoolize', abort=False )
         pkgconfig = ToolProbe( 'PKGCONFIG.exe', 'pkg-config', abort=False )
 
         xcodebuild = ToolProbe( 'XCODEBUILD.exe', 'xcodebuild', abort=False )
