diff --git a/.config b/.config
old mode 100644
new mode 100755
index d06b0c2754..d46a9c7d5d
--- a/.config
+++ b/.config
@@ -2281,7 +2281,7 @@ CONFIG_DROPBEAR_ECC=y
 CONFIG_PACKAGE_firewall=y
 CONFIG_PACKAGE_fstools=y
 CONFIG_PACKAGE_fwtool=y
-CONFIG_PACKAGE_gconv-modules=y
+# CONFIG_PACKAGE_gconv-modules is not set
 CONFIG_PACKAGE_iucode-tool=y
 CONFIG_PACKAGE_jsonfilter=y
 # CONFIG_PACKAGE_lede-keyring is not set
diff --git a/package/my/i2pd/Makefile b/package/my/i2pd/Makefile
index 8f27bd6c1a..7355902ab8 100755
--- a/package/my/i2pd/Makefile
+++ b/package/my/i2pd/Makefile
@@ -17,7 +17,6 @@ PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
 PKG_SOURCE_URL:=https://github.com/PurpleI2P/i2pd.git
 PKG_SOURCE_VERSION:=$(PKG_VERSION)
-#PKG_MIRROR_HASH:=d389615366e603468c6d213e6545dfea99090d4a83aac121b6e2e4e5d0aac925
 PKG_LICENSE:=BSD-3-clause
 
 include $(INCLUDE_DIR)/package.mk
@@ -26,7 +25,7 @@ define Package/i2pd
 	SECTION:=net
 	CATEGORY:=Network
 	DEPENDS:=+libopenssl +boost +boost-system +boost-filesystem \
-	         +boost-program_options +boost-date_time +libatomic +zlib
+	         +boost-program_options +boost-date_time +libatomic +zlib +libminiupnpc
 	TITLE:=full-featured C++ implementation of I2P client
 	URL:=https://github.com/PurpleI2P/i2pd
 	USERID:=i2pd:i2pd
@@ -46,7 +45,7 @@ define Package/i2pd/conffiles
 endef
 
 TARGET_LDFLAGS+=-latomic
-MAKE_FLAGS+=USE_AESNI=no USE_AVX=no
+MAKE_FLAGS+=USE_AESNI=no USE_AVX=no USE_UPNP=yes
 
 define Package/i2pd/install
 	$(INSTALL_DIR) $(1)/usr/sbin
diff --git a/package/my/miniupnpc/Makefile b/package/my/miniupnpc/Makefile
index f03c84cd16..d656a9a37b 100755
--- a/package/my/miniupnpc/Makefile
+++ b/package/my/miniupnpc/Makefile
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=miniupnpc
-PKG_VERSION:=2.0.20171212
-PKG_RELEASE:=1
+PKG_VERSION:=2.0.20180203
+PKG_RELEASE:=2
 
 PKG_SOURCE_URL:=http://miniupnp.free.fr/files
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_HASH:=cef135ae46c12358a485e809619c01145238aceb10e6e8e2946ae40c4dbe477d
+#PKG_HASH:=cef135ae46c12358a485e809619c01145238aceb10e6e8e2946ae40c4dbe477d
 PKG_MAINTAINER:=Steven Barth <cyrus@openwrt.org>
 
 PKG_INSTALL:=1
diff --git a/toolchain/gcc/common.mk b/toolchain/gcc/common.mk
index 48da9f35ec..b1dce7dea1 100755
--- a/toolchain/gcc/common.mk
+++ b/toolchain/gcc/common.mk
@@ -114,6 +114,7 @@ GCC_CONFIGURE:= \
 		--disable-multilib \
 		--disable-libmpx \
 		--disable-nls \
+		--enable-checking=release \
 		--with-default-libstdcxx-abi=gcc4-compatible \
 		--with-tune=haswell \
 		--with-arch=bonnell \
diff --git a/toolchain/gcc/final (2018.01.18) [21.20.24].rar b/toolchain/gcc/final (2018.01.18) [21.20.24].rar
deleted file mode 100755
index 245767c4b9..0000000000
Binary files a/toolchain/gcc/final (2018.01.18) [21.20.24].rar and /dev/null differ
diff --git a/toolchain/gcc/final/Makefile b/toolchain/gcc/final/Makefile
index a490f60c83..636b62f910 100755
--- a/toolchain/gcc/final/Makefile
+++ b/toolchain/gcc/final/Makefile
@@ -22,6 +22,8 @@ GCC_CONFIGURE += \
 	--enable-libstdcxx-time \
 	--with-libelf=$(TOPDIR)/staging_dir/host
 
+TARGET_CXXFLAGS+=-std=gnu++98
+
 ifdef CONFIG_USE_MUSL
   GCC_MAKE += gcc_cv_libc_provides_ssp=yes
 endif
diff --git a/toolchain/gcc/patches/7.3.0 (2018.02.06) [08.20.44].rar b/toolchain/gcc/patches/7.3.0 (2018.02.06) [08.20.44].rar
deleted file mode 100755
index e96bd7173e..0000000000
Binary files a/toolchain/gcc/patches/7.3.0 (2018.02.06) [08.20.44].rar and /dev/null differ
diff --git a/toolchain/glibc/common.mk b/toolchain/glibc/common.mk
index 053c8ff083..e88d6353ed 100755
--- a/toolchain/glibc/common.mk
+++ b/toolchain/glibc/common.mk
@@ -4,13 +4,14 @@
 # This is free software, licensed under the GNU General Public License v2.
 # See /LICENSE for more information.
 #
+# https://github.com/bminor/glibc/tree/release/2.27/master
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=glibc
 PKG_VERSION:=2.27
 
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_VERSION:=00c5a2d
+PKG_SOURCE_VERSION:=0d40d0e
 PKG_SOURCE_URL:=https://github.com/bminor/glibc
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION)-$(PKG_SOURCE_VERSION).tar.xz
 
