--- softhddevice/audio.c.orig	2016-06-12 23:06:00.000000000 +0200
+++ softhddevice/audio.c	2016-06-14 10:45:15.854131610 +0200
@@ -1228,13 +1228,12 @@
 	//Debug(3, "audio: %s [\n", __FUNCTION__);
 	AlsaPCMHandle = NULL;		// other threads should check handle
 	snd_pcm_close(handle);
-	if (AudioAlsaCloseOpenDelay) {
-	    usleep(50 * 1000);		// 50ms delay for alsa recovery
+	int numRetries = 0;
+	while (!(handle = AlsaOpenPCM(passthrough)) && numRetries++ < 10) {
+	    usleep(20 * 1000);
 	}
-	// FIXME: can use multiple retries
-	if (!(handle = AlsaOpenPCM(passthrough))) {
+	if (!handle)
 	    return -1;
-	}
 	AlsaPCMHandle = handle;
 	//Debug(3, "audio: %s ]\n", __FUNCTION__);
     }
