--- softhddevice/codec.c	2014-05-16 08:43:04.686101960 +0200
+++ softhddevice/codec.c	2014-05-16 09:42:34.455238309 +0200
@@ -227,7 +227,7 @@
 	if (video_ctx->pkt) {
 	    frame->pkt_pts = video_ctx->pkt->pts;
 	} else {
-	    frame->pkt_pts = AV_NOPTS_VALUE;
+	    //frame->pkt_pts = AV_NOPTS_VALUE;
 	}
 	return 0;
     }
@@ -254,7 +254,7 @@
 	if (video_ctx->pkt) {
 	    frame->pkt_pts = video_ctx->pkt->pts;
 	} else {
-	    frame->pkt_pts = AV_NOPTS_VALUE;
+	    //frame->pkt_pts = AV_NOPTS_VALUE;
 	}
 	return 0;
     }
