From 66e5994f161a5969560bb4f55dd2f1d1951abf38 Mon Sep 17 00:00:00 2001
From: arvidn <arvid@cs.umu.se>
Date: Sun, 17 Jul 2016 11:32:01 -0700
Subject: [PATCH] fix crash in session::get_ip_filter when not having set one
Debian-Bug: http://bugs.debian.org/834630

Index: libtorrent-rasterbar/src/session_impl.cpp
===================================================================
--- libtorrent-rasterbar.orig/src/session_impl.cpp	2016-08-23 20:20:10.231408634 -0400
+++ libtorrent-rasterbar/src/session_impl.cpp	2016-08-23 20:20:10.227408578 -0400
@@ -1237,6 +1237,7 @@
 	ip_filter const& session_impl::get_ip_filter()
 	{
 		TORRENT_ASSERT(is_single_thread());
+		if (!m_ip_filter) m_ip_filter = boost::make_shared<ip_filter>();
 		return *m_ip_filter;
 	}
 
Index: libtorrent-rasterbar/test/test_ip_filter.cpp
===================================================================
--- libtorrent-rasterbar.orig/test/test_ip_filter.cpp	2016-08-23 20:20:10.231408634 -0400
+++ libtorrent-rasterbar/test/test_ip_filter.cpp	2016-08-23 20:20:10.227408578 -0400
@@ -35,6 +35,7 @@
 
 #include "test.hpp"
 #include "libtorrent/socket_io.hpp"
+#include "libtorrent/session.hpp"
 
 /*
 
@@ -89,6 +90,18 @@
 	}
 }
 
+TORRENT_TEST(session_get_ip_filter)
+{
+	using namespace libtorrent;
+	session ses;
+	ip_filter const& ipf = ses.get_ip_filter();
+#if TORRENT_USE_IPV6
+	TEST_EQUAL(boost::get<0>(ipf.export_filter()).size(), 1);
+#else
+	TEST_EQUAL(ipf.export_filter().size(), 1);
+#endif
+}
+
 TORRENT_TEST(ip_filter)
 {
 	using namespace libtorrent;
