diff -Naur Python3-3.6.4-orig/Misc/python-config.sh.in Python3-3.6.4/Misc/python-config.sh.in
--- Python3-3.6.4-orig/Misc/python-config.sh.in	2018-02-14 03:24:20.450308126 +0300
+++ Python3-3.6.4/Misc/python-config.sh.in	2017-12-19 07:53:56.000000000 +0300
@@ -33,9 +33,9 @@
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 exec_prefix_real=${prefix_real}
-includedir=$(echo "@includedir@" | sed "s#$prefix#$prefix_real#")
-libdir=$(echo "@libdir@" | sed "s#$prefix#$prefix_real#")
-CFLAGS=$(echo "@CFLAGS@" | sed "s#$prefix#$prefix_real#")
+includedir=$(echo "@includedir@" | sed "s#^$prefix#$prefix_real#")
+libdir=$(echo "@libdir@" | sed "s#$^prefix#$prefix_real#")
+CFLAGS=$(echo "@CFLAGS@" | sed "s#^$prefix#$prefix_real#")
 VERSION="@VERSION@"
 LIBM="@LIBM@"
 LIBC="@LIBC@"
@@ -49,7 +49,7 @@
 PY_ENABLE_SHARED="@PY_ENABLE_SHARED@"
 LDVERSION="@LDVERSION@"
 LIBDEST=${prefix_real}/lib/python${VERSION}
-LIBPL=$(echo "@LIBPL@" | sed "s#$prefix#$prefix_real#")
+LIBPL=$(echo "@LIBPL@" | sed "s#^$prefix#$prefix_real#")
 SO="@EXT_SUFFIX@"
 PYTHONFRAMEWORK="@PYTHONFRAMEWORK@"
 INCDIR="-I$includedir/python${VERSION}${ABIFLAGS}"
