--- xbmc-13.0rc1.orig/lib/cpluff/Makefile.am	2014-04-30 15:43:53.000000000 +0200
+++ xbmc-13.0rc1/lib/cpluff/Makefile.am	2014-05-05 13:03:15.329450196 +0200
@@ -6,7 +6,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = libcpluff loader console po test docsrc doc
+SUBDIRS = libcpluff loader console test docsrc doc
 DIST_SUBDIRS = $(SUBDIRS) examples
 
 EXTRA_DIST = COPYRIGHT.txt INSTALL.txt ChangeLog.txt Roadmap.txt autogen.sh plugin.xsd
