diff -Naur reaver-1.4.orig/configure reaver-1.4/configure
--- reaver-1.4.orig/configure	2016-09-13 21:16:37.706466966 +0300
+++ reaver-1.4/configure	2016-09-13 21:17:24.110361195 +0300
@@ -2781,7 +2781,7 @@
 
 
 CFLAGS="-Wall $CFLAGS"
-LDFLAGS="-ldl -lm -lpcap -lsqlite3 $LDFLAGS"
+LDFLAGS="-ldl -lm -lpcap-ng -lsqlite3 $LDFLAGS"
 
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for pcap_open_live in -lpcap" >&5
diff -Naur reaver-1.4.orig/configure.ac reaver-1.4/configure.ac
--- reaver-1.4.orig/configure.ac	2016-09-13 21:16:37.706466966 +0300
+++ reaver-1.4/configure.ac	2016-09-13 21:17:02.934903103 +0300
@@ -4,7 +4,7 @@
 AC_LANG(C)
 
 CFLAGS="-Wall $CFLAGS"
-LDFLAGS="-ldl -lm -lpcap -lsqlite3 $LDFLAGS"
+LDFLAGS="-ldl -lm -libcap-ng -lsqlite3 $LDFLAGS"
 
 AC_CHECK_LIB(pcap, pcap_open_live, [], [echo "error: pcap library not found!"; exit -1])
 AC_CHECK_LIB(sqlite3, sqlite3_open, [], [echo "error: sqlite3 library not found!"; exit -1])
