--- autoconf-2.69.orig/Makefile.in	2012-04-25 04:40:26.000000000 +0200
+++ autoconf-2.69/Makefile.in	2015-12-06 17:42:13.000000000 +0100
@@ -245,7 +245,7 @@ top_srcdir = @top_srcdir@
 # bin/ must be run first, as it builds executables needed for tests.
 # autom4te uses autotest.m4f to generate `testsuite', so build tests last.
 # Rules in man/ use scripts from both bin/ and tests/, so *it* goes last.
-SUBDIRS = bin . lib doc tests man
+SUBDIRS = bin . lib
 ACLOCAL_AMFLAGS = -I m4
 EXTRA_DIST = ChangeLog.0 ChangeLog.1 ChangeLog.2 ChangeLog.3 \
 	     BUGS COPYINGv3 COPYING.EXCEPTION \
