--- samba-4.2.1.orig/wscript_build	2014-12-20 09:07:55.000000000 -0600
+++ samba-4.2.1/wscript_build	2015-04-30 23:51:25.915427028 -0500
@@ -123,18 +123,7 @@ bld.RECURSE('libcli/samsync')
 bld.RECURSE('libcli/registry')
 bld.RECURSE('source4/lib/policy')
 bld.RECURSE('libcli/named_pipe_auth')
-
-if bld.CONFIG_GET('KRB5_VENDOR') in (None, 'heimdal'):
-    if bld.CONFIG_GET("HEIMDAL_KRB5_CONFIG") and bld.CONFIG_GET("USING_SYSTEM_KRB5"):
-        # When both HEIMDAL_KRB5_CONFIG and KRB5_CONFIG are set and not equal,
-        # it means one is Heimdal-specific (krb5-config.heimdal, for example)
-        # and there is system heimdal
-        bld.PROCESS_SEPARATE_RULE('system_heimdal')
-    else:
-        bld.PROCESS_SEPARATE_RULE('embedded_heimdal')
-else:
-    bld.PROCESS_SEPARATE_RULE('system_mitkrb5')
-
+bld.PROCESS_SEPARATE_RULE('embedded_heimdal')
 bld.RECURSE('libcli/smbreadline')
 bld.RECURSE('codepages')
 if bld.AD_DC_BUILD_IS_ENABLED():
