--- samba-4.2.1.orig/source3/lib/util.c	2015-01-16 09:18:39.000000000 +0100
+++ samba-4.2.1/source3/lib/util.c	2015-04-26 13:26:28.000000000 +0200
@@ -562,6 +562,7 @@ char *get_mydnsdomname(TALLOC_CTX *ctx)
 	}
 }
 
+#if 0
 #if (defined(HAVE_NETGROUP) && defined(WITH_AUTOMOUNT))
 /******************************************************************
  Remove any mount options such as -rsize=2048,wsize=2048 etc.
@@ -684,6 +685,7 @@ char *automount_lookup(TALLOC_CTX *ctx,
 }
 #endif /* WITH_NISPLUS_HOME */
 #endif
+#endif
 
 bool process_exists(const struct server_id pid)
 {
