diff --git a/project/cmake/scripts/linux/Install.cmake b/project/cmake/scripts/linux/Install.cmake
old mode 100644
new mode 100755
index d8df359..eb70b8b
--- a/project/cmake/scripts/linux/Install.cmake
+++ b/project/cmake/scripts/linux/Install.cmake
@@ -136,11 +136,6 @@ install(FILES ${CORE_SOURCE_DIR}/privacy-policy.txt
         DESTINATION ${datarootdir}/${APP_NAME_LC}
         COMPONENT kodi)
 
-# Install kodi-tools-texturepacker
-if(NOT WITH_TEXTUREPACKER)
-  install(PROGRAMS ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/texturepacker/TexturePacker
-          DESTINATION ${bindir}
-          COMPONENT kodi-tools-texturepacker)
 endif()
 
 # Install kodi-addon-dev headers
