--- a/xbmc/cores/VideoPlayer/VideoRenderers/DebugRenderer.cpp	2017-05-11 10:02:20.000000000 +0200
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/DebugRenderer.cpp	2017-05-15 14:09:27.039080736 +0200
@@ -108,7 +108,7 @@
 {
   std::vector<COverlay*> render;
   std::vector<SElement>& list = m_buffers[idx];
-  int posY = 0;
+  int posY = 550;
   for (std::vector<SElement>::iterator it = list.begin(); it != list.end(); ++it)
   {
     COverlay* o = nullptr;
