diff -Naur kodi-cb1d6d0-orig/project/cmake/modules/FindPython.cmake kodi-cb1d6d0/project/cmake/modules/FindPython.cmake
--- kodi-cb1d6d0-orig/project/cmake/modules/FindPython.cmake	2017-02-18 11:33:34.693550462 +0300
+++ kodi-cb1d6d0/project/cmake/modules/FindPython.cmake	2017-02-18 11:35:38.913133963 +0300
@@ -14,16 +14,12 @@
 find_path(PYTHON_INCLUDE_DIR NAMES Python.h PATHS ${PC_PYTHON_INCLUDE_DIRS} ${DEPENDS_PATH}/include/python2.7)
 
 if(KODI_DEPENDSBUILD)
-  find_library(FFI_LIBRARY ffi REQUIRED)
-  find_library(EXPAT_LIBRARY expat REQUIRED)
-  find_library(INTL_LIBRARY intl REQUIRED)
-  find_library(GMP_LIBRARY gmp REQUIRED)
 
   if(NOT CORE_SYSTEM_NAME STREQUAL android)
     set(PYTHON_DEP_LIBRARIES pthread dl util)
   endif()
 
-  set(PYTHON_LIBRARIES ${PYTHON_LIBRARY} ${FFI_LIBRARY} ${EXPAT_LIBRARY} ${INTL_LIBRARY} ${GMP_LIBRARY} ${PYTHON_DEP_LIBRARIES})
+  set(PYTHON_LIBRARIES ${PYTHON_LIBRARY} ${GMP_LIBRARY} ${PYTHON_DEP_LIBRARIES})
 else()
   find_package(PythonLibs 2.7 REQUIRED)
   list(APPEND PYTHON_LIBRARIES ${PC_PYTHON_STATIC_LIBRARIES})
