--- a/cmake/libutils.cmake	2017-02-27 11:33:58.651851125 +0100
+++ b/cmake/libutils.cmake	2017-02-27 11:35:53.451459310 +0100
@@ -78,13 +78,7 @@
   ELSE()
     SET(EXPORTS ${CMAKE_CURRENT_BINARY_DIR}/${TARGET}_exports_file.cc)
     SET(CONTENT)
-    FOREACH(FUNC ${API_FUNCTIONS})
-      SET(CONTENT "${CONTENT} extern void* ${FUNC}\;\n")
-    ENDFOREACH()
     SET(CONTENT "${CONTENT} void *${TARGET}_api_funcs[] = {\n")
-    FOREACH(FUNC ${API_FUNCTIONS})
-     SET(CONTENT "${CONTENT} &${FUNC},\n")
-    ENDFOREACH()
     SET(CONTENT "${CONTENT} (void *)0\n}\;")
     CONFIGURE_FILE_CONTENT(${CONTENT} ${EXPORTS})
     SET(${VAR} ${EXPORTS})
