--- rufus-d32e9dcb451b7d3a8d37241be73e5a3ae7837e8f/src/format.h	2015-04-04 17:53:29.013406930 -0400
+++ rufus-d32e9dcb451b7d3a8d37241be73e5a3ae7837e8f.patch/src/format.h	2015-04-04 17:53:49.985406527 -0400
@@ -17,7 +17,7 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-#include <Windows.h>
+#include <windows.h>
 #include <winioctl.h>				// for MEDIA_TYPE
 
 #pragma once
