Author: John E. Davis <jedsoft.org>
Description: Fix for segfault in slang/jed
Last-Updated: 2014-10-21
Forwarded: not-needed

Index: slang2-2.3.1/src/slstrops.c
===================================================================
--- slang2-2.3.1.orig/src/slstrops.c
+++ slang2-2.3.1/src/slstrops.c
@@ -1384,6 +1384,8 @@ static int strtrim_internal (int do_beg,
    if (cd.lut == NULL)
      return -1;
 
+   if (cd.lut == NULL)
+       return -1;
    status = arraymap_str_func_str (func_strtrim, &cd);
    if (free_lut) SLwchar_free_lut (cd.lut);
    return status;
