--- chromium-62.0.3202.9/chrome/browser/ui/libgtkui/gtk_ui.cc.gtk2	2017-09-06 21:06:25.000000000 +0200
+++ chromium-62.0.3202.9/chrome/browser/ui/libgtkui/gtk_ui.cc	2017-09-14 02:34:17.238146928 +0200
@@ -66,6 +66,7 @@
 
 #if GTK_MAJOR_VERSION == 2
 #include "chrome/browser/ui/libgtkui/native_theme_gtk2.h"  // nogncheck
+#include "chrome/browser/ui/libgtkui/nav_button_layout_manager.h"
 #elif GTK_MAJOR_VERSION == 3
 #include "chrome/browser/ui/libgtkui/native_theme_gtk3.h"  // nogncheck
 #include "chrome/browser/ui/libgtkui/nav_button_layout_manager_gtk3.h"  // nogncheck
