From 339e1788fa90004969c4c4d852f858ada5faa0e2 Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Sat, 29 Oct 2016 11:20:37 +1100
Subject: [PATCH 02/24] Make preemptible kernel default.

---
 kernel/Kconfig.preempt |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

Index: linux-4.11-ck2/kernel/Kconfig.preempt
===================================================================
--- linux-4.11-ck2.orig/kernel/Kconfig.preempt	2017-05-26 19:04:57.324244921 +1000
+++ linux-4.11-ck2/kernel/Kconfig.preempt	2017-05-26 19:04:57.324244921 +1000
@@ -1,7 +1,7 @@
 
 choice
 	prompt "Preemption Model"
-	default PREEMPT_NONE
+	default PREEMPT
 
 config PREEMPT_NONE
 	bool "No Forced Preemption (Server)"
@@ -17,7 +17,7 @@ config PREEMPT_NONE
 	  latencies.
 
 config PREEMPT_VOLUNTARY
-	bool "Voluntary Kernel Preemption (Desktop)"
+	bool "Voluntary Kernel Preemption (Nothing)"
 	help
 	  This option reduces the latency of the kernel by adding more
 	  "explicit preemption points" to the kernel code. These new
@@ -31,7 +31,8 @@ config PREEMPT_VOLUNTARY
 	  applications to run more 'smoothly' even when the system is
 	  under load.
 
-	  Select this if you are building a kernel for a desktop system.
+	  Select this for no system in particular (choose Preemptible
+	  instead on a desktop if you know what's good for you).
 
 config PREEMPT
 	bool "Preemptible Kernel (Low-Latency Desktop)"
