--- a/arch/x86/boot/compressed/eboot.c	2016-06-14 23:21:32.373442868 +0200
+++ b/arch/x86/boot/compressed/eboot.c	2016-06-14 23:21:43.366875394 +0200
@@ -127,7 +127,6 @@
 
 	status = efi_call_early(allocate_pool, EFI_LOADER_DATA, size, &rom);
 	if (status != EFI_SUCCESS) {
-		efi_printk(sys_table, "Failed to alloc mem for rom\n");
 		return status;
 	}
 
@@ -235,7 +234,6 @@
 
 	status = efi_call_early(allocate_pool, EFI_LOADER_DATA, size, &rom);
 	if (status != EFI_SUCCESS) {
-		efi_printk(sys_table, "Failed to alloc mem for rom\n");
 		return status;
 	}
 
