diff --git a/glob/glob.c b/glob/glob.c
index f3911bc..8adbde3 100644
--- a/glob/glob.c
+++ b/glob/glob.c
@@ -208,7 +208,7 @@ my_realloc (p, n)
 #endif /* __GNU_LIBRARY__ || __DJGPP__ */
 
 
-#if !defined __alloca && !defined __GNU_LIBRARY__
+#if !defined __alloca && defined __GNU_LIBRARY__
 
 # ifdef	__GNUC__
 #  undef alloca
diff --git a/job.c b/job.c
index 39966f6..1f316b3 100644
--- a/job.c
+++ b/job.c
@@ -2758,7 +2758,11 @@ construct_command_argv_internal (char *line, char **restp, const char *shell,
 
   /* See if it is safe to parse commands internally.  */
   if (shell == 0)
-    shell = default_shell;
+    {
+      shell = default_shell;
+      if (shellflags == 0)
+	shellflags = "-c";
+    }
 #ifdef WINDOWS32
   else if (strcmp (shell, default_shell))
   {
@@ -3370,13 +3374,14 @@ construct_command_argv_internal (char *line, char **restp, const char *shell,
 #endif
             if (PRESERVE_BSNL)
               {
-                *(ap++) = '\\';
+                *(ap++) = '\'';
                 /* Only non-batch execution needs another backslash,
                    because it will be passed through a recursive
                    invocation of this function.  */
                 if (!batch_mode_shell)
                   *(ap++) = '\\';
                 *(ap++) = '\n';
+                *(ap++) = '\'';
               }
             ++p;
             continue;
diff --git a/main.c b/main.c
index 4163622..71753a7 100644
--- a/main.c
+++ b/main.c
@@ -1982,6 +1982,20 @@ main (int argc, char **argv, char **envp)
     }
 #endif
 
+#ifdef PIPE_BUF
+  if (job_slots > PIPE_BUF)
+#elif defined _POSIX_PIPE_BUF
+  if (job_slots > _POSIX_PIPE_BUF)
+#else
+  if (job_slots > 512)
+#endif
+    {
+      O (error, NILF,
+            _("More parallel jobs (-jN) than this platform can handle requested."));
+      O (error, NILF, _("Resetting to single job (-j1) mode."));
+      job_slots = 1;
+    }
+
 #ifdef MAKE_JOBSERVER
   /* If we have >1 slot but no jobserver-fds, then we're a top-level make.
      Set up the pipe and install the fds option for our children.  */
diff --git a/makeint.h b/makeint.h
index 6e936a8..4f2f453 100644
--- a/makeint.h
+++ b/makeint.h
@@ -549,7 +549,7 @@ long int lseek ();
 #endif  /* Not GNU C library or POSIX.  */
 
 #ifdef  HAVE_GETCWD
-# if !defined(VMS) && !defined(__DECC)
+# if !defined(VMS) && !defined(__DECC) && !defined(getcwd)
 char *getcwd ();
 # endif
 #else
