--- chromium-64.0.3282.71/third_party/angle/src/compiler/translator/Types.h.gcc7	2018-01-04 00:07:24.000000000 +0100
+++ chromium-64.0.3282.71/third_party/angle/src/compiler/translator/Types.h	2018-01-07 10:11:45.477781558 +0100
@@ -236,7 +236,7 @@
     {
     }
 
-    TBasicType getBasicType() const { return type; }
+    constexpr TBasicType getBasicType() const { return type; }
     void setBasicType(TBasicType t);
 
     TPrecision getPrecision() const { return precision; }
--- chromium-64.0.3282.71/cc/paint/raw_memory_transfer_cache_entry.cc.gcc7	2018-01-04 00:05:53.000000000 +0100
+++ chromium-64.0.3282.71/cc/paint/raw_memory_transfer_cache_entry.cc	2018-01-07 10:43:22.160924166 +0100
@@ -4,6 +4,8 @@
 
 #include "cc/paint/raw_memory_transfer_cache_entry.h"
 
+#include <cstring>
+
 namespace cc {
 
 ClientRawMemoryTransferCacheEntry::ClientRawMemoryTransferCacheEntry(
@@ -25,7 +27,7 @@
   if (size != data_.size())
     return false;
 
-  memcpy(data, data_.data(), size);
+  std::memcpy(data, data_.data(), size);
   return true;
 }
 
@@ -46,7 +48,7 @@
                                                      size_t size,
                                                      uint8_t* data) {
   data_.resize(size);
-  memcpy(data_.data(), data, size);
+  std::memcpy(data_.data(), data, size);
   return true;
 }
 
