--- chromium-61.0.3163.59/third_party/WebKit/Source/platform/wtf/typed_arrays/ArrayBufferContents.h.bak	2017-08-23 21:06:14.000000000 +0200
+++ chromium-61.0.3163.59/third_party/WebKit/Source/platform/wtf/typed_arrays/ArrayBufferContents.h	2017-08-28 21:08:42.366482005 +0200
@@ -63,7 +63,7 @@
           allocation_length_(0),
           data_(data),
           data_length_(0),
-          kind_(AllocationKind::kNormal),
+          kind_(ArrayBufferContents::AllocationKind::kNormal),
           deleter_(deleter) {}
     DataHandle(void* allocation_base,
                size_t allocation_length,
@@ -94,11 +94,11 @@
              reinterpret_cast<uintptr_t>(allocation_base_) +
                  allocation_length_);
       switch (kind_) {
-        case AllocationKind::kNormal:
+        case ArrayBufferContents::AllocationKind::kNormal:
           DCHECK(deleter_);
           deleter_(data_);
           return;
-        case AllocationKind::kReservation:
+        case ArrayBufferContents::AllocationKind::kReservation:
           ReleaseReservedMemory(allocation_base_, allocation_length_);
           return;
       }
--- chromium-61.0.3163.59/third_party/webrtc/modules/audio_processing/aec3/aec_state.cc.bak	2017-08-23 21:08:14.000000000 +0200
+++ chromium-61.0.3163.59/third_party/webrtc/modules/audio_processing/aec3/aec_state.cc	2017-08-29 13:07:50.789475942 +0200
@@ -10,7 +10,7 @@
 
 #include "modules/audio_processing/aec3/aec_state.h"
 
-#include <math.h>
+#include <cmath>
 
 #include <numeric>
 #include <vector>
