--- chromium-68.0.3440.7/chrome/browser/ui/libgtkui/native_theme_gtk2.cc.gcc7	2018-05-31 12:45:32.000000000 +0200
+++ chromium-68.0.3440.7/chrome/browser/ui/libgtkui/native_theme_gtk2.cc	2018-06-02 18:47:10.156567303 +0200
@@ -161,6 +161,9 @@
       return GetTextColor(GetMenuItem(), INSENSITIVE);
     case kColorId_MenuBackgroundColor:
       return GetBgColor(GetMenu(), NORMAL);
+    case ui::NativeTheme::kColorId_TouchableMenuItemLabelColor:
+    case ui::NativeTheme::kColorId_ActionableSubmenuVerticalSeparatorColor:
+      return kInvalidColorIdColor;
 
     // Label
     case kColorId_LabelEnabledColor:
