diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 26171d6afc..f3f9d8004e 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -272,7 +272,8 @@ config("compiler") {
     } else if (is_posix && !is_chromeos && !is_nacl) {
       # TODO(phajdan.jr): Use -fstack-protector-strong when our gcc supports it.
       # See also https://crbug.com/533294
-      cflags += [ "--param=ssp-buffer-size=4" ]
+      cflags += [ "--param=ssp-buffer-size=32" ]
+      cflags += [ "-Wl,--copy-dt-needed-entries" ]
 
       # The x86 toolchain currently has problems with stack-protector.
       if (is_android && current_cpu == "x86") {
