diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
old mode 100644
new mode 100755
index 3dbf872f70..f9cd01c285
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -256,7 +256,7 @@ config("compiler") {
     } else if ((is_posix && !is_chromeos && !is_nacl) || is_fuchsia) {
       # TODO(phajdan.jr): Use -fstack-protector-strong when our gcc supports it.
       # See also https://crbug.com/533294
-      cflags += [ "--param=ssp-buffer-size=4" ]
+      cflags += [ "--param=ssp-buffer-size=32" ]
 
       # The x86 toolchain currently has problems with stack-protector.
       if (is_android && current_cpu == "x86") {
@@ -696,9 +696,28 @@ config("compiler_cpu_abi") {
     if (current_cpu == "x64") {
       cflags += [
         "-m64",
-        "-march=x86-64",
+        "-march=westmere",
+        "-mmmx",
+        "-msse",
+        "-msse2",
+        "-msse3",
+        "-mssse3", 
+        "-msse4.2", 
+        "-msse4.1",
+        "-mfpmath=sse",
+        "-Wno-attributes",
+        "-Wno-comment",
+        "-Wno-strict-overflow",
+        "-Wno-deprecated-declarations",
+        "-fdiagnostics-color=always",
+        "-fno-unwind-tables",
+        "-fno-asynchronous-unwind-tables",
+        "-DNO_UNWIND_TABLES",
+      ]
+      ldflags += [
+        "-m64",
+        "-march=westmere",
       ]
-      ldflags += [ "-m64" ]
     } else if (current_cpu == "x86") {
       cflags += [ "-m32" ]
       ldflags += [ "-m32" ]

