--- a/third_party/tcmalloc/chromium/src/base/linux_syscall_support.h	2017-10-30 12:45:10.272681148 +0100
+++ b/third_party/tcmalloc/chromium/src/base/linux_syscall_support.h	2017-10-30 13:16:35.895907914 +0100
@@ -752,11 +752,11 @@
    */
   #define LSS_RETURN(type, res)                                               \
     do {                                                                      \
-      if ((unsigned long)(res) >= (unsigned long)(-4095)) {                   \
-        LSS_ERRNO = -(res);                                                   \
+      if ((unsigned long)res >= (unsigned long)(-4095)) {                     \
+        LSS_ERRNO = -res;                                                     \
         res = -1;                                                             \
       }                                                                       \
-      return (type) (res);                                                    \
+      return (type)res;                                                       \
     } while (0)
   #elif defined(__mips__)
   /* On MIPS, failing system calls return -1, and set errno in a
