diff -Naur chromium-55.0.2883.75-orig/build/config/compiler/BUILD.gn chromium-55.0.2883.75/build/config/compiler/BUILD.gn
--- chromium-55.0.2883.75-orig/build/config/compiler/BUILD.gn	2017-02-26 16:57:05.804416938 +0300
+++ chromium-55.0.2883.75/build/config/compiler/BUILD.gn	2017-02-26 17:00:06.206380979 +0300
@@ -201,7 +201,7 @@
     } else if (is_posix && !is_chromeos && !is_nacl) {
       # TODO(phajdan.jr): Use -fstack-protector-strong when our gcc supports it.
       # See also https://crbug.com/533294
-      cflags += [ "--param=ssp-buffer-size=4" ]
+      cflags += [ " " ]
 
