diff --git a/ui/accessibility/platform/atk_util_auralinux_gtk2.cc b/ui/accessibility/platform/atk_util_auralinux_gtk2.cc
old mode 100644
new mode 100755
index ac11b5626f..786c7e9a62
--- a/ui/accessibility/platform/atk_util_auralinux_gtk2.cc
+++ b/ui/accessibility/platform/atk_util_auralinux_gtk2.cc
@@ -58,6 +58,8 @@ void FinishAccessibilityInitOnMainThread(
   init_func();
 }
 
+namespace ui {
+
 bool AtkUtilAuraLinux::PlatformShouldEnableAccessibility() {
   std::unique_ptr<base::Environment> env(base::Environment::Create());
   std::string gtk_modules;
