diff -Naur chromium-61.0.3163.91-orig/build/config/compiler/BUILD.gn chromium-61.0.3163.91/build/config/compiler/BUILD.gn
--- chromium-61.0.3163.91-orig/build/config/compiler/BUILD.gn	2017-09-19 18:36:32.231551807 +0300
+++ chromium-61.0.3163.91/build/config/compiler/BUILD.gn	2017-09-19 18:42:13.177102501 +0300
@@ -203,19 +203,19 @@
       if (is_debug) {
         cflags += [ "-fstack-protector-strong" ]
       } else {
-        cflags += [ "-fstack-protector" ]
+        cflags += [ "-fno-plt" ]
       }
     } else if (is_posix && !is_chromeos && !is_nacl) {
       # TODO(phajdan.jr): Use -fstack-protector-strong when our gcc supports it.
       # See also https://crbug.com/533294
-      cflags += [ "--param=ssp-buffer-size=4" ]
+      cflags += [ "-fstack-protector-strong" ]
 
       # The x86 toolchain currently has problems with stack-protector.
       if (is_android && current_cpu == "x86") {
         cflags += [ "-fno-stack-protector" ]
       } else if (current_os != "aix") {
         # Not available on aix.
-        cflags += [ "-fstack-protector" ]
+        cflags += [ "-fno-plt" ]
       }
     }
 
@@ -575,7 +575,7 @@
     if (current_cpu == "x64") {
       cflags += [
         "-m64",
-        "-march=x86-64",
+        "-march=core-avx-i",
       ]
       ldflags += [ "-m64" ]
     } else if (current_cpu == "x86") {
