diff --git a/third_party/android_protobuf/BUILD.gn b/third_party/android_protobuf/BUILD.gn
index 1f29810a4d..1160f537ba 100644
--- a/third_party/android_protobuf/BUILD.gn
+++ b/third_party/android_protobuf/BUILD.gn
@@ -8,7 +8,7 @@ if (current_toolchain == host_toolchain) {
       "-Wno-null-conversion",
       "-Wno-tautological-undefined-compare",
       "-Wno-unused-function",
-      "-Wno-unused-local-typedef",
+      "-Wno-unused-local-typedefs",
     ]
   }
 
diff --git a/third_party/harfbuzz-ng/BUILD.gn b/third_party/harfbuzz-ng/BUILD.gn
index c8bd9c6f9d..962fa1bc4a 100644
--- a/third_party/harfbuzz-ng/BUILD.gn
+++ b/third_party/harfbuzz-ng/BUILD.gn
@@ -32,7 +32,7 @@ if (use_system_harfbuzz) {
         # Harfbuzz uses unused typedefs for its static asserts (and its
         # static asserts are strange enough that they can't be replaced
         # by static_assert).
-        "-Wno-unused-local-typedef",
+        "-Wno-unused-local-typedefs",
       ]
     }
     if (is_win) {
diff --git a/third_party/mesa/BUILD.gn b/third_party/mesa/BUILD.gn
index 148b7ec8fa..abf0b44a6c 100644
--- a/third_party/mesa/BUILD.gn
+++ b/third_party/mesa/BUILD.gn
@@ -136,7 +136,7 @@ config("mesa_internal_warnings") {
       "-Wno-mismatched-tags",  # Fixed upstream.
 
       # mesa's STATIC_ASSERT() macro expands to an ununused typedef.
-      "-Wno-unused-local-typedef",
+      "-Wno-unused-local-typedefs",
 
       # mesa's flex scanner code uses "register" a lot.
       "-Wno-deprecated-register",
diff --git a/third_party/yasm/BUILD.gn b/third_party/yasm/BUILD.gn
index ab2f9414bd..77b263ea0f 100644
--- a/third_party/yasm/BUILD.gn
+++ b/third_party/yasm/BUILD.gn
@@ -237,7 +237,7 @@ if (current_toolchain == host_toolchain) {
     if (is_clang) {
       cflags = [
         # reg3264type in x86expr.c is unused.
-        "-Wno-unused-local-typedef",
+        "-Wno-unused-local-typedefs",
       ]
     } else if (is_linux) {
       cflags = [
