Last-Update: 2014-04-02
Forwarded: not-needed
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: enable "Group undo" by default (mcedit)

--- a/src/editor/edit.c
+++ b/src/editor/edit.c
@@ -93,9 +93,9 @@
 int option_state_full_filename = 1;
 
 int enable_show_tabs_tws = 1;
 int option_check_nl_at_eof = 0;
-int option_group_undo = 0;
+int option_group_undo = 1;
 int show_right_margin = 0;
 
 char *option_backup_ext = NULL;
 char *option_filesize_threshold = NULL;
