--- GConf-3.2.6.orig/Makefile.in        2013-01-22 02:59:40.000000000 +0100
+++ GConf-3.2.6/Makefile.in     2014-06-03 15:06:15.000000000 +0200
@@ -398,9 +398,9 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 with_sysconfsubdir = @with_sysconfsubdir@
-SUBDIRS = gconf backends po doc examples $(am__append_1) \
+SUBDIRS = gconf backends doc examples $(am__append_1) \
        $(am__append_2)
-DIST_SUBDIRS = tests gconf backends po doc examples defaults gsettings
+DIST_SUBDIRS = tests gconf backends doc examples defaults gsettings
 EXTRA_DIST = \
        TODO                \
        gtk-doc.make        \

