diff --git a/Makefile b/Makefile
old mode 100644
new mode 100755
index 12f7f75..e94dc13
--- a/Makefile
+++ b/Makefile
@@ -1,17 +1,17 @@
 # Config
-CC = /usr/bin/gcc
+CC = x86_64-openelec-linux-gnu-gcc
 CFLAGS = -Wall
 INSTALL = /usr/bin/install
 PREFIX = /usr
-XBMC = /var/lib/xbmc/.xbmc/userdata/keymaps
+XBMC = /var/lib/kodi/.kodi/userdata/keymaps
 
 # Install stuff
-keymap = xbmc/keyboard.xml
+keymap = kodi/keyboard.xml
 service = systemd/ps3remote.service
 systemd = /usr/lib/systemd/system
 bin = $(PREFIX)/bin
 
-LDLIBS = -ludev
+LDLIBS = -ludev -lsystemd
 srcs = hid.c hidraw.c main.c uinput.c
 srcs := $(addprefix src/,$(srcs))
 objs = $(srcs:.c=.o)
@@ -29,7 +29,7 @@ install: $(target)
 	$(INSTALL) -o root -m 644 $(service) $(systemd)
 
 installxbmc: install
-	$(INSTALL) -o xbmc -m 644 $(keymap) $(XBMC)
+	$(INSTALL) -o kodi -m 644 $(keymap) $(XBMC)
 
 uninstall:
 	rm -f $(bin)/$(target)
