diff --git a/Makefile b/Makefile
index 47d1f33..2deed84 100644
--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,9 @@
 # Config
-CC = /usr/bin/gcc
+CC = x86_64-libreelec-linux-gnu-gcc
 CFLAGS = -Wall
 INSTALL = /usr/bin/install
 PREFIX = /usr
-XBMC = /var/lib/xbmc/.xbmc/userdata/keymaps
+XBMC = /storage/.kodi/userdata/keymaps
 
 # Install stuff
 keymap = xbmc/keyboard.xml
@@ -18,7 +18,7 @@ objs = $(srcs:.c=.o)
 target = ps3remote
 
 # Rules
-all: CFLAGS += -DDEBUG=0 -O3
+all: CFLAGS += -DDEBUG=0 -O3 -pipe
 all: $(target)
 
 debug: CFLAGS += -g -DDEBUG -Og
