diff -Naur rpcbind-1.2.5-orig/systemd/rpcbind.service.in rpcbind-1.2.5/systemd/rpcbind.service.in
--- rpcbind-1.2.5-orig/systemd/rpcbind.service.in	2018-10-10 06:22:55.087580148 +0300
+++ rpcbind-1.2.5/systemd/rpcbind.service.in	2018-08-15 17:51:19.000000000 +0300
@@ -12,7 +12,7 @@
 [Service]
 Type=notify
 # distro can provide a drop-in adding EnvironmentFile=-/??? if needed.
-ExecStart=@_sbindir@/rpcbind $RPCBIND_OPTIONS -w -f
+ExecStart=@_sbindir@/rpcbind $RPCBIND_OPTIONS -f
 
 [Install]
 WantedBy=multi-user.target
