From 68caf4ec0563fdffe8904850312d38e1e30d02be Mon Sep 17 00:00:00 2001
From: fritsch <fritsch@kodi.tv>
Date: Sat, 26 Aug 2017 22:41:31 +0200
Subject: [PATCH 1/3] gen8_mfd: Check ptr before dereferencing

---
 src/gen8_mfd.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/gen8_mfd.c b/src/gen8_mfd.c
index c93c26df..78719b2f 100644
--- a/src/gen8_mfd.c
+++ b/src/gen8_mfd.c
@@ -1725,10 +1725,11 @@ gen8_mfd_vc1_pred_pipe_state(VADriverContextP ctx,
     if (gen7_mfd_context->reference_surface[0].surface_id != VA_INVALID_ID) {
         if (picture_type == 1 || picture_type == 2) { /* P/B picture */
             struct gen7_vc1_surface *gen7_vc1_surface = gen7_mfd_context->reference_surface[0].obj_surface->private_data;
-
-            intensitycomp_single_fwd = gen7_vc1_surface->intensity_compensation;
-            luma_scale1 = gen7_vc1_surface->luma_scale;
-            luma_shift1 = gen7_vc1_surface->luma_shift;
+            if (gen7_vc1_surface) {
+                intensitycomp_single_fwd = gen7_vc1_surface->intensity_compensation;
+                luma_scale1 = gen7_vc1_surface->luma_scale;
+                luma_shift1 = gen7_vc1_surface->luma_shift;
+            }
         }
     }
 

From a6863055737e05f9f70fc1206616f9905460431c Mon Sep 17 00:00:00 2001
From: fritsch <fritsch@kodi.tv>
Date: Sat, 26 Aug 2017 23:01:56 +0200
Subject: [PATCH 2/3] gen7_mfd: Check ptr before dereferencing

---
 src/gen7_mfd.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/gen7_mfd.c b/src/gen7_mfd.c
index fe4d66d3..1e6911e2 100644
--- a/src/gen7_mfd.c
+++ b/src/gen7_mfd.c
@@ -1681,10 +1681,11 @@ gen7_mfd_vc1_pred_pipe_state(VADriverContextP ctx,
     if (gen7_mfd_context->reference_surface[0].surface_id != VA_INVALID_ID) {
         if (picture_type == 1 || picture_type == 2) { /* P/B picture */
             struct gen7_vc1_surface *gen7_vc1_surface = gen7_mfd_context->reference_surface[0].obj_surface->private_data;
-
-            intensitycomp_single_fwd = gen7_vc1_surface->intensity_compensation;
-            luma_scale1 = gen7_vc1_surface->luma_scale;
-            luma_shift1 = gen7_vc1_surface->luma_shift;
+            if (gen7_vc1_surface) {
+                intensitycomp_single_fwd = gen7_vc1_surface->intensity_compensation;
+                luma_scale1 = gen7_vc1_surface->luma_scale;
+                luma_shift1 = gen7_vc1_surface->luma_shift;
+            }
         }
     }
 

From b08d913ef5a998e17098a725a88eb6ef92d8d4ac Mon Sep 17 00:00:00 2001
From: fritsch <Peter.Fruehberger@gmail.com>
Date: Mon, 4 Sep 2017 07:06:50 +0200
Subject: [PATCH 3/3] gen6_mfd: Check ptr before dereferencing

---
 src/gen6_mfd.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/gen6_mfd.c b/src/gen6_mfd.c
index cb9713aa..bcc455be 100644
--- a/src/gen6_mfd.c
+++ b/src/gen6_mfd.c
@@ -1619,10 +1619,11 @@ gen6_mfd_vc1_pred_pipe_state(VADriverContextP ctx,
     if (gen6_mfd_context->reference_surface[0].surface_id != VA_INVALID_ID) {
         if (picture_type == 1 || picture_type == 2) { /* P/B picture */
             struct gen6_vc1_surface *gen6_vc1_surface = gen6_mfd_context->reference_surface[0].obj_surface->private_data;
-
-            intensitycomp_single_fwd = gen6_vc1_surface->intensity_compensation;
-            luma_scale1 = gen6_vc1_surface->luma_scale;
-            luma_shift1 = gen6_vc1_surface->luma_shift;
+            if (gen6_vc1_surface) {
+                intensitycomp_single_fwd = gen6_vc1_surface->intensity_compensation;
+                luma_scale1 = gen6_vc1_surface->luma_scale;
+                luma_shift1 = gen6_vc1_surface->luma_shift;
+            }
         }
     }
 
