diff -up M2Crypto-0.21.1/tests/test_x509.py.x509_name M2Crypto-0.21.1/tests/test_x509.py
--- M2Crypto-0.21.1/tests/test_x509.py.x509_name	2014-06-09 17:15:56.649087188 +0200
+++ M2Crypto-0.21.1/tests/test_x509.py	2014-06-09 21:29:43.516145260 +0200
@@ -142,7 +142,9 @@ class X509TestCase(unittest.TestCase):
         cn.set_data("Hello There!")
         assert cn.get_data().as_text() == "Hello There!", cn.get_data().as_text()
 
-        assert n.as_hash() == 1697185131
+        # OpenSSL 1.0.1h switched from encoding strings as PRINTABLESTRING (the
+        # first hash value) to UTF8STRING (the second one) 
+        assert n.as_hash() in (1697185131, 1370641112), n.as_hash()
         
         self.assertRaises(IndexError, lambda: n[100])
         self.assert_(n[10])
