--- libressl-2.4.2.orig/Makefile.in	2016-08-01 01:08:10.000000000 +0200
+++ libressl-2.4.2/Makefile.in	2016-08-22 18:34:29.881419200 +0200
@@ -361,7 +361,7 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = crypto ssl tls include apps tests man
+SUBDIRS = crypto ssl tls include
 ACLOCAL_AMFLAGS = -I m4
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcrypto.pc libssl.pc libtls.pc openssl.pc
