From 37c650849d48d545539c91c970e499cb68940321 Mon Sep 17 00:00:00 2001
From: Chips <darcelf@gmail.com>
Date: Sun, 10 Jul 2016 21:36:26 +0200
Subject: [PATCH] Solve JIT crash by setting O2 to the crashing function

---
 src/drawing.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/drawing.cpp b/src/drawing.cpp
index 562d4f9..6e34e16 100644
--- a/src/drawing.cpp
+++ b/src/drawing.cpp
@@ -1634,7 +1634,7 @@ static __inline__ void pfield_doline (int lineno)
 }
 
 
-void init_row_map (void)
+__attribute__((optimize("O2"))) void init_row_map (void)
 {
   int i, j;
 
