diff -Naur a/Makefile.am b/Makefile.am
--- a/Makefile.am	2016-11-11 09:29:04.000000000 -0800
+++ b/Makefile.am	2016-11-12 10:04:00.825202361 -0800
@@ -200,17 +200,11 @@
 
 TESTS = $(unit_tests)
 
-manual_pages = doc/iwmon.1
-
-dist_man_MANS = $(manual_pages)
-
-EXTRA_DIST = $(manual_pages:.1=.txt)
-
 AM_CFLAGS = -fvisibility=hidden
 
 MAINTAINERCLEANFILES = Makefile.in configure config.h.in aclocal.m4
 
-DISTCLEANFILES = $(BUILT_SOURCES) $(unit_tests) $(manual_pages)
+DISTCLEANFILES = $(BUILT_SOURCES) $(unit_tests)
 
 BUILT_SOURCES = ell/internal
 
@@ -236,8 +230,5 @@
 		fi \
 	done > $@
 
-%.1: %.txt
-	$(AM_V_GEN)$(A2X) --doctype manpage --format manpage $(srcdir)/$<
-
 maintainer-clean-local:
 	-rm -rf build-aux ell
