commit 9cc43c85f4e979499c8d5fae28fd9aba1111dfe5
Author: Lukas Rusak <lorusak@gmail.com>
Date:   Sat Nov 12 12:28:58 2016 -0800

    makefile: Install iwd-dbus.conf into /etc/dbus-1/system.d

diff --git a/Makefile.am b/Makefile.am
index a56f5bf..2f63be0 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -86,6 +86,9 @@ src_iwd_LDADD = ell/libell-internal.la -ldl
 
 sysconf_DATA = src/iwd.conf
 
+dbuspolicydir=/etc/dbus-1/system.d
+dbuspolicy_DATA = src/iwd-dbus.conf
+
 client_iwctl_SOURCES = client/main.c
 client_iwctl_LDADD = ell/libell-internal.la
 
