diff -Naur a/monitor/main.c b/monitor/main.c
--- a/monitor/main.c	2017-02-23 09:21:18.000000000 -0800
+++ b/monitor/main.c	2017-02-28 23:42:56.351184791 -0800
@@ -78,7 +78,7 @@
 	const struct genlmsghdr *genlmsg = data;
 	const struct nlattr *nla;
 	char name[GENL_NAMSIZ];
-	uint16_t id = GENL_ID_GENERATE;
+	uint16_t id = 0;
 
 	if (nlmon)
 		return;
@@ -101,7 +101,7 @@
 		}
 	}
 
-	if (id == GENL_ID_GENERATE)
+	if (id == 0)
 		return;
 
 	if (!strcmp(name, NL80211_GENL_NAME)) {
@@ -694,7 +694,7 @@
 	const char *analyze_path = NULL;
 	const char *ifname = NULL;
 	struct iwmon_interface monitor_interface = { };
-	uint16_t nl80211_family = GENL_ID_GENERATE;
+	uint16_t nl80211_family = 0;
 	struct l_signal *signal;
 	sigset_t mask;
 	int exit_status;
@@ -732,7 +732,7 @@
 				}
 				nl80211_family = strtoul(optarg, NULL, 10);
 			}
-			if (nl80211_family == GENL_ID_GENERATE) {
+			if (nl80211_family == 0) {
 				usage();
 				return EXIT_FAILURE;
 			}
diff -Naur a/monitor/nlmon.c b/monitor/nlmon.c
--- a/monitor/nlmon.c	2017-02-23 09:21:18.000000000 -0800
+++ b/monitor/nlmon.c	2017-02-28 23:42:56.350184782 -0800
@@ -4160,7 +4160,7 @@
 	const struct genlmsghdr *genlmsg = data;
 	const struct nlattr *nla;
 	char name[GENL_NAMSIZ];
-	uint16_t id = GENL_ID_GENERATE;
+	uint16_t id = 0;
 
 	if (genlmsg->cmd != CTRL_CMD_NEWFAMILY)
 		return;
@@ -4177,7 +4177,7 @@
 		}
 	}
 
-	if (id == GENL_ID_GENERATE)
+	if (id == 0)
 		return;
 
 	if (!strcmp(name, NL80211_GENL_NAME))
