diff --git a/src/defaults.py b/src/defaults.py
old mode 100644
new mode 100755
index 0324421..b31f7d4
--- a/src/defaults.py
+++ b/src/defaults.py
@@ -40,8 +40,8 @@ bluetooth = {
 services = {
     'ENABLED': True,
     'KERNEL_CMD': '/proc/cmdline',
-    'SAMBA_NMDB': '/usr/sbin/nmbd',
-    'SAMBA_SMDB': '/usr/sbin/smbd',
+    'SAMBA_NMDB': '/usr/bin/nmbd',
+    'SAMBA_SMDB': '/usr/bin/smbd',
     'D_SAMBA_WORKGROUP': 'WORKGROUP',
     'D_SAMBA_SECURE': '0',
     'D_SAMBA_USERNAME': 'libreelec',
