--- util-linux-2.26/term-utils/agetty.c.org	2015-02-27 09:44:28.005955842 -0500
+++ util-linux-2.26/term-utils/agetty.c	2015-02-27 09:45:28.454200135 -0500
@@ -1379,7 +1379,7 @@
 	 * Wait for a while, then read everything the modem has said so far and
 	 * try to extract the speed of the dial-in call.
 	 */
-	sleep(1);
+	usleep(1000);
 	if ((nread = read(STDIN_FILENO, buf, sizeof(buf) - 1)) > 0) {
 		buf[nread] = '\0';
 		for (bp = buf; bp < buf + nread; bp++)
@@ -1728,7 +1728,7 @@
 	 * the baud rate).
 	 */
 	if ((op->flags & F_VCONSOLE) == 0)
-		sleep(1);
+		usleep(1000);
 	tcflush(STDIN_FILENO, TCIFLUSH);
 
 	eightbit = (op->flags & F_EIGHTBITS);
--- util-linux-2.28/term-utils/agetty.c~	2016-04-04 08:29:50.000000000 +0000
+++ util-linux-2.28/term-utils/agetty.c	2016-10-29 13:44:35.655558375 +0000
@@ -36,6 +36,7 @@
 #include <netdb.h>
 #include <ifaddrs.h>
 #include <net/if.h>
+#include <malloc.h>
 
 #include "strutils.h"
 #include "all-io.h"
@@ -1804,6 +1805,7 @@
 		/* Always show login prompt. */
 		write_all(STDOUT_FILENO, LOGIN, sizeof(LOGIN) - 1);
 	}
+	malloc_trim(0);
 }
 
 /* Select next baud rate. */
