diff -purN util-linux-2.26.1.org/term-utils/agetty.c util-linux-2.26.1/term-utils/agetty.c
--- util-linux-2.26.1.org/term-utils/agetty.c	2015-03-12 05:20:04.843410762 -0400
+++ util-linux-2.26.1/term-utils/agetty.c	2015-03-14 11:59:11.184831054 -0400
@@ -1594,13 +1594,19 @@ static void print_issue_file(struct opti
 #ifdef	ISSUE
 	FILE *fd;
 #endif
+
+        putchar('\033');
+        putchar('[');
+        putchar('0');
+        putchar('m');
+        
 	if ((op->flags & F_NONL) == 0) {
 		/* Issue not in use, start with a new line. */
 		write_all(STDOUT_FILENO, "\r\n", 2);
 	}
 
 #ifdef	ISSUE
-	if ((op->flags & F_ISSUE) && (fd = fopen(op->issue, "r"))) {
+	if (op->flags & F_ISSUE) {
 		int c, oflag = tp->c_oflag;	    /* Save current setting. */
 
 		if ((op->flags & F_VCONSOLE) == 0) {
@@ -1608,12 +1614,22 @@ static void print_issue_file(struct opti
 			tp->c_oflag |= (ONLCR | OPOST);
 			tcsetattr(STDIN_FILENO, TCSADRAIN, tp);
 		}
-
-		while ((c = getc(fd)) != EOF) {
-			if (c == '\\')
-				output_special_char(getc(fd), op, tp, fd);
-			else
-				putchar(c);
+		fd = fopen(op->issue, "r");
+		if (fd) {
+			while ((c = getc(fd)) != EOF) {
+				if (c == '\\')
+					output_special_char(getc(fd), op, tp, fd);
+				else
+					putchar(c);
+			}
+		} else {
+			output_special_char('S', op, tp, fd);
+			putchar(' ');
+			output_special_char('n', op, tp, fd);
+			putchar(' ');
+			output_special_char('l', op, tp, fd);
+			putchar('\r');
+			putchar('\n');
 		}
 		fflush(stdout);
 
@@ -1623,7 +1639,8 @@ static void print_issue_file(struct opti
 			/* Wait till output is gone. */
 			tcsetattr(STDIN_FILENO, TCSADRAIN, tp);
 		}
-		fclose(fd);
+		if (fd)
+			fclose(fd);
 	}
 #endif	/* ISSUE */
 }
@@ -2216,6 +2233,9 @@ static void output_iface_ip(struct ifadd
  */
 static char *get_escape_argument(FILE *fd, char *buf, size_t bufsz)
 {
+	if (fd == NULL)
+		return NULL;
+
 	size_t i = 0;
 	int c = fgetc(fd);
 
