--- util-linux-2.29/libblkid/src/probe.c~	2016-11-02 12:57:31.000000000 +0000
+++ util-linux-2.29/libblkid/src/probe.c	2017-04-15 17:42:44.274109543 +0000
@@ -1805,15 +1805,6 @@
 }
 
 
-/* converts DCE UUID (uuid[16]) to human readable string
- * - the @len should be always 37 */
-#ifdef HAVE_LIBUUID
-void blkid_unparse_uuid(const unsigned char *uuid, char *str,
-			size_t len __attribute__((__unused__)))
-{
-	uuid_unparse(uuid, str);
-}
-#else
 void blkid_unparse_uuid(const unsigned char *uuid, char *str, size_t len)
 {
 	snprintf(str, len,
@@ -1824,7 +1815,6 @@
 		uuid[8], uuid[9],
 		uuid[10], uuid[11], uuid[12], uuid[13], uuid[14],uuid[15]);
 }
-#endif
 
 /* like uuid_is_null() from libuuid, but works with arbitrary size of UUID */
 int blkid_uuid_is_empty(const unsigned char *buf, size_t len)
