diff --git a/chrome/browser/ui/libgtkui/settings_provider_gsettings.cc b/chrome/browser/ui/libgtkui/settings_provider_gsettings.cc
index 0809f7a..81110f2 100644
--- a/chrome/browser/ui/libgtkui/settings_provider_gsettings.cc
+++ b/chrome/browser/ui/libgtkui/settings_provider_gsettings.cc
@@ -44,10 +44,13 @@ SettingsProviderGSettings::SettingsProviderGSettings(GtkUi* delegate)
                                          base::nix::DESKTOP_ENVIRONMENT_CINNAMON
                                      ? kCinnamonPreferencesSchema
                                      : kGnomePreferencesSchema;
+  GSettingsSchema *schema =
+    g_settings_schema_source_lookup(g_settings_schema_source_get_default(),
+				    settings_schema, FALSE);
 
-  if (!g_settings_schema_source_lookup(g_settings_schema_source_get_default(),
-                                       settings_schema, FALSE) ||
-      !(button_settings_ = g_settings_new(settings_schema))) {
+  if (!schema ||
+      !(button_settings_ = g_settings_new(settings_schema)) ||
+      !g_settings_schema_has_key(schema, kButtonLayoutKey)) {
     ParseAndStoreButtonValue(kDefaultButtonString);
   } else {
     // Get the inital value of the keys we're interested in.
