---
 Configurations/unix-Makefile.tmpl |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/Configurations/unix-Makefile.tmpl
+++ b/Configurations/unix-Makefile.tmpl
@@ -164,7 +164,8 @@ HTMLDIR=$(DOCDIR)/html
 # MANSUFFIX is for the benefit of anyone who may want to have a suffix
 # appended after the manpage file section number.  "ssl" is popular,
 # resulting in files such as config.5ssl rather than config.5.
-MANSUFFIX=
+MANSUFFIX=ssl
+MANSECTION=SSL
 HTMLSUFFIX=html
 
 
@@ -630,7 +631,7 @@ UNINSTALL_DOCS=\
 	@\
 	OUTSUFFIX='.$${SEC}$(MANSUFFIX)'; \
 	OUTTOP="$(DESTDIR)$(MANDIR)"; \
-	GENERATE='pod2man --name=$$NAME --section=$$SEC --center=OpenSSL --release=$(VERSION)'; \
+	GENERATE='pod2man --name=$$NAME --section="$$SEC""$(MANSECTION)" --center=OpenSSL --release=$(VERSION)'; \
 	$(PROCESS_PODS)
 
 uninstall_man_docs:
