--- glib-2.56.0/gio/inotify/inotify-missing.c~	2017-07-13 23:03:39.000000000 +0000
+++ glib-2.56.0/gio/inotify/inotify-missing.c	2018-04-06 02:02:35.427232385 +0000
@@ -27,7 +27,7 @@
 #include "inotify-path.h"
 #include "glib-private.h"
 
-#define SCAN_MISSING_TIME 4 /* 1/4 Hz */
+#define SCAN_MISSING_TIME 120 /* 1/4 Hz */
 
 static gboolean im_debug_enabled = FALSE;
 #define IM_W if (im_debug_enabled) g_warning
