Description: zero new space allocated in prompt buffer
Origin: 8d4c98540de9bcdba8565facc91fbc45855d27e2
Author: Paulo Andrade <pandrade@redhat.com>
Reviewed-By: Daniel Shahaf <d.s@daniel.shahaf.name>

--- a/Src/prompt.c
+++ b/Src/prompt.c
@@ -920,6 +920,7 @@
 	if(need & 255)
 	    need = (need | 255) + 1;
 	bv->buf = realloc(bv->buf, bv->bufspc += need);
+	memset(bv->buf + bv->bufspc - need, 0, need);
 	bv->bp = bv->buf + bo;
 	if(bo1 != -1)
 	    bv->bp1 = bv->buf + bo1;
