diff -Naur strace-4.16=orig/configure.ac strace-4.16/configure.ac
--- strace-4.16=orig/configure.ac	2017-02-21 19:27:40.289567376 +0300
+++ strace-4.16/configure.ac	2017-02-21 19:32:27.725561427 +0300
@@ -30,12 +30,10 @@
 # THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 AC_PREREQ(2.57)
-AC_INIT([strace],
-	m4_esyscmd([./git-version-gen .tarball-version]),
-	[strace-devel@lists.sourceforge.net],
 	[strace],
 	[https://strace.io])
 AC_COPYRIGHT([Copyright (C) 1999-2017 The strace developers.])
+AC_INIT([strace],[4.8])
 AC_CONFIG_SRCDIR([strace.c])
 AC_CONFIG_AUX_DIR([.])
 AC_CONFIG_HEADERS([config.h])
