diff -Naur strace-4.17-orig/configure.ac strace-4.17/configure.ac
--- strace-4.17-orig/configure.ac	2017-05-25 20:42:32.355042753 +0300
+++ strace-4.17/configure.ac	2017-05-25 20:43:42.192871490 +0300
@@ -31,13 +31,9 @@
 # THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 AC_PREREQ(2.57)
-AC_INIT([strace],
-	m4_esyscmd([./git-version-gen .tarball-version]),
-	[strace-devel@lists.sourceforge.net],
-	[strace],
-	[https://strace.io])
 m4_define([copyright_year], m4_esyscmd([./copyright-year-gen .year]))
 AC_COPYRIGHT([Copyright (C) 1999-]copyright_year[ The strace developers.])
+AC_INIT([strace],[4.8])
 AC_CONFIG_SRCDIR([strace.c])
 AC_CONFIG_AUX_DIR([.])
 AC_CONFIG_HEADERS([config.h])
